---
title: "Count body and end signal for the various samples and tracks"
author: "Manfred Schmid"
date: "9 August 2017"
output: html_document
---

```{r setup}
knitr::opts_chunk$set(echo = TRUE, fig.path = 'Figures_body_end_counting/', dev='pdf')
```

```{r, message=FALSE,warning=FALSE}
suppressWarnings(library('tidyverse'))
suppressWarnings(library('magrittr'))
suppressWarnings(library('knitr'))
suppressWarnings(library('broom'))

```


# Create body and end annotations

Define gene body region: TSS to TES-200bp
Define gene end region: TES+/-200bp


Annotations used are the classical Steinmetz transcription unit annotations lifted to UCSC release *sacCer3*.

Body and ends are derived using awk and bedtools
```{bash, eval=FALSE}
#!/bin/bash

## for raw data

########################################################################################################################
####    ANNOTATIONS FOR BODY vs END
########################################################################################################################


# annotations for body and end counts
bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin.bed"
awk '{OFS="\t"}{if($6=="+"){$3-=200;if(($3-$2)>100) print $0}}' $bed | cut -f1-4 > ${bed/.bed/_body_endsm200_plus.bed}
awk '{OFS="\t"}{if($6=="-"){$2+=200;if(($3-$2)>100) print $0}}' $bed | cut -f1-4 > ${bed/.bed/_body_endsm200_minus.bed}
awk '{OFS="\t"}{if($6=="+"){$2=$3-200;$3+=200; print $0}}' $bed | cut -f1-4 > ${bed/.bed/_ends_pm200_plus.bed}
awk '{OFS="\t"}{if($6=="-"){$3=$2+200;$2-=200; print $0}}' $bed | cut -f1-4 > ${bed/.bed/_ends_pm200_minus.bed}


head -3 $bed
#chrI    5074    6237    ST3634:SUTs:SUT432:SUT432       0       -
#chrI    7275    9260    ST3635:ORF-T:YAL067C:SEO1       0       -
#chrI    9367    9600    ST0001:SUTs:SUT001:SUT001       0       +


head -3 ${bed/.bed/_body_endsm200_plus.bed}
#chrI    30071   30704   ST0002:CUTs:CUT001:CUT001
#chrI    31151   32784   ST0003:ORF-T:YAL062W:GDH3
#chrI    33359   34696   ST0004:ORF-T:YAL061W:BDH2
##note: ST0001 is gone because body is shorter than 100bp!

head -3 ${bed/.bed/_body_endsm200_minus.bed}
#chrI    5274    6237    ST3634:SUTs:SUT432:SUT432
#chrI    7475    9260    ST3635:ORF-T:YAL067C:SEO1
#chrI    10931   11140   ST3636:CUTs:CUT436:CUT436


head -3 ${bed/.bed/_ends_pm200_plus.bed}
#chrI    9400    9800    ST0001:SUTs:SUT001:SUT001
#chrI    30704   31104   ST0002:CUTs:CUT001:CUT001
#chrI    32784   33184   ST0003:ORF-T:YAL062W:GDH3
##note: ST0001 ends are still counted


head -3 ${bed/.bed/_ends_pm200_minus.bed}
#chrI    4874    5274    ST3634:SUTs:SUT432:SUT432
#chrI    7075    7475    ST3635:ORF-T:YAL067C:SEO1
#chrI    10531   10931   ST3636:CUTs:CUT436:CUT436
```


Original Steinmetz annotations never overlap on the same strand. Here I extend the ends by 200bp so overlaps are likely to occur. Since gene end reads dominate in those libraries, we don't want to count overlapping "gene end" regions as "gene body." Hence we remove all ends+/-200bp covered regions from gene body regions.  


First inspect the overlaps:  
```{bash, eval = FALSE}
##inspect the overlaps
bedtools intersect -wo -a ${bed/.bed/_body_endsm200_plus.bed} -b ${bed/.bed/_ends_pm200_plus.bed} > body_vs_end_intersect_plus.txt
bedtools intersect -wo -a ${bed/.bed/_body_endsm200_minus.bed} -b ${bed/.bed/_ends_pm200_minus.bed} > body_vs_end_intersect_minus.txt

head body_vs_end_intersect_plus.txt
#chrI    35095   36192   ST0005:ORF-T:YAL060W:BDH1       chrI    34696   35096   ST0004:ORF-T:YAL061W:BDH2       1
#chrI    36543   37128   ST0006:ORF-T:YAL059W:ECM1       chrI    36192   36592   ST0005:ORF-T:YAL060W:BDH1       49
#chrI    37407   38832   ST0007:ORF-T:YAL058W:CNE1       chrI    37128   37528   ST0006:ORF-T:YAL059W:ECM1       121
#chrI    39215   41768   ST0008:ORF-T:YAL056W:GPB2       chrI    38832   39232   ST0007:ORF-T:YAL058W:CNE1       17
#chrI    42159   42632   ST0009:ORF-T:YAL055W:PEX22      chrI    41768   42168   ST0008:ORF-T:YAL056W:GPB2       9
#chrI    62789   65406   ST0015:ORF-T:YAL041W:CDC24      chrI    62446   62846   ST0014:ORF-T:YAL042W:ERV46      57
#chrI    79693   80446   ST0020:ORF-T:YAL034W-A:MTW1     chrI    79334   79734   ST0019:ORF-T:YAL035W:FUN12      41
#chrI    94605   95342   ST0024:ORF-T:YAL027W:YAL027W    chrI    94286   94686   ST0023:ORF-T:YAL028W:FRT2       81
#chrI    130699  131884  ST0029:ORF-T:YAL012W:CYS3       chrI    130364  130764  ST0028:ORF-T:YAL013W:DEP1       65
#chrI    132195  133908  ST0030:ORF-T:YAL011W:SWC3       chrI    131884  132284  ST0029:ORF-T:YAL012W:CYS3       89

wc -l body_vs_end_intersect_plus.txt
#1011 body_vs_end_intersect_plus.txt

head body_vs_end_intersect_minus.txt
#chrI    33275   34380   ST3640:SUTs:SUT435:SUT435       chrI    34179   34579   ST3641:CUTs:CUT438:CUT438       201
#chrI    42939   43548   ST3643:ORF-T:YAL054C:ACS1       chrI    43347   43747   ST3644:ORF-T:YAL054C:ACS1       201
#chrI    51937   52658   ST3645:ORF-T:YAL049C:YAL049C    chrI    52457   52857   ST3646:ORF-T:YAL048C:GEM1       201
#chrI    52857   54842   ST3646:ORF-T:YAL048C:GEM1       chrI    54753   55153   ST3647:ORF-T:YAL047C:SPC72      89
#chrI    55153   56922   ST3647:ORF-T:YAL047C:SPC72      chrI    56721   57121   ST3648:ORF-T:YAL046C:YAL046C    201
#chrI    58073   58482   ST3649:ORF-T:YAL044C:GCV3       chrI    58409   58809   ST3650:ORF-T:YAL043C:PTA1       73
#chrI    65665   67850   ST3651:ORF-T:YAL040C:CLN3       chrI    67649   68049   ST3652:CUTs:CUT440:CUT440       201
#chrI    101681  105986  ST3661:ORF-T:YAL024C:LTE1       chrI    105961  106361  ST3662:ORF-T:YAL023C:PMT2       25
#chrI    106361  108666  ST3662:ORF-T:YAL023C:PMT2       chrI    108657  109057  ST3663:ORF-T:YAL022C:FUN26      9
#chrI    109057  110466  ST3663:ORF-T:YAL022C:FUN26      chrI    110417  110817  ST3664:ORF-T:YAL021C:CCR4       49

wc -l body_vs_end_intersect_minus.txt
#1074 body_vs_end_intersect_minus.txt

wc -l ${bed/.bed/_body_endsm200_plus.bed}
#    3381 /Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_body_endsm200_plus.bed
wc -l ${bed/.bed/_body_endsm200_minus.bed}
#    3385 /Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_body_endsm200_minus.bed
```


--> roughly 1/3 of all body regions have overlapping end regions!  



#### filter out ends_pm200 from body annotations:  
```{bash, eval = FALSE}
bedtools subtract -a ${bed/.bed/_body_endsm200_plus.bed} -b ${bed/.bed/_ends_pm200_plus.bed} > ${bed/.bed/_body_endsm200_cleaned_plus.bed}
bedtools subtract -a ${bed/.bed/_body_endsm200_minus.bed} -b ${bed/.bed/_ends_pm200_minus.bed} > ${bed/.bed/_body_endsm200_cleaned_minus.bed}

#sanity check
##plus strand for ST0006 overlaps with extended end of ST0005
##body:
grep ST0006 ${bed/.bed/_body_endsm200_plus.bed}
#chrI    36543   37128   ST0006:ORF-T:YAL059W:ECM1

##ends:
grep ST0005 ${bed/.bed/_ends_pm200_plus.bed}
#chrI    36192   36592   ST0005:ORF-T:YAL060W:BDH1

##cleaned_body:
grep ST0006 ${bed/.bed/_body_endsm200_cleaned_plus.bed}
#chrI    36592   37128   ST0006:ORF-T:YAL059W:ECM1




##minus strand for ST3640
grep ST3640 ${bed/.bed/_body_endsm200_minus.bed}
#chrI    33275   34380   ST3640:SUTs:SUT435:SUT435


##ends:
grep ST3641 ${bed/.bed/_ends_pm200_minus.bed}
#chrI    34179   34579   ST3641:CUTs:CUT438:CUT438

##cleaned_body:
grep ST3640 ${bed/.bed/_body_endsm200_cleaned_minus.bed}
#chrI    33275   34179   ST3640:SUTs:SUT435:SUT435
```


#### inspection of the cleaned annotation

```{bash, eval = FALSE}
### total nr of regions
wc -l ${bed/.bed/_body_endsm200_cleaned_plus.bed}
#    3359 /Users/schmidm/Documents/Results/Lexogen_RNAseq/anno/Steinmetz_annotations_sacCer3_latin_body_endsm200_cleaned_plus.bed
wc -l ${bed/.bed/_body_endsm200_cleaned_minus.bed}
#    3366 /Users/schmidm/Documents/Results/Lexogen_RNAseq/anno/Steinmetz_annotations_sacCer3_latin_body_endsm200_cleaned_minus.bed
##lost some 40 body annotations completely


### total bp in the different parts
cat ${bed/.bed/_body_endsm200_plus.bed} | awk '{sum+=($3-$2)}END{print sum}'
#4745432
cat ${bed/.bed/_body_endsm200_cleaned_plus.bed} | awk '{sum+=($3-$2)}END{print sum}'
#4632362
cat ${bed/.bed/_body_endsm200_minus.bed} | awk '{sum+=($3-$2)}END{print sum}'
#4720823
cat ${bed/.bed/_body_endsm200_cleaned_minus.bed} | awk '{sum+=($3-$2)}END{print sum}'
#4594953
```
--> lost some 200000 bp of body annotations by this overlap cleaning.  


#### how many body regions are "split apart" ?
```{bash, eval = FALSE}
##how many body regions are "split apart" ?
cat ${bed/.bed/_body_endsm200_plus.bed} | cut -f 4 | uniq -c | grep -v "  1"
##no output, ie 0
cat ${bed/.bed/_body_endsm200_minus.bed} | cut -f 4 | uniq -c | grep -v "  1"
##no output ie 0
```
--> none!  
This may at least have happened some times, but explanation is the Steinmetz annotations are NEVER overlapping on the same strand... if an end region is overlapping due to 200bp extension, its only overlapping at one edge and never internally.


final sorting:  
```{bash, eval = FALSE}
body_plus_bed=${bed/.bed/_body_endsm200_cleaned_plus.bed}
body_minus_bed=${bed/.bed/_body_endsm200_cleaned_minus.bed}
ends_plus_bed=${bed/.bed/_ends_pm200_plus.bed}
ends_minus_bed=${bed/.bed/_ends_pm200_minus.bed}

#final sort for later use
sort -k1,1 -k2,2n ${body_plus_bed} -o ${body_plus_bed}
sort -k1,1 -k2,2n ${body_minus_bed} -o ${body_minus_bed}
sort -k1,1 -k2,2n ${ends_plus_bed} -o ${ends_plus_bed}
sort -k1,1 -k2,2n ${ends_minus_bed} -o ${ends_minus_bed}
```



#### number of genomic pA masked sites in each body and end region
Using this little script *bed_count_positions.sh*:  
```{bash, eval = FALSE}
#!/usr/bin/env bash

# counts the total number of bases that are covered in file $2 for each interval in file $1
# and uses bedtools intersect and a awk script to sum up adding the sum at the end
# note expects a 4-column bed file for file $1
# output goes to file $3
#ie line from file $2:
#   chrI 10 12
# and for an interval in $1
#   chrI 5 20 id . +
# will give chrI 5 20 id 2

bedtools intersect -wao -a $1 -b $2 | \
awk '{
  if($4==id){
    sum+=$8
  }else{
    print chr"\t"start"\t"end"\t"id"\t"sum;
    chr=$1; start=$2; end=$3; id=$4; sum=$8
  }
}END{
  print chr"\t"start"\t"end"\t"id"\t"sum
}' | sed 1d > $3
```

```{bash, eval = FALSE}
body_plus_bed=${bed/.bed/_body_endsm200_cleaned_plus.bed}
body_minus_bed=${bed/.bed/_body_endsm200_cleaned_minus.bed}
ends_plus_bed=${bed/.bed/_ends_pm200_plus.bed}
ends_minus_bed=${bed/.bed/_ends_pm200_minus.bed}

## count of genomic A-masked sites per annotation
pA_sites_plus="/Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_plus.bed"
pA_sites_minus="/Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_minus.bed"
grep ^chr ${pA_sites_plus} | sort -k1,1 -k2,2n > ${pA_sites_plus/.bed/_Sc.bed}
grep ^chr ${pA_sites_minus} | sort -k1,1 -k2,2n > ${pA_sites_minus/.bed/_Sc.bed}


## count of genomic A-masked sites for cleaned gene body regions
bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count_positions.sh \
${body_plus_bed} ${pA_sites_plus/.bed/_Sc.bed} ${body_plus_bed/_cleaned_plus.bed/_cleaned_plus_pAfilter.counts}

bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count_positions.sh \
${body_minus_bed} ${pA_sites_minus/.bed/_Sc.bed} ${body_minus_bed/_cleaned_minus.bed/_cleaned_minus_pAfilter.counts}

cat ${body_plus_bed/_cleaned_plus.bed/_cleaned_plus_pAfilter.counts} \
${body_minus_bed/_cleaned_minus.bed/_cleaned_minus_pAfilter.counts} > \
/Users/schmidm/Documents/Results/Lexogen_RNAseq/anno/clean_body_pAfilter.counts



## count of genomic A-masked sites for effective gene end regions
bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count_positions.sh \
${ends_plus_bed} ${pA_sites_plus/.bed/_Sc.bed} ${ends_plus_bed/_plus.bed/_plus_pAfilter.counts}

bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count_positions.sh \
${ends_minus_bed} ${pA_sites_minus/.bed/_Sc.bed} ${ends_minus_bed/_minus.bed/_minus_pAfilter.counts}

cat ${ends_plus_bed/_plus.bed/_plus_pAfilter.counts} \
${ends_minus_bed/_minus.bed/_minus_pAfilter.counts} > \
/Users/schmidm/Documents/Results/Lexogen_RNAseq/anno/clean_ends_pAfilter.counts
```



# counting

done using a simple bash script bed_count.sh that uses awk + bedtools
```{bash bed_count.sh, eval=FALSE}
#!/usr/bin/env bash

#expands a bedgraph file to single nucleotide intervals of file $2
# and uses bedtools map to count up features over file $1
# using sum over column $3
# output goes to file $4
#ie line from file $2:
#   chrI 10 12 20 2 +
#will be converted to:
#   chrI 10 11 20 2 +
#   chrI 11 12 20 2 +
# and counts from this file over interval in $1
#   chrI 5 20 . . +
# will give chrI 5 20 . . + 4

awk '{OFS="\t"}
{ start=$2;
  end=$3;
  for(i=start;i<end;i++){
    $2=i;
    $3=i+1;
    print $0
  }
}' $2 | \
sort -k1,1 -k2,2n | \
bedtools map -a $1 -b - -c $3 -o sum > $4
```

#### counting raw reads files
```{bash, eval = FALSE}
#### NON-NORMALIZED FILES
body_plus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_body_endsm200_cleaned_plus.bed"
body_minus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_body_endsm200_cleaned_minus.bed"
ends_plus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_ends_pm200_plus.bed"
ends_minus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_ends_pm200_minus.bed"


# A-filtered count files
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph

for f in *plus_KevinRoyAfiltered.bedgraph
  do
     echo "end reads counting"
     echo $f
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_plus_bed $f 4 plus_gene_endpm200.counts

     fminus="${f/_plus/_minus}"
     echo $fminus
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_minus_bed $fminus 4 minus_gene_endpm200.counts

     echo "merging ends"
     cat plus_gene_endpm200.counts minus_gene_endpm200.counts > \
     bedtools_counts/${f/_plus_KevinRoyAfiltered.bedgraph/_endpm200_count_sum_per_anno.txt}

     echo "body reads counting"
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_plus_bed $f 4 plus_gene_body.counts

     echo $fminus
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_minus_bed $fminus 4 minus_gene_body.counts

     echo "merging body"
     cat plus_gene_body.counts minus_gene_body.counts > \
     bedtools_counts/${f/_plus_KevinRoyAfiltered.bedgraph/_body_endsm200_cleaned_count_sum_per_anno.txt}

     echo "done"
  done

rm plus_gene_endpm200.counts
rm minus_gene_endpm200.counts
rm plus_gene_body.counts
rm minus_gene_body.counts
```


#### counting Sp-normalized files
```{bash, eval=FALSE}
## NORMALIZED to S.pombe track files
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph

#### annotation files
body_plus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_body_endsm200_cleaned_plus.bed"
body_minus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_body_endsm200_cleaned_minus.bed"
ends_plus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_ends_pm200_plus.bed"
ends_minus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_ends_pm200_minus.bed"

#### counts for input FILES
for f in *input*_plus_KevinRoyAfiltered.bedgraph
  do
     echo "end reads counting"
     echo $f
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_plus_bed $f 4 plus_gene_endpm200.counts

     fminus="${f/_plus/_minus}"
     echo $fminus
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_minus_bed $fminus 4 minus_gene_endpm200.counts

     echo "merging ends"
     cat plus_gene_endpm200.counts minus_gene_endpm200.counts > \
     bedtools_counts/${f/_plus_KevinRoyAfiltered.bedgraph/_endpm200_count_sum_per_anno.txt}

     echo "body reads counting"
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_plus_bed $f 4 plus_gene_body.counts

     echo $fminus
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_minus_bed $fminus 4 minus_gene_body.counts

     echo "merging body"
     cat plus_gene_body.counts minus_gene_body.counts > \
     bedtools_counts/${f/_plus_KevinRoyAfiltered.bedgraph/_body_endsm200_cleaned_count_sum_per_anno.txt}
     echo "done"
  done

rm plus_gene_endpm200.counts
rm minus_gene_endpm200.counts
rm plus_gene_body.counts
rm minus_gene_body.counts




#### counts for BGsub IP
for f in *ip*_plus_KevinRoyAfiltered_BGsub.bedgraph
  do
     fminus="${f/_plus/_minus}"
     echo "plus strand file "$f
     echo "minus strand file "$fminus

     echo "end reads counting"
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_plus_bed $f 4 plus_gene_endpm200.counts
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_minus_bed $fminus 4 minus_gene_endpm200.counts

     echo "merging ends"
     cat plus_gene_endpm200.counts minus_gene_endpm200.counts > \
     bedtools_counts/${f/_plus_KevinRoyAfiltered_BGsub.bedgraph/_BGsub_endpm200_count_sum_per_anno.txt}

     echo "body reads counting"
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_plus_bed $f 4 plus_gene_body.counts
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_minus_bed $fminus 4 minus_gene_body.counts

     echo "merging body"
     cat plus_gene_body.counts minus_gene_body.counts > \
     bedtools_counts/${f/_plus_KevinRoyAfiltered_BGsub.bedgraph/_BGsub_body_endsm200_cleaned_count_sum_per_anno.txt}

     echo "done"
  done

rm plus_gene_endpm200.counts
rm minus_gene_endpm200.counts
rm plus_gene_body.counts
rm minus_gene_body.counts

#### counts for NON-BGsub IP
for f in *ip*_plus_KevinRoyAfiltered.bedgraph
  do
     fminus="${f/_plus/_minus}"
     echo "plus strand file "$f
     echo "minus strand file "$fminus

     echo "end reads counting"
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_plus_bed $f 4 plus_gene_endpm200.counts
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_minus_bed $fminus 4 minus_gene_endpm200.counts

     echo "merging ends"
     cat plus_gene_endpm200.counts minus_gene_endpm200.counts > \
     bedtools_counts/${f/_plus_KevinRoyAfiltered.bedgraph/_endpm200_count_sum_per_anno.txt}

     echo "body reads counting"
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_plus_bed $f 4 plus_gene_body.counts
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_minus_bed $fminus 4 minus_gene_body.counts

     echo "merging body"
     cat plus_gene_body.counts minus_gene_body.counts > \
     bedtools_counts/${f/_plus_KevinRoyAfiltered.bedgraph/_body_endsm200_cleaned_count_sum_per_anno.txt}

     echo "done"
  done

rm plus_gene_endpm200.counts
rm minus_gene_endpm200.counts
rm plus_gene_body.counts
rm minus_gene_body.counts
```






#### counting pA-masked files
```{bash, eval=FALSE}
## NORMALIZED to S.pombe and pAplus masked track files
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph

#### annotation files
body_plus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_body_endsm200_cleaned_plus.bed"
body_minus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_body_endsm200_cleaned_minus.bed"
ends_plus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_ends_pm200_plus.bed"
ends_minus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_ends_pm200_minus.bed"

#### pAmasked NORMALIZED to S.pombe FILES
for f in only_pAminus_positions/*_plus_*_wo_noPaP_0rapa_sites.bedgraph
  do
     fminus="${f/_plus/_minus}"
     echo "plus strand file "$f
     echo "minus strand file "$fminus

     echo "end reads counting"
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_plus_bed $f 4 plus_gene_endpm200.counts
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $ends_minus_bed $fminus 4 minus_gene_endpm200.counts

     echo "merging ends"
     outfile=$(echo $f | sed s/_plus_KevinRoyAfiltered//g | sed s/_wo_noPaP_0rapa_sites.bedgraph//g | sed s/only_pAminus_positions\\///g | sed s/xPap/xPappAmasked/g)
     cat plus_gene_endpm200.counts minus_gene_endpm200.counts > \
     bedtools_counts/${outfile}_endpm200_count_sum_per_anno.txt

     echo "body reads counting"
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_plus_bed $f 4 plus_gene_body.counts
     bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $body_minus_bed $fminus 4 minus_gene_body.counts

     echo "merging body"
     cat plus_gene_body.counts minus_gene_body.counts > \
     bedtools_counts/${outfile}_body_endsm200_cleaned_count_sum_per_anno.txt

     echo "done"
  done

rm plus_gene_endpm200.counts
rm minus_gene_endpm200.counts
rm plus_gene_body.counts
rm minus_gene_body.counts
```




# Load counts into R

```{r}
read_bedgraph_sums <- function(path, file_list, sfx) {
  lapply(file_list, function(fname) {read.table(paste0(path,fname,sep=''), header=F, col.names=c('chr', 'start', 'end', 'id', 'sum'), stringsAsFactors = FALSE) %>% 
      tbl_df %>%
  mutate(condition = sub(sfx, '', fname),
         condition = sub('norm_', '', condition),
         sum = ifelse(sum == '.', 0, as.numeric(sum)))}) %>%
  bind_rows()
}
```



## raw counts

#### load and combine end counts
```{r}
path <- '/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph/bedtools_counts/'
sfx <- '_endpm200_count_sum_per_anno.txt$'

file_list <- dir(path)[grep(sfx, dir(path))]

file_list
```


```{r, warning=FALSE}
(end_counts_df <- read_bedgraph_sums(path, file_list, sfx))
```


#### load and combine body counts
```{r}
sfx <- '_body_endsm200_cleaned_count_sum_per_anno.txt$'

file_list <- dir(path)[grep(sfx, dir(path))]

file_list
```

```{r, warning=FALSE}
(body_counts_df <- read_bedgraph_sums(path, file_list, sfx))
```


#### combine body and end 

```{r}
(raw_counts_df <- bind_rows(mutate(body_counts_df, part = 'body'), 
                     mutate(end_counts_df, part = 'end')) %>%
   mutate(pA = case_when(grepl('^noPap', .$condition) ~ 'pA+', 
                         grepl('^xPap_', .$condition) ~ 'pA+ + pA-',
                         TRUE ~ '??')))
```


```{r}
save(raw_counts_df, file='data/raw_signal_body_end_all.RData')
```


## Sp-normalized counts

#### load and combine end counts
```{r}
path <- '/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bedgraph/bedtools_counts/'
sfx <- '_endpm200_count_sum_per_anno.txt$'

file_list <- dir(path)[grep(sfx, dir(path))]

file_list
```


```{r, warning=FALSE}
(end_counts_df <- read_bedgraph_sums(path, file_list, sfx))
```


#### load and combine body counts
```{r}
sfx <- '_body_endsm200_cleaned_count_sum_per_anno.txt$'

file_list <- dir(path)[grep(sfx, dir(path))]

file_list
```

```{r, warning=FALSE}
(body_counts_df <- read_bedgraph_sums(path, file_list, sfx))
```


#### combine body and end 

```{r}
(both_df <- bind_rows(mutate(body_counts_df, part = 'body'), 
                     mutate(end_counts_df, part = 'end')) %>%
   mutate(pA = case_when(grepl('^noPap', .$condition) ~ 'pA+', 
                         grepl('^xPappAminus', .$condition) ~ 'pA-',
                         grepl('^xPappAmasked', .$condition) ~ 'pA-X',
                         grepl('^xPap_', .$condition) ~ 'pA+ + pA-',
                         TRUE ~ 'NA')))
```


```{r}
save(both_df, file='data/norm_signal_body_end_raw_all.RData')
```


## Sanity check S.pombe normalization

simply checks that $SpombeNormCounts = RawCounts / SpombeSizeFactor$  

#### Load the S.pombe size factors
```{r}
(sf <- read.table('data/Sp_genes_sf.txt', col.names=c('condition', 'sf')) %>%
  tbl_df)
```


#### check it
```{r}
(raw_vs_norm <- left_join(raw_counts_df %>%
  dplyr::rename(raw_reads = sum),
  sf) %>%
  left_join(., both_df %>%
  dplyr::rename(norm_reads = sum)))
```


```{r}
filter(raw_vs_norm, norm_reads != raw_reads/sf)
```

apparently its not exactly the same values, probably rounding makes the difference ....

```{r}
mutate(raw_vs_norm, re_norm_reads = raw_reads/sf) %>%
  group_by(condition, part) %>%
  do(tidy(cor(.$norm_reads, .$re_norm_reads))) %>%
  kable
```

seems to be perfectly correlated.  


```{r}
mutate(raw_vs_norm, rel_diff = (norm_reads - (raw_reads/sf))/norm_reads) %>%
  group_by(condition, part) %>%
  do(tidy(summary(.$rel_diff))) %>%
  arrange(maximum) %>%
  kable
```

-> can see that maximum relative difference at a single feature between norm and recomputed normalized is around 4E-6, ie 4 in a million, aka .0004%.
-> its roughly stable between all libraries so appears to be a rounding thingy



## Sanity check BGsub and pA minus method

simply checks that 
1. norm. counts for $BGsub = normIP  -  normnegativeIP$  
2. norm. counts for $pAminus = pAplusandminus  -  pAplus$  

#### BGsub
```{r}
IP_raw <- filter(both_df, grepl('ip', condition) & !(grepl('BGsub', condition))) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop')

IP_neg <- filter(both_df, grepl('ip_neg0', condition)) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop') %>%
  dplyr::select(-rapa, -rep) %>%
  dplyr::rename(ip_neg_sum = sum)

IP_BGsub <- filter(both_df, grepl('ip', condition) & (grepl('BGsub', condition))) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop') %>%
  dplyr::rename(BGsub_sum = sum)
```


```{r}
IP_BGsub_redone <- left_join(IP_raw, IP_neg) %>%
  mutate(BGsub_sum_redone = sum - ip_neg_sum) %>%
  dplyr::select(-sum, -ip_neg_sum)
```

```{r}
(BGsub_comparison <- left_join(IP_BGsub_redone, IP_BGsub) %>%
   mutate(relative_error = (BGsub_sum_redone-BGsub_sum)/BGsub_sum))
```

```{r}
BGsub_comparison %>%
  filter(is.finite(BGsub_sum_redone), is.finite(BGsub_sum)) %>%
  group_by(pA, part, Pap, strain, rapa, rep) %>%
  #summarize(cnt=n()) %>% kable
  do(tidy(cor.test(.$BGsub_sum, .$BGsub_sum_redone))) %>%
  kable
```

```{r}
BGsub_comparison %>%
  filter(is.finite(BGsub_sum_redone), is.finite(BGsub_sum)) %>%
  group_by(pA, part, Pap, strain, rapa, rep) %>%
  do(tidy(summary(.$relative_error))) %>% kable
```

--> perfectly correlated and same values
--> OK


## work only with BGsub IPs

```{r}
(both_df %<>% 
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep', 'BGsub'), sep='_') %>%
  separate(id, c('id', 'type', 'name', 'common_name'), sep=':') %>%
  filter(fraction == 'input' | (fraction == 'ip' & BGsub == 'BGsub'))  %>%
  dplyr::select(-BGsub))
```


## Normalize reads to region length and correct for genomic A masked positions

#### Original plain annotations
```{r}
(anno <- read.table("/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin.bed", col.names = c('chr', 'start', 'end', 'anno', 'pad', 'strand')) %>%
  tbl_df %>% 
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':') %>%
  mutate(type = ifelse(grepl('^snR', name), 'snRNA', as.character(type)),
         type = factor(type, levels=c('ORF-T', 'SUTs', 'CUTs', 'snRNA', 'other'))) %>%
   arrange(id))
```



### cleaned region lengths and A-masked positions

Note: for body regions stripped for overlapping ends, end regions are all 400bp. This info is contained in the A-mask count files.

A-masked positions: nothing will ever be counted from those positions, so in practice these positions do not count to the length of each region -> subtract them:  
```{r}
(effective_body_len <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq/anno/clean_body_pAfilter.counts', col.names=c('chr', 'start', 'end', 'anno', 'body_pA_masked')) %>%
   tbl_df %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':') %>%
   mutate(pAfiltered_cleaned_body_len = end - start - body_pA_masked))
```

```{r}
(effective_end_len <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq/anno/clean_ends_pAfilter.counts', col.names=c('chr', 'start', 'end', 'anno', 'end_pA_masked')) %>%
   tbl_df %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':') %>%
   mutate(pAfiltered_end_len = end - start - end_pA_masked))
```


#### combine all of the above

```{r}
(effective_lens <- full_join(
  dplyr::select(effective_body_len, id, pAfiltered_cleaned_body_len),
  dplyr::select(effective_end_len, id, pAfiltered_end_len)) %>%
   arrange(id)
  )
```


### pAplus positions mask

```{r}
(body_pAplus_masked <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph/noPap_0rapa_sites_per_cleaned_body.txt', col.names=c('chr', 'start', 'end', 'anno', 'body_pAplus_cnt')) %>%
   tbl_df %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':') %>%
   mutate(body_pAplus_cnt = ifelse(body_pAplus_cnt == '.', as.numeric(0), as.numeric(body_pAplus_cnt))))
```



```{r}
(end_pAplus_masked <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph/noPap_0rapa_sites_per_end.txt', col.names=c('chr', 'start', 'end', 'anno', 'end_pAplus_cnt')) %>%
   tbl_df %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':') %>%
   mutate(end_pAplus_cnt = ifelse(end_pAplus_cnt == '.', as.numeric(0), as.numeric(end_pAplus_cnt))))
```


#### combined pA plus masks 
```{r}
(pAplus_masked <- full_join(
  dplyr::select(body_pAplus_masked, id, body_pAplus_cnt),
  dplyr::select(end_pAplus_masked, id, end_pAplus_cnt)) %>%
   arrange(id)
  )
```


Final annotations, containing that info:  
```{r}
(anno <- left_join(anno, effective_lens) %>%
   left_join(pAplus_masked))
```


```{r}
(both_df %<>% 
  dplyr::select(id, sum, part, pA, Pap, strain, fraction, rapa, rep) %>%
  left_join(., anno) %>%
  mutate(scaled_reads = case_when(.$part == 'body' & pA != 'pA-X' ~ sum/pAfiltered_cleaned_body_len,
                               .$part == 'end'  & pA != 'pA-X' ~ sum/pAfiltered_end_len,
                               .$part == 'body' & pA == 'pA-X' ~ sum/(pAfiltered_cleaned_body_len-body_pAplus_cnt),
                               .$part == 'end'  & pA == 'pA-X' ~ sum/(pAfiltered_end_len-end_pAplus_cnt))) %>%
  dplyr::select(id, name, common_name, type, part, pA, Pap, strain, fraction, rapa, rep, sum, scaled_reads))
```


#### sanity check non-exisiting gene bodies
…. note some regions pAfiltered_cleaned_body_len are NA … ie there is no body, either completely overlapping with ends or otherwise the annotation is too short (or only consists of masked genomic A, although this probably never happens) !! 
-> there should be no reads for those
-> sanity check all with pAfiltered_cleaned_body_len == NA should never have counts

```{r}
body_len_NA_ids <- filter(anno, is.na(pAfiltered_cleaned_body_len)) %>% distinct(id, common_name, type)

table(body_len_NA_ids$type)
```

mostly CUTs, many SUTs, some snRNAs very few ORFs.  


```{r}
filter(both_df, part == 'body', id %in% body_len_NA_ids$id)
```

```{r}
filter(both_df, part == 'end', id %in% body_len_NA_ids$id)
```


--> seems correct, we never count body for those where the region does not exist. But ends are apprently counted ...




#### look at some single genes

```{r}
filter(both_df,
       common_name == 'PMA1',
       rapa == 0, rep == 1) %>%
  kable
```

```{r}
filter(both_df,
       common_name == 'PGK1',
       rapa == 0, rep == 1) %>%
  kable
```


```{r}
filter(both_df,
       common_name == 'SNR11',
       rapa == 0, rep == 1) %>%
  kable
```


## save the counts
```{r}
save(both_df, file='data/norm_signal_body_end_pA+_pA-_BGsub.RData')
```



# End vs body pA+ vs pA-

Global distribution of pA+ and pA- signal in gene body an gene end regions.

## sum signal per annotation class and sample
```{r}
(both_totals <- both_df %>%
  #mutate(sum = ifelse(sum < 0, 0, sum)) %>%
  group_by(type, Pap, strain,fraction, rapa, part, pA, rep) %>% 
  summarize(total_Spnorm_signal = sum(sum)))
```

```{r barplot total all samples }
both_totals %>%
  group_by(part, pA, fraction, strain, rapa, rep) %>%
  summarize(all_types_total_Spnorm_signal = sum(total_Spnorm_signal)) %>%
ggplot(., aes(x=part, y=all_types_total_Spnorm_signal, fill=part)) + 
  geom_bar(stat='identity') +
  facet_grid(pA+fraction~strain+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```

```{r barplot total all samples rep1 rapa0}
both_totals %>%
  filter(rapa == 0, rep == 1) %>%
  group_by(part, pA, fraction, strain, rapa, rep) %>%
  summarize(all_types_total_Spnorm_signal = sum(total_Spnorm_signal)) %>%
ggplot(., aes(x=part, y=all_types_total_Spnorm_signal, fill=part)) + 
  geom_bar(stat='identity') +
  facet_grid(pA+fraction~strain+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```

```{r barplot totals per type all samples rep1 rapa0}
both_totals %>%
  filter(rapa == 0, rep == 1) %>%
ggplot(., aes(x=part, y=total_Spnorm_signal, fill=part)) + 
  geom_bar(stat='identity') +
  facet_grid(pA+fraction~type+strain+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```

```{r barplot totals ORF-T all samples rep1 rapa0}
both_totals %>%
  filter(rapa == 0, rep == 1, type == 'ORF-T') %>%
ggplot(., aes(x=part, y=total_Spnorm_signal, fill=part)) + 
  geom_bar(stat='identity') +
  facet_grid(pA+fraction~type+strain+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```


#### total signal per type

```{r barplot signal per type and part}
both_totals %>%
  filter(rapa == '0', rep == '1') %>%
  ggplot(., aes(x=type, y=total_Spnorm_signal, fill=type)) + 
  geom_bar(stat='identity') +
  facet_grid(pA+fraction~part+strain+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() 
```

```{r barplot signal per type wo part}
both_totals %>%
  filter(rapa == '0', rep == '1') %>%
  ggplot(., aes(x=type, y=total_Spnorm_signal, fill=type)) + 
  geom_bar(stat='identity') +
  facet_grid(pA+fraction~strain+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() 
```


#### pAplus vs pAminus body vs end pie

```{r}
pAplusandminus_totals <- filter(both_totals, pA == 'pA+ + pA-') %>%
  ungroup %>%
  dplyr::select(type, strain, fraction, rapa, rep, part, total_Spnorm_signal) %>%
    mutate(total_Spnorm_signal = ifelse(total_Spnorm_signal < 0, 0, total_Spnorm_signal)) %>%
  dplyr::rename(pAplusandminus_total = total_Spnorm_signal)

pAminus_totals <- filter(both_totals, pA == 'pA-') %>%
  ungroup %>%
  dplyr::select(type, strain, fraction, rapa, rep, part, total_Spnorm_signal) %>%
    mutate(total_Spnorm_signal = ifelse(total_Spnorm_signal < 0, 0, total_Spnorm_signal)) %>%
  dplyr::rename(pAminus_total = total_Spnorm_signal)
```

```{r}
pie_percents_pAplusvsminus <- left_join(pAplusandminus_totals, pAminus_totals) %>%
  mutate(pAplus_total = ifelse(pAplusandminus_total > pAminus_total,
                               pAplusandminus_total - pAminus_total,
                               0)) %>%
  dplyr::select(-pAplusandminus_total) %>%
  gather(pA, total_Spnorm_signal, -type, -strain, -fraction, -rapa, -rep, -part) %>%
  unite(part, c(part, pA), sep='_') %>%
  group_by(type, strain, fraction, rapa, rep) %>%
  mutate(total_Spnorm_signal_percent = 100 * total_Spnorm_signal / sum(total_Spnorm_signal))
```

```{r}
pie_percents_pAplusvsminus %>%
  ungroup %>%
  arrange(type, strain, fraction, rapa, rep, part) %>%
  kable
```


```{r piechart body vs end pAplus vs pAminus}
pie_percents_pAplusvsminus %>%
  filter(rapa == '0') %>% 
ggplot(., aes(x=factor(1), y=total_Spnorm_signal_percent, fill=part)) + 
  geom_bar(stat='identity') +
  coord_polar(theta='y') +
  facet_grid(type+strain~fraction+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```



#### E-PAP libraries piechart
```{r}
pie_percents_pAplusandminus <- filter(both_totals, pA == 'pA+ + pA-') %>%
  group_by(pA, type, strain, fraction, rapa, rep) %>%
  mutate(total_Spnorm_signal_percent = 100 * total_Spnorm_signal / sum(total_Spnorm_signal))

kable(pie_percents_pAplusandminus)
```


```{r EPAP libraries piechart}
pie_percents_pAplusandminus %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=factor(1), y=total_Spnorm_signal_percent, fill=part)) + 
  geom_bar(stat='identity') +
  coord_polar(theta='y') +
  facet_grid(type+strain~fraction+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```


#### noPAP libraries piechart

```{r}
pie_percents_pAplus <- filter(both_totals, pA == 'pA+') %>%
  group_by(pA, type, strain, fraction, rapa, rep) %>%
  mutate(total_Spnorm_signal_percent = 100 * total_Spnorm_signal / sum(total_Spnorm_signal))

kable(pie_percents_pAplus)
```


```{r noPAP libraries piechart}
pie_percents_pAplus %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=factor(1), y=total_Spnorm_signal_percent, fill=part)) + 
  geom_bar(stat='identity') +
  coord_polar(theta='y') +
  facet_grid(type+strain~fraction+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```


#### pAminus piechart

```{r}
pie_percents_pAminus <- filter(both_totals, pA == 'pA-') %>%
  mutate(total_Spnorm_signal = ifelse(total_Spnorm_signal < 0, 0, total_Spnorm_signal)) %>%
  group_by(pA, type, strain, fraction, rapa, rep) %>%
  mutate(total_Spnorm_signal_percent = 100 * total_Spnorm_signal / sum(total_Spnorm_signal))

kable(pie_percents_pAminus)
```

```{r pAminus libraries piechart}
pie_percents_pAminus %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=factor(1), y=total_Spnorm_signal_percent, fill=part)) + 
  geom_bar(stat='identity') +
  coord_polar(theta='y') +
  facet_grid(type+strain~fraction+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```


#### pAminus pAplus masked piechart


```{r}
pie_percents_pAminus_masked <- filter(both_totals, pA == 'pA-X') %>%
  mutate(total_Spnorm_signal = ifelse(total_Spnorm_signal < 0, 0, total_Spnorm_signal)) %>%
  group_by(pA, type, strain, fraction, rapa, rep) %>%
  mutate(total_Spnorm_signal_percent = 100 * total_Spnorm_signal / sum(total_Spnorm_signal))

kable(pie_percents_pAminus_masked)
```

```{r pAminus pAplus masked libraries piechart}
pie_percents_pAminus_masked %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=factor(1), y=total_Spnorm_signal_percent, fill=part)) + 
  geom_bar(stat='identity') +
  coord_polar(theta='y') +
  facet_grid(type+strain~fraction+rapa+rep, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw() +
  theme(axis.text.x = element_blank())
```




```{r}
sessionInfo()
```



