---
title: 'Correlation with Decay rates and txn measures'
author: "Manfred Schmid"
output: html_document
---

`r format(Sys.time(), "%d %B, %Y")`

```{r setup, echo=TRUE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, 
                      fig.path=paste0('Figures_correlation_with_halflife_txn/'), 
                      dev='pdf', 
                      echo=TRUE, warning=FALSE, message=FALSE, 
                      error=TRUE)
```



```{r load packages, echo=T, warning=F, message=F}
suppressWarnings(library(tidyverse))
suppressWarnings(library(magrittr))
suppressWarnings(library(knitr))
suppressWarnings(library(broom))
suppressWarnings(library(LSD))
```


## Loading half-life datasets

#### DTA 

from Miller et al., Molecular Systems Biology 7:458, 2010
```{r}
(miller <- read_tsv('/Users/schmidm/Documents/genomewide_datasets/halflife_and_transcription_rates/Cramer_DTA/Cramer_half-life.txt', col_names=TRUE) %>%
  mutate(Miller_DR = log(2)/halflife0) %>%
  dplyr::select(name, Miller_DR))
```


#### cDTA 

from Sun et al., Mol. Cell 52, 52–62, 2013
```{r}
sun <- read_tsv('/Users/schmidm/Documents/genomewide_datasets/halflife_and_transcription_rates/Cramer_cDTA_Sun2013/Cramer_decay_rates.txt', col_names=TRUE)

sun <- sun[,c(1,3)]
colnames(sun) <- c('name', 'Sun_DR')
(sun %<>% mutate(Sun_DR = as.numeric(sub(',', '.', Sun_DR))))
```


```{r}
DR_literature <- left_join(miller, sun)
```

```{r}
heatscatter(DR_literature$Miller_DR, DR_literature$Sun_DR, xlim=c(0,.6), ylim=c(0,.6), cor=TRUE)
```

```{r}
cor(DR_literature$Miller_DR, DR_literature$Sun_DR, use='pairwise.complete')
```

```{r}
cor(DR_literature$Miller_DR, DR_literature$Sun_DR, use='pairwise.complete', method='spearman')
```


#### Presnyak

from Presnyak et al., 2015, Cell 160, 1111–1124
```{r}
(presnyak <- read_tsv('/Users/schmidm/Documents/genomewide_datasets/halflife_and_transcription_rates/Coller_half_life/Coller_halflife.txt', col_names=TRUE, locale=locale(decimal_mark = ",")) %>%
   dplyr::rename(name = Gene_ID) %>%
   mutate(Presnyak_pA_DR = log(2)/pA_Half_life,
          Presnyak_total_DR =  log(2)/Total_Half_life) %>%
   dplyr::select(name, Presnyak_pA_DR, Presnyak_total_DR)) 
```


```{r}
DR_literature %<>% left_join(., presnyak)
```

```{r}
heatscatter(DR_literature$Presnyak_pA_DR, DR_literature$Sun_DR, xlim=c(0,1), ylim=c(0,1), cor=TRUE)
```

```{r}
heatscatter(DR_literature$Presnyak_total_DR, DR_literature$Sun_DR, xlim=c(0,1), ylim=c(0,1), cor=TRUE)
```

```{r}
heatscatter(DR_literature$Presnyak_pA_DR, DR_literature$Miller_DR, xlim=c(0,1), ylim=c(0,1), cor=TRUE)
```

```{r}
heatscatter(DR_literature$Presnyak_total_DR, DR_literature$Miller_DR, xlim=c(0,1), ylim=c(0,1), cor=TRUE)
```


## Loading Mex and Nab depletion log2FC

#### load the DESeq2 results
```{r}
load(file='data/DESeq2_10min_results_df.RData', verbose=TRUE)

dds_results_df
```

```{r}
load(file='data/DESeq2_10min_ip_rel_input_results_df.RData', verbose=TRUE)

dds_ip_rel_in_df
```



## Correlation with Decay Rates

```{r}
(LFC_spread <- bind_rows(dds_results_df, dds_ip_rel_in_df) %>%
   filter(type == 'ORF-T') %>%
  dplyr::select(id, name, type, log2FoldChange, comparison) %>%
  spread(comparison, log2FoldChange) %>%
  left_join(., DR_literature)) 
```


```{r}
cor_dist <- function(x, ...) {
  as.dist((1 - cor(x, ...))/2)
}
```

#### spearman correlation

```{r}
(cor_df <- LFC_spread %>%
  dplyr::select(-name, -id, -type, -Presnyak_pA_DR) %>%
  as.matrix %>%
  na.omit %>%
  cor(., method='spearman') %>%
  data.frame %>%
  rownames_to_column(var='study1') %>%
  gather(study2, cor, -study1) %>%
  tbl_df)
```


```{r spearman correlation hclust}
spearman_dists <- LFC_spread %>%
  dplyr::select(-name, -id, -type, -Presnyak_pA_DR) %>%
  as.matrix %>%
  na.omit %>%
  cor_dist(., method='spearman')

hc <- hclust(spearman_dists)

plot(hc)

hc_order <- hc$labels[hc$order]
```


```{r spearman correlation matrix}
cor_df %>%
  mutate(study1 = factor(study1, levels=hc_order),
         study2 = factor(study2, levels=hc_order)) %>%
ggplot(., 
       aes(x=study1, y=study2, fill=cor)) +
  geom_tile() +
  theme(axis.text.x=element_text(angle=45, hjust=1)) +
  scale_fill_gradient2(low='red', mid='white', high='blue', limits=c(-1,1)) +
  geom_text(aes(label=round(cor,2)))
```


### only genes sig over background

```{r}
load('data/DESeq2_10min_IP_rel_neg_results_df.RData', verbose=TRUE)

dds_rel_neg_results_df
```


```{r}
(ip_sig_over_bkgd <- filter(dds_rel_neg_results_df,
       log2FoldChange > 0, padj < .1,
       type == 'ORF-T'
       ) %>%
  dplyr::select(name, comparison))
```


#### IPs only gene sig over background at 0

```{r}
ids_mex_sig0 <- filter(ip_sig_over_bkgd, comparison == 'Mex_ip_0relneg') %$% name

ids_nab_sig0 <- filter(ip_sig_over_bkgd, comparison == 'Nab_ip_0relneg') %$% name
  
ids_both_sig0 <- intersect(ids_mex_sig0, ids_nab_sig0)

length(ids_both_sig0)
```
```{r}
(cor_df <- LFC_spread %>%
   filter(name %in% ids_both_sig0) %>%
  dplyr::select(-name, -id, -type, -Presnyak_pA_DR) %>%
  as.matrix %>%
  na.omit %>%
  cor(., method='spearman') %>%
  data.frame %>%
  rownames_to_column(var='study1') %>%
  gather(study2, cor, -study1) %>%
  tbl_df)
```

```{r spearman correlation hclust sig over BG}
spearman_dists <- LFC_spread %>%
  filter(name %in% ids_both_sig0) %>%
  dplyr::select(-name, -id, -type, -Presnyak_pA_DR) %>%
  as.matrix %>%
  na.omit %>%
  cor_dist(., method='spearman')

hc <- hclust(spearman_dists)

plot(hc)

hc_order <- hc$labels[hc$order]
```


```{r spearman correlation matrix only genes sig over bkgd 0}
cor_df %>%
  mutate(study1 = factor(study1, levels=hc_order),
         study2 = factor(study2, levels=hc_order)) %>%
ggplot(., 
       aes(x=study1, y=study2, fill=cor)) +
  geom_tile() +
  theme(axis.text.x=element_text(angle=45, hjust=1)) +
  scale_fill_gradient2(low='red', mid='white', high='blue', limits=c(-1,1)) +
  geom_text(aes(label=round(cor,2)))
```



#### IPs only gene sig over background at 0 and 15

```{r}
ids_sigall <- ip_sig_over_bkgd %>%
  group_by(name) %>%
  summarize(cnt = n()) %>%
  filter(cnt >= 4)

nrow(ids_sigall)
```
```{r}
(cor_df <- LFC_spread %>%
   filter(name %in% ids_sigall$name) %>%
  dplyr::select(-name, -id, -type, -Presnyak_pA_DR) %>%
  as.matrix %>%
  na.omit %>%
  cor(., method='spearman') %>%
  data.frame %>%
  rownames_to_column(var='study1') %>%
  gather(study2, cor, -study1) %>%
  tbl_df)
```


```{r spearman correlation sig all hclust}
spearman_dists <- LFC_spread %>%
  filter(name %in% ids_sigall$name) %>%
  dplyr::select(-name, -id, -type, -Presnyak_pA_DR) %>%
  as.matrix %>%
  na.omit %>%
  cor_dist(., method='spearman')

hc <- hclust(spearman_dists)

plot(hc)

hc_order <- hc$labels[hc$order]
```


```{r spearman correlation matrix only genes sig over bkgd all}
cor_df %>%
  mutate(study1 = factor(study1, levels=hc_order),
         study2 = factor(study2, levels=hc_order)) %>%
ggplot(., 
       aes(x=study1, y=study2, fill=cor)) +
  geom_tile() +
  theme(axis.text.x=element_text(angle=45, hjust=1)) +
  scale_fill_gradient2(low='red', mid='white', high='blue', limits=c(-1,1)) +
  geom_text(aes(label=round(cor,2)))
```


## Correlation with Transcription

load the transcription estimates from Markdown Published_txn_estimates_gene_body_counting.Rmd

```{r}
load(file='../../Lexogen_RNAseq_2/RProject/data/published_gene_body_txn_estimates.RData', verbose=TRUE)

txn_estimates
```

```{r}
(txn_estimates %<>%
  separate(id, c('id', 'type', 'name', 'common_name'), sep=':') %>%
  filter(type == 'ORF-T'))
```

```{r}
LFC_spread %<>%
   dplyr::select(-contains('DR'), -contains('ip_rel_in')) %>%
   left_join(txn_estimates, .)
```

#### spearman correlation

```{r}
(cor_df <- LFC_spread %>%
  dplyr::select(-name, -id, -type, -common_name) %>%
  as.matrix %>%
  na.omit %>%
  cor(., method='spearman') %>%
  data.frame %>%
  rownames_to_column(var='study1') %>%
  gather(study2, cor, -study1) %>%
  tbl_df)
```


```{r txn spearman hclust}
spearman_dists <- LFC_spread %>%
  dplyr::select(-name, -id, -type, -common_name) %>%
  as.matrix %>%
  na.omit %>%
  cor_dist(., method='spearman')

hc <- hclust(spearman_dists)

plot(hc)

hc_order <- hc$labels[hc$order]
```


```{r txn spearman correlation matrix}
cor_df %>%
  mutate(study1 = factor(study1, levels=hc_order),
         study2 = factor(study2, levels=hc_order)) %>%
ggplot(., 
       aes(x=study1, y=study2, fill=cor)) +
  geom_tile() +
  theme(axis.text.x=element_text(angle=45, hjust=1)) +
  scale_fill_gradient2(low='red', mid='white', high='blue', limits=c(-1,1)) +
  geom_text(aes(label=round(cor,2)))
```


### only genes sig over background

#### IPs only gene sig over background at 0

```{r}
(cor_df <- LFC_spread %>%
   filter(name %in% ids_both_sig0) %>%
  dplyr::select(-name, -id, -type, -common_name) %>%
  as.matrix %>%
  na.omit %>%
  cor(., method='spearman') %>%
  data.frame %>%
  rownames_to_column(var='study1') %>%
  gather(study2, cor, -study1) %>%
  tbl_df)
```


```{r txn sig 0 hclust}
spearman_dists <- LFC_spread %>%
  filter(name %in% ids_both_sig0) %>%
  dplyr::select(-name, -id, -type, -common_name) %>%
  as.matrix %>%
  na.omit %>%
  cor_dist(., method='spearman')

hc <- hclust(spearman_dists)

plot(hc)

hc_order <- hc$labels[hc$order]
```


```{r txn spearman correlation matrix only genes sig over bkgd 0}
cor_df %>%
  mutate(study1 = factor(study1, levels=hc_order),
         study2 = factor(study2, levels=hc_order)) %>%
ggplot(., 
       aes(x=study1, y=study2, fill=cor)) +
  geom_tile() +
  theme(axis.text.x=element_text(angle=45, hjust=1)) +
  scale_fill_gradient2(low='red', mid='white', high='blue', limits=c(-1,1)) +
  geom_text(aes(label=round(cor,2)))
```



#### IPs only gene sig over background at 0 and 15

```{r}
(cor_df <- LFC_spread %>%
   filter(name %in% ids_sigall$name) %>%
  dplyr::select(-name, -id, -type, -common_name) %>%
  as.matrix %>%
  na.omit %>%
  cor(., method='spearman') %>%
  data.frame %>%
  rownames_to_column(var='study1') %>%
  gather(study2, cor, -study1) %>%
  tbl_df)
```


```{r txn sig all hclust}
spearman_dists <- LFC_spread %>%
  filter(name %in% ids_sigall$name) %>%
  dplyr::select(-name, -id, -type, -common_name) %>%
  as.matrix %>%
  na.omit %>%
  cor_dist(., method='spearman')

hc <- hclust(spearman_dists)

plot(hc)

hc_order <- hc$labels[hc$order]
```


```{r txn spearman correlation matrix only genes sig over bkgd all}
cor_df %>%
  mutate(study1 = factor(study1, levels=hc_order),
         study2 = factor(study2, levels=hc_order)) %>%
ggplot(., 
       aes(x=study1, y=study2, fill=cor)) +
  geom_tile() +
  theme(axis.text.x=element_text(angle=45, hjust=1)) +
  scale_fill_gradient2(low='red', mid='white', high='blue', limits=c(-1,1)) +
  geom_text(aes(label=round(cor,2)))
```


```{r}
sessionInfo()
```
