---
title: 'Nucleosomes'
author: "Manfred Schmid"
output: html_document
---

`r format(Sys.time(), "%d %B, %Y")`


```{r, echo=TRUE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, 
                      fig.path=paste0('Figures_nucleosomes_v2/'), 
                      dev='pdf', echo=TRUE, warning=FALSE, message=FALSE, 
                      error=TRUE)
```


```{r load packages, echo=T, warning=F, message=F}
suppressWarnings(library(tidyverse))
suppressWarnings(library(magrittr))
suppressWarnings(library(knitr))
suppressWarnings(library(zoo))
```


## Loading bwtool matrices

For this part I used bwtool for metagene assemblies. Its simply much faster than deeptools, caveat is that it has fewer features, but enough for the part here.

a little fun to load metagene matrices generated by bwtool

### compound load functions 

#### gather a bwtool matrix result to tidy tbl

```{r}
gather_bwtool_matrix <- function(d, upstream, downstream) {
  colnames(d)[1:6] <- c('chr', 'start', 'end', 'id', 'pad', 'strand')
  colnames(d)[7:ncol(d)] <- as.character(-upstream:(downstream-1))
  d <- d[,-c(1:3,5:6)]
  
  d %>% 
    separate(id, c('nuc', 'id'), sep=':') %>%
    tbl_df %>%
    gather(pos, value, -id, -nuc) %>%
    mutate(pos = as.integer(pos),
           value = as.numeric(value),
           nuc = ifelse(grepl('\\*', nuc), 
                        'term', 
                        nuc)) %>% 
    filter(!is.na(value)) 
}
```

#### read file to tbl_df

```{r}
load_bwtool_nuc_matrix <- function (fname, upstream, downstream, both_strands = TRUE, cutvalues_below = 0) {
  print('loading plus strand file')
  d <- read_tsv(fname, col_names = FALSE) %>%
    gather_bwtool_matrix(upstream, downstream) %>%
    mutate(strand = '+')
  
  minus_fname <- gsub('plus', 'minus', fname)
  print(paste('try loading minus strand file ', minus_fname))
  if( both_strands & file.exists(minus_fname) ) {
     d_minus <- read_tsv(minus_fname, col_names = FALSE) %>%
      gather_bwtool_matrix(upstream, downstream) %>%
      mutate(pos = -pos, strand = '-')
     
     d %<>% bind_rows(., d_minus)
  } else {
    print('--> no minus strand file found working with plus only')
  }

  print('merging and computing metagene values')
  if ( !is.na(cutvalues_below) ) {
    d %<>% filter(value > cutvalues_below)
  }
  
  d_agg_in <- d %>% 
    group_by(nuc, pos) %>%
    summarize(events=n(),
              sum_value = sum(value),
              sum_log2_value = sum(log2(value)))
}
```



# Nucleosome Positions


We are using the data from Pugh lab publication Jiang and Pugh, 2009. This is a meta-study assembling different datasets into a common nucleoseome-positioning set.  
Processing from Additional Data 1 Excel sheet, exported to a txt file was as follows:  
```{bash, eval = FALSE}
#extract from supplemental Additonal Data 1 Excel file from Jiang and Pugh 2009 publication
#this one has nucleosomes info from several publication and makes a common list
#saved fthe first sheed "Measured nucleosomes" as Pugh_measured_nucleosomes.txt
#nucleosomes marked with * in position are  terminal nucleosome

cat Pugh_measured_nucleosomes.txt | tr "\r" "\n" | cut -f 1-20 | head
#Name    chrom   start   end     position        peakheight      nc      occ     fuzziness       Set1    rc1     occ1    Set2    rc2     occ2    Set3    rc3     occ3    Set4    rc4
#N1:192  chr01   119     265             1,11    3       80,5    41      151     10      89                              192     109     72
#N1:382  chr01   309     455             3,61    5       71      21,916  376     260     100     427     7       100     372     50      13
#N1:630  chr01   557     703             4,31    5       62      10,756  622     359     100     647     2       79      636     44      7
#N1:872  chr01   799     945             3,81    5       65      15,643                          864     11      100     845     65      28      880     5
#N1:1033 chr01   960     1106            3,44    4       52,5    11,325  1022    74      100                             1034    42      5
#N1:1191 chr01   1118    1264            2,99    3       100     1,732   1190    17      100
#N1:1340 chr01   1267    1413            2,05    3       50      24,007  1340    74      100                             1380    16      0
#N1:1585 chr01   1512    1658    "+5*:YAL068C; " 3,83    4       50      5,916   1594    67      100                             1582    22      0
#N1:1902 chr01   1829    1975    "+3:YAL068C; "  2,67    4       50      35,968  1893    325     100                             1972    8       0

cat Pugh_measured_nucleosomes.txt | tr "\r" "\n" | cut -f 2 | uniq -c

cat Pugh_measured_nucleosomes.txt | tr "\r" "\n" | cut -f 2-5 | sed s/"; "/";"/g | head

cat Pugh_measured_nucleosomes.txt | tr "\r" "\n" | cut -f 2-6 | sed 1d | sed s/"; "/";"/g | awk '{OFS="\t"}{$2-=1;print $0}'> Pugh_measured_nucleosomes_roman0.bed
#I strongly assume that the coordinates are NOT 0-based (its called start position and end position), so I convert to bed 0-based half-open...

python ~/ms_tools/convert_chr_names2.py -f roman0 -t latin Pugh_measured_nucleosomes_roman0.bed > Pugh_measured_nucleosomes.bed

head -20 Pugh_measured_nucleosomes.bed
#chrI    118     265     1,11
#chrI    308     455     3,61
#chrI    556     703     4,31
#chrI    798     945     3,81
#chrI    959     1106    3,44
#chrI    1117    1264    2,99
#chrI    1266    1413    2,05
#chrI    1511    1658    "+5*:YAL068C;"       3,83
#chrI    1828    1975    "+3:YAL068C;"       2,67
#chrI    2007    2154    "+2:YAL068C;"       2,86
#chrI    2167    2314    "+1:YAL068C;-1:YAL067W-A;"       2,76
#chrI    2322    2469    "-1:YAL068C;+1:YAL067W-A;"       3,74
#chrI    2597    2744    "+3:YAL067W-A;"       4,92
#chrI    2761    2908    "+4*:YAL067W-A;"       5,04
#chrI    2954    3101    4,36
#chrI    3139    3286    4,59
#chrI    3296    3443    3,77
#chrI    3535    3682    2,05
#chrI    3718    3865    4,35
#chrI    4008    4155    3,01

#many nucleosomes are not associated with a gene
# others are associated with multiple genes, although this is probably rare
awk '$4 ~ /\+/' Pugh_measured_nucleosomes.bed | wc -l
#   50334

awk '$4 ~ /\+/ && $4 ~ /\-/' Pugh_measured_nucleosomes.bed | wc -l
#   4794



sort -k1 Pugh_measured_nucleosomes.bed | cut -f 1 | uniq -c
#1183 chrI
#4134 chrII
#1600 chrIII
#7748 chrIV
#2249 chrIX
#2924 chrV
#1368 chrVI
#5527 chrVII
#2863 chrVIII
#3617 chrX
#3371 chrXI
#5447 chrXII
#4713 chrXIII
#4007 chrXIV
#5533 chrXV
#4826 chrXVI

#Here we only select the nucleosomes with a plus annotation as first annotation in the Excel sheet position column
cat Pugh_measured_nucleosomes.bed | awk '{OFS="\t"}{if($4 ~ /\"\+/){sub(";.*","",$4);sub("^\"","",$4);print $0}}' | cut -f 1-4 > Pugh_measured_plus_nucleosomes.bed

head Pugh_measured_plus_nucleosomes.bed
#chrI    1511    1658    +5*:YAL068C
#chrI    1828    1975    +3:YAL068C
#chrI    2007    2154    +2:YAL068C
#chrI    2167    2314    +1:YAL068C
#chrI    2597    2744    +3:YAL067W-A
#chrI    2761    2908    +4*:YAL067W-A
#chrI    5107    5254    +7*:SUT432
#chrI    5258    5405    +6:SUT432
#chrI    5410    5557    +5:SUT432
#chrI    5582    5729    +4:SUT432




awk '$4 ~ /\+1:/' Pugh_measured_nucleosomes.bed > Pugh_measured_nucleosomes_1plus.bed


#nucleosome coverage in various ways
cat Pugh_measured_nucleosomes.bed | \
awk '{OFS="\t"}{
  if($4 ~ /:/){
    val=$5
  }else{
    val=$4
  };
  sub(",",".",val);
  print $1"\t"$2"\t"$3"\t"val
}' > Pugh_measured_nucleosomes_coverage.bedGraph

/Users/schmidm/ms_tools/bedGraphToBigWig Pugh_measured_nucleosomes_coverage.bedGraph /Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_genome_latin.info Pugh_measured_nucleosomes_coverage.bw

# !! fails because chromosomes a few nt too long
# sacCer1 and sacCer2 have same problem !?
# perhaps caused by calling nucleosomes procedure ... ie extend to defined width 146 ?? ...
awk '{print $3-$2}' Pugh_measured_nucleosomes_coverage.bedGraph | uniq -c
#61110 147


## in any case, it should be compatible with sacCer3 and hence I simply do
bedtools slop -i Pugh_measured_nucleosomes_coverage.bedGraph -g /Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_genome_latin.info -b 0 > Pugh_measured_nucleosomes_coverage_trimmed.bedGraph


diff Pugh_measured_nucleosomes_coverage.bedGraph Pugh_measured_nucleosomes_coverage_trimmed.bedGraph
#1183c1183
#< chrI  230082  230229  1.99
#---
#> chrI  230082  230218  1.99 ##removing 11bp on right
#5317c5317
#< chrII 813075  813222  1.00
#---
#> chrII 813075  813184  1.00 ##removing 38bp on right
#6917c6917
#< chrIII        316488  316635  2.02
#---
#> chrIII        316488  316620  2.02 ##removing 25bp on right
#18957c18957
#< chrVI 270089  270236  1.02
#---
#> chrVI 270089  270161  1.02  ##removing 75bp on right
#29596c29596
#< chrIX 439771  439918  2.71
#---
#> chrIX 439771  439888  2.71  ##removing 30bp on right
#46744c46744
#< chrXIII       924287  924434  2.08
#---
#> chrXIII       924287  924431  2.08   ##removing 3bp on right
#50751c50751
#< chrXIV        784187  784334  1.06
#---
#> chrXIV        784187  784333  1.06 ##removing 1bp on right

### overall very little trimming done


/Users/schmidm/ms_tools/bedGraphToBigWig \
Pugh_measured_nucleosomes_coverage_trimmed.bedGraph \
/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_genome_latin.info \
Pugh_measured_nucleosomes.bw
```



# nucleosome density rel TSS and TES

First check out how those data map relative Steinmetz annotations. From all we know that should be some very sharp positioning, its more like a sanity check.  

#### make a nucleosome center track
```{bash, eval = FALSE}
### NUCLEOSOME CENTERS
##note all nucleosomes are made 146bp wide, this script sets the center at the dyad position

awk '{half=($3-$2-1)/2; $2+=half;$3-=half;print $0}' Pugh_measured_nucleosomes_coverage_trimmed.bedGraph > \
Pugh_measured_nucleosomes_center_coverage.bedGraph

/Users/schmidm/ms_tools/bedGraphToBigWig \
Pugh_measured_nucleosomes_center_coverage.bedGraph \
/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_genome_latin.info \
Pugh_measured_nucleosomes_center.bw

```


#### nucleosomes metagene profiles
```{bash, eval = FALSE}
cd /Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes

bed="/Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_annotations_sacCer3_latin_ORFs.bed"

awk '{OFS="\t"}{$4=$7":"$8":"$5":"$4;$5="0";print $0}' $bed | cut -f 1-6 > ${bed/.bed/_correctformat.bed}

bwtool matrix 1000:1000 -starts -keep-bed ${bed/.bed/_correctformat.bed} Pugh_measured_nucleosomes_center.bw bwtool_nuc_center_rel_ORF_TSS.txt

bwtool matrix 1000:1000 -ends -keep-bed ${bed/.bed/_correctformat.bed} Pugh_measured_nucleosomes_center.bw bwtool_nuc_center_rel_ORF_TES.txt

```


```{r, eval = FALSE}
nuc_rel_tss <- load_bwtool_nuc_matrix('/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/bwtool_nuc_center_rel_ORF_TSS.txt', 1000, 1000, both_strands = FALSE)


nuc_rel_tes <- load_bwtool_nuc_matrix('/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/bwtool_nuc_center_rel_ORF_TES.txt', 1000, 1000, both_strands = FALSE)

save(nuc_rel_tss, file='data/nucleosome_centers_rel_TSS.RData')
save(nuc_rel_tes, file='data/nucleosome_centers_rel_TES.RData')
```

```{r, eval = TRUE}
load('data/nucleosome_centers_rel_TSS.RData', verbose=TRUE)
load('data/nucleosome_centers_rel_TES.RData', verbose=TRUE)
```


make common theme for prettier and annotated TSS plots
```{r}
nuc_rel_TSS <- data.frame(
  name = c('+1', '+2', '+3', '+4', '+5', '+6', 'up1', 'up2', 'up3'),
  center = c(50, 215, 380, 545, 710, 875, -225, -390, -555)
  ) %>%
  mutate(end = center + 74,
         start = center - 74)

nucleosome_rel_tss_theme <- list(scale_x_continuous(minor_breaks=NULL, breaks=nuc_rel_TSS$center),
                         scale_y_continuous(breaks=NULL),
  geom_vline(xintercept = nuc_rel_TSS$center, color='orange', linetype=2),
  geom_vline(xintercept = 0, color='black'), 
  geom_rect(data=nuc_rel_TSS,
            aes(xmin = start, xmax = end,
                ymin=-5, ymax=0), fill = 'gray',
            inherit.aes = FALSE),
  geom_text(data=nuc_rel_TSS,
            aes(x=center, y=-2.5, label=name),
            inherit.aes = FALSE))
```

```{r nucleosomes around ORF-T TSS}
nuc_rel_tss %>%
  ggplot(., aes(x=pos, y=sum_log2_value)) + 
  geom_line() +
  xlab('bp tp TSS') +
  ylab('nucleosome centers\n( sum of log2(peak height) )') + 
  theme_bw() +
  nucleosome_rel_tss_theme
```

--> extrememly well-defined phasing (165 nt periodicity) of nucleosome centers with respect to TSS, which lies inside +1 nucleosome.


```{r nucleosomes around ORF-T TES}
nuc_rel_tes %>%
  mutate(roll_sum_log2_value = rollmean(sum_log2_value, k=20, fill = NA)) %>%
  ggplot(., aes(x=pos, y=roll_sum_log2_value)) + 
  geom_line(color='black') +
  theme_bw() +
  xlab('bp to TES') +
  ylab('nucleosome centers\n( sum of log2(peak height) )') +
    nucleosome_rel_tss_theme[c(1:4)]
```

--> phasing and a nucleosome-depleted region is also present at the TES but the strength of the depletion and positioning of the nucleosomes around is much(!) less pronounced compared to the TSS.  

--> strikingly the TES phases with pretty much some distance as TSS ie TSS are often identical to TES from adjacent gene !?



# 3'seq data rel nucleosomes

### testing using bwtools

#### bwtool matrix instead
```{bash, eval = FALSE}
#!/usr/bin/env bash

cd ~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/

bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus_nucleosomes.bed"
bwtool matrix 1000:1000 -starts -keep-bed $bed norm_xPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub.bw bwtool_out/all_plus_nuc_mat.txt

bwtool matrix 1000:1000 -ends -keep-bed $bed norm_xPap_Mex67AA_ip_0_1_minus_KevinRoyAfiltered_BGsub.bw bwtool_out/all_minus_nuc_mat.txt


##reverse
bwtool matrix 1000:1000 -ends -keep-bed $bed norm_xPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub.bw bwtool_out/all_plus_nuc_mat_rev.txt

bwtool matrix 1000:1000 -starts -keep-bed $bed norm_xPap_Mex67AA_ip_0_1_minus_KevinRoyAfiltered_BGsub.bw bwtool_out/all_minus_nuc_mat_rev.txt


### INPUT
bwtool matrix 1000:1000 -starts -keep-bed $bed norm_xPappAminus_Mex67AA_input_0_1_plus_KevinRoyAfiltered.bw bwtool_out/all_plus_nuc_totalpAmin_mat.txt
```



```{r}
plus <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/bwtool_out/norm_xPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub_all_plus_nuc_mat.txt')

(dplus <- gather_bwtool_matrix(plus, 1000, 1000) %>%
  mutate(strand = '+'))
```

```{r}
minus <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/bwtool_out/norm_xPap_Mex67AA_ip_0_1_minus_KevinRoyAfiltered_BGsub_all_minus_nuc_mat.txt')

(dminus <- gather_bwtool_matrix(minus, 1000, 1000) %>%
  mutate(pos = -pos, strand = '-'))
```


```{r}
d_agg <- bind_rows(dplus, dminus) %>% 
  filter(value > 0) %>%
  group_by(nuc, pos, strand) %>%
  summarize(events=n(),
            sum_value = sum(value),
            sum_log2_value = sum(log2(value)))
```


```{r 3pseq data all nucleosomes vs strand}
ggplot(d_agg, aes(x=pos, y=events, color=nuc)) + 
  geom_line() +
  facet_grid(.~strand)
```

```{r 3pseq data plus1to4 nucleosomes vs strand}
d_agg %>%
  filter(as.numeric(nuc) < 5) %>%
  group_by(nuc, strand) %>%
  mutate(roll_mean_events = rollmean(events, k=10, fill = NA)) %>%
  ggplot(., aes(x=pos, y=roll_mean_events, color=nuc)) + 
  geom_line() +
  facet_grid(nuc ~ strand) +
  scale_color_brewer(palette = 'BrBG', direction = -1) +
  theme_bw()
```

--> OK we can savely assume that strands are handled "correct"

## common theme for prettier and plot funs for annotated nucleosome array plots
```{r}
nuc_rel_nuc <- data.frame(
  name = c('dn1', 'dn2', 'dn3', 'up1', 'up2', 'up3'),
  border = c(0, 165, 330, -165, -330, -495)
)

nucleosome_rel_nuc_theme <- list(scale_x_continuous(minor_breaks=NULL, breaks=nuc_rel_nuc$border),
                         scale_y_continuous(breaks=NULL),
  geom_vline(xintercept = nuc_rel_nuc$border, color='orange', linetype=2),
  geom_vline(xintercept = 0, color='black'))
```

```{r}
plus_nuc_plot <- function (df) {
  df %>%
  filter(as.numeric(nuc) < 5) %>%
  group_by(nuc) %>%
  mutate(roll_sum_log2_value = rollmean(sum_log2_value, k=10, fill = NA)) %>%
  ggplot(., aes(x=pos, y=roll_sum_log2_value, color=nuc)) + 
    geom_line() +
    scale_color_brewer(palette = 'BrBG', direction = -1) +
  facet_grid(nuc~.) +
  theme_bw() +
  xlab('bp to nucleosome upstream border') +
    nucleosome_rel_nuc_theme
}
```


```{r}
term_nuc_plot <- function (df) {
df %>%
  filter(nuc == 'term') %>%
  group_by(nuc) %>%
  mutate(roll_sum_log2_value = rollmean(sum_log2_value, k=10, fill = NA)) %>%
  ggplot(., aes(x=pos, y=roll_sum_log2_value)) + 
  geom_line(color='black') +
  facet_grid(nuc~.) +
  theme_bw() +
  xlab('bp to terminal nucleosome upstream border') +
    nucleosome_rel_nuc_theme
}
```



## pAplusandminus IP

```{bash, eval = FALSE}
#!/usr/bin/env bash

cd ~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/

bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus_nucleosomes.bed"

bw="norm_xPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub.bw"
bwtool matrix 1000:1000 -starts -keep-bed $bed $bw bwtool_out/${bw/.bw/_all_plus_nuc_mat.txt}

bw_minus=${bw/plus/minus}
bwtool matrix 1000:1000 -ends -keep-bed $bed $bw_minus bwtool_out/${bw_minus/.bw/_all_minus_nuc_mat.txt}
```


```{r}
mat_fname <- 'norm_xPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub_all_plus_nuc_mat.txt'

sample_name <- sub('_KevinRoyAfiltered', '', mat_fname) %>%
  sub('_all_plus_nuc_mat.txt', '', .)

RData_filename <- paste0('data/', sample_name, '_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
agg_3seq <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(agg_3seq, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r ip pAplusandminus plus nucleosomes}
plus_nuc_plot(agg_3seq) +
  ggtitle(sample_name)
```

```{r ip pAplusandminus terminal nucleosome}
term_nuc_plot(agg_3seq) +
  ggtitle(sample_name)
```




## rel downstream border


```{bash, eval = FALSE}
#!/usr/bin/env bash

cd ~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/

bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus_nucleosomes.bed"

bw="norm_xPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub.bw"
bwtool matrix 1000:1000 -ends -keep-bed $bed $bw bwtool_out/${bw/.bw/_all_plus_nuc_ends_mat.txt}

bw_minus=${bw/plus/minus}
bwtool matrix 1000:1000 -starts -keep-bed $bed $bw_minus bwtool_out/${bw_minus/.bw/_all_minus_nuc_ends_mat.txt}

```


```{r}
mat_fname <- 'norm_xPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub_all_plus_nuc_ends_mat.txt'

sample_name <- sub('_KevinRoyAfiltered', '', mat_fname) %>%
  sub('_all_plus_nuc', '', .) %>%
  sub('_mat.txt', '', .)

RData_filename <- paste0('data/', sample_name, '_rel_nucleosome_end_bwtool.RData')
```

```{r, eval = FALSE}
agg_relend_3seq <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(agg_relend_3seq, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r ip pAplusandminus plus nucleosomes rel downstream border}
plus_nuc_plot(agg_relend_3seq) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome downstream border')
```

```{r ip pAplusandminus terminal nucleosome rel downstream border}
term_nuc_plot(agg_relend_3seq) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome downstream border')
```

## pAplus IP

```{bash, eval = FALSE}
#!/usr/bin/env bash

cd ~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/

bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus_nucleosomes.bed"

bw="norm_noPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub.bw"
bwtool matrix 1000:1000 -starts -keep-bed $bed $bw bwtool_out/${bw/.bw/_all_plus_nuc_mat.txt}

bw_minus=${bw/plus/minus}
bwtool matrix 1000:1000 -ends -keep-bed $bed $bw_minus bwtool_out/${bw_minus/.bw/_all_minus_nuc_mat.txt}
```


```{r}
mat_fname <- 'norm_noPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub_all_plus_nuc_mat.txt'

sample_name <- sub('_KevinRoyAfiltered', '', mat_fname) %>%
  sub('_all_plus_nuc_mat.txt', '', .)

RData_filename <- paste0('data/', sample_name, '_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
agg_ip_pAplus <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(agg_ip_pAplus, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r ip pAplus plus nucleosomes}
plus_nuc_plot(agg_ip_pAplus) +
  ggtitle(sample_name)
```

```{r ip pAplus terminal nucleosome}
term_nuc_plot(agg_ip_pAplus) +
  ggtitle(sample_name)
```



## pAminus IP

```{bash, eval = FALSE}
#!/usr/bin/env bash

cd ~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/

bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus_nucleosomes.bed"

bw="norm_xPappAminus_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub.bw"
bwtool matrix 1000:1000 -starts -keep-bed $bed $bw bwtool_out/${bw/.bw/_all_plus_nuc_mat.txt}

bw_minus=${bw/plus/minus}
bwtool matrix 1000:1000 -ends -keep-bed $bed $bw_minus bwtool_out/${bw_minus/.bw/_all_minus_nuc_mat.txt}
```


```{r}
mat_fname <- 'norm_xPappAminus_Mex67AA_ip_0_1_plus_KevinRoyAfiltered_BGsub_all_plus_nuc_mat.txt'

sample_name <- sub('_KevinRoyAfiltered', '', mat_fname) %>%
  sub('_all_plus_nuc_mat.txt', '', .)

RData_filename <- paste0('data/', sample_name, '_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
agg_ip_pAminus <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(agg_ip_pAminus, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r ip pAminus plus nucleosomes}
plus_nuc_plot(agg_ip_pAminus) +
  ggtitle(sample_name)
```

```{r ip pAminus terminal nucleosome}
term_nuc_plot(agg_ip_pAminus) +
  ggtitle(sample_name)
```





## pAplus input samples

```{bash, eval = FALSE}
bw="norm_noPap_Mex67AA_input_0_1_plus_KevinRoyAfiltered.bw"
bwtool matrix 1000:1000 -starts -keep-bed $bed $bw bwtool_out/${bw/.bw/_all_plus_nuc_mat.txt}

bw_minus=${bw/_plus/_minus}
bwtool matrix 1000:1000 -ends -keep-bed $bed $bw_minus bwtool_out/${bw_minus/.bw/_all_minus_nuc_mat.txt}
```


```{r}
mat_fname <- 'norm_noPap_Mex67AA_input_0_1_plus_KevinRoyAfiltered_all_plus_nuc_mat.txt'

sample_name <- sub('_KevinRoyAfiltered', '', mat_fname) %>%
  sub('_all_plus_nuc_mat.txt', '', .)

RData_filename <- paste0('data/', sample_name, '_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
agg_pAplus_in <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(agg_pAplus_in, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r input pAplus plus nucleosomes}
plus_nuc_plot(agg_pAplus_in) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```

```{r input pAplus terminal nucleosome}
term_nuc_plot(agg_pAplus_in) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```




## 3'seq pAminus input samples

```{bash, eval = FALSE}
bw="norm_xPappAminus_Mex67AA_input_0_1_plus_KevinRoyAfiltered.bw"
bwtool matrix 1000:1000 -starts -keep-bed $bed $bw bwtool_out/${bw/.bw/_all_plus_nuc_mat.txt}

bw_minus=${bw/_plus/_minus}
bwtool matrix 1000:1000 -ends -keep-bed $bed $bw_minus bwtool_out/${bw_minus/.bw/_all_minus_nuc_mat.txt}
```


```{r}
mat_fname <- 'norm_xPappAminus_Mex67AA_input_0_1_plus_KevinRoyAfiltered_all_plus_nuc_mat.txt'

sample_name <- sub('_KevinRoyAfiltered', '', mat_fname) %>%
  sub('_all_plus_nuc_mat.txt', '', .)

RData_filename <- paste0('data/', sample_name, '_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
agg_pAminus_in <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(agg_pAminus_in, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r input pAminus plus nucleosomes}
plus_nuc_plot(agg_pAminus_in) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```

```{r input pAminus terminal nucleosome}
term_nuc_plot(agg_pAminus_in) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```


## 3'seq pAmasked input samples



# CRAC data rel nucleosomes
```{bash, eval = FALSE}
#!/usr/bin/env bash
bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus_nucleosomes.bed"
bwtool matrix 1000:1000 -starts -keep-bed $bed GSM1706520_Rpo21_nabFT_plus.bw bwtool_out/GSM1706520_Rpo21_nabFT_plus_nuc_mat.txt

bwtool matrix 1000:1000 -ends -keep-bed $bed GSM1706520_Rpo21_nabFT_minus.bw bwtool_out/GSM1706520_Rpo21_nabFT_minus_nuc_mat.txt
```


```{r}
mat_fname <- 'GSM1706520_Rpo21_nabFT_plus_nuc_mat.txt'

sample_name <- sub('_plus_nuc_mat.txt', '', mat_fname) 

RData_filename <- paste0('data/', sample_name, '_CRAC_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
crac_nucs <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Tollervey_RNAP_CRAC/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(crac_nucs, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r CRAC Milligan plus nucleosomes}
plus_nuc_plot(crac_nucs) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```

```{r CRAC Milligan terminal nucleosome}
term_nuc_plot(crac_nucs) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```




# NETseq data rel nucleosomes
```{bash, eval = FALSE}
#!/usr/bin/env bash

cd /Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Churchman_NETseq

bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus_nucleosomes.bed"
bwtool matrix 1000:1000 -starts -keep-bed $bed GSM617027_WT_NC_plus_sacCer3.bw bwtool_out/GSM617027_WT_NC_plus_nuc_mat.txt

bwtool matrix 1000:1000 -ends -keep-bed $bed GSM617027_WT_NC_minus_sacCer3.bw bwtool_out/GSM617027_WT_NC_minus_nuc_mat.txt


bwtool matrix 1000:1000 -starts -keep-bed $bed GSM617030_DST1D_plus_sacCer3.bw bwtool_out/GSM617030_DST1D_plus_nuc_mat.txt

bwtool matrix 1000:1000 -ends -keep-bed $bed GSM617030_DST1D_minus_sacCer3.bw bwtool_out/GSM617030_DST1D_minus_nuc_mat.txt
```


#### NETseq wt
```{r}
mat_fname <- 'GSM617027_WT_NC_plus_nuc_mat.txt'

sample_name <- sub('_plus_nuc_mat.txt', '', mat_fname) 

RData_filename <- paste0('data/', sample_name, '_NETseq_WT_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
NETseq_wt_nucs <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Churchman_NETseq/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(NETseq_wt_nucs, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r NETseq wt plus nucleosomes}
plus_nuc_plot(NETseq_wt_nucs) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```

```{r NETseq wt terminal nucleosome}
term_nuc_plot(NETseq_wt_nucs) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```



#### NETseq dst1∆
```{r}
mat_fname <- 'GSM617030_DST1D_plus_nuc_mat.txt'

sample_name <- sub('_plus_nuc_mat.txt', '', mat_fname) 

RData_filename <- paste0('data/', sample_name, '_NETseq_dst1_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
NETseq_dst1_nucs <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Churchman_NETseq/bwtool_out/', mat_fname), 1000, 1000, both_strands = TRUE)

save(NETseq_dst1_nucs, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r NETseq dst1 plus nucleosomes}
plus_nuc_plot(NETseq_dst1_nucs) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```

```{r NETseq dst1 terminal nucleosome}
term_nuc_plot(NETseq_dst1_nucs) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```


## Rpb3 ChIPseq data rel nucleosomes
```{bash, eval = FALSE}
#!/usr/bin/env bash

cd /Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Henikoff_Rpb3FLAG_ChIPseq

bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_nucleosomes_plus_unique_with_strand.bed"
bwtool matrix 1000:1000 -starts -keep-bed $bed GSM2551210_WT_A_140.bw bwtool_out/GSM2551210_WT_A_140_unique_plus_nuc_mat.txt
```


#### Rpb3 ChIPseq
```{r}
mat_fname <- 'GSM2551210_WT_A_140_unique_plus_nuc_mat.txt'

sample_name <- sub('_unique_plus_nuc_mat.txt', '', mat_fname) 

RData_filename <- paste0('data/', sample_name, '_Rpb3_ChIPseq_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
Rpb3_ChIPseq_nucs <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Henikoff_Rpb3FLAG_ChIPseq/', mat_fname), 1000, 1000, both_strands = TRUE)

save(Rpb3_ChIPseq_nucs, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r Rpb3 ChIPseq plus nucleosomes}
Rpb3_ChIPseq_nucs %>%
  filter(nuc != '-1', nuc != '0') %>%
  plus_nuc_plot(.) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```

```{r Rpb3 ChIPseq terminal nucleosome}
term_nuc_plot(Rpb3_ChIPseq_nucs) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```



```{r}
sessionInfo()
```



## Rpb3 ChIP on chip rel nucleosomes
```{bash, eval = FALSE}
#!/usr/bin/env bash

cd /Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Cramer_ChIPchip

##nucleosomes

bed="/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_nucleosomes_plus_unique_with_strand.bed"

bwtool matrix 1000:1000 -starts -keep-bed $bed Cramer_E_TABM_1033_Rpb3_InputAndMockNormalized_sacCer3.bw Cramer_Rpb3_unique_plus_nuc_mat.txt
```


```{r}
mat_fname <- 'Cramer_Rpb3_unique_plus_nuc_mat.txt'

sample_name <- sub('_unique_plus_nuc_mat.txt', '', mat_fname) 

RData_filename <- paste0('data/', sample_name, '_ChIPonchip_rel_nucleosome_start_bwtool.RData')
```

```{r, eval = FALSE}
Rpb3_ChIPonchip_nucs <- load_bwtool_nuc_matrix(paste0('/Users/schmidm/Documents/genomewide_datasets/ChIP_CLiP_yeast/Cramer_ChIPchip/', mat_fname), 1000, 1000, both_strands = TRUE)

save(Rpb3_ChIPonchip_nucs, file=RData_filename)
```


```{r, eval = TRUE}
load(RData_filename, verbose=TRUE)
```


```{r Rpb3 ChIPonchip plus nucleosomes}
Rpb3_ChIPonchip_nucs %>%
  filter(nuc != '-1', nuc != '0') %>%
  plus_nuc_plot(.) +
  ggtitle(sample_name) +
  facet_grid(nuc~., scales='free')+
  xlab('bp to nucleosome border')
```

```{r Rpb3 ChIPonchip terminal nucleosome}
term_nuc_plot(Rpb3_ChIPonchip_nucs) +
  ggtitle(sample_name) +
  xlab('bp to nucleosome border')
```



```{r}
sessionInfo()
```
