---
title: "Create pAminus Tracks by Masking pA+ Positions"
author: "Manfred Schmid"
date: "9 August 2017"
output: html_document
---

## pAminus_masked = pA++pA- minus all signals from positions with pA+ signal

Using S.pombe normalized values: get positions from pA+ positions from 0rapa libraries

ups:
--> no specific cut-off used here!  
--> very conservative, masks even files where pA+ contributes very little signal in EPAP libraries


#### counting pA-masked files
```{bash, eval=FALSE}
## NORMALIZED to S.pombe and pAplus masked track files
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph/

## noPap filter using control (rapa=0min) samples only
plus_files=`ls *noPap_*_*plus*.bedgraph | grep -v 15 | grep -v 70 | tr "\n" " "`

plus_names=`echo $plus_files | sed 's/norm_//g' | sed 's/_plus_KevinRoyAfiltered.bedgraph//g' `
echo $plus_files
#noPap_Mex67AA_input_0_1_plus_KevinRoyAfiltered.bedgraph noPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered.bedgraph noPap_Mex67AA_ip_neg0_1_plus_KevinRoyAfiltered.bedgraph noPap_Nab2AA_input_0_1_plus_KevinRoyAfiltered.bedgraph noPap_Nab2AA_ip_0_1_plus_KevinRoyAfiltered.bedgraph

##get all positions for plus and minus strand
bedtools unionbedg -header -i $plus_files -names $plus_names | \
awk '{sum=$4+$5+$6+$7+$8; print $1"\t"$2"\t"$3"\t"sum}' | sed 1d | sort -k1,1 -k2,2n > noPaP_0rapa_sites_plus.bedgraph

minus_files=${plus_files/plus/minus}
echo $minus_files
#noPap_Mex67AA_input_0_1_minus_KevinRoyAfiltered.bedgraph noPap_Mex67AA_ip_0_1_plus_KevinRoyAfiltered.bedgraph noPap_Mex67AA_ip_neg0_1_plus_KevinRoyAfiltered.bedgraph noPap_Nab2AA_input_0_1_plus_KevinRoyAfiltered.bedgraph noPap_Nab2AA_ip_0_1_plus_KevinRoyAfiltered.bedgraph

minus_names=`echo $minus_files | sed 's/norm_//g' | sed 's/_plus_KevinRoyAfiltered.bedgraph//g' `

bedtools unionbedg -header -i $minus_files -names $minus_names | \
awk '{sum=$4+$5+$6+$7+$8; print $1"\t"$2"\t"$3"\t"sum}' | sed 1d | sort -k1,1 -k2,2n > noPaP_0rapa_sites_minus.bedgraph

head noPaP_0rapa_sites_*.bedgraph
#==> noPaP_0rapa_sites_minus.bedgraph <==
#chrI    1719    1720    1
#chrI    1756    1757    1
#chrI    5027    5028    1
#chrI    6234    6235    1
#chrI    6241    6242    2
#chrI    6246    6247    2
#chrI    6286    6287    2
#chrI    6517    6518    3
#chrI    6775    6776    1
#chrI    6956    6957    1
#
#==> noPaP_0rapa_sites_plus.bedgraph <==
#chrI    1756    1757    1
#chrI    6234    6235    1
#chrI    6241    6242    2
#chrI    6246    6247    2
#chrI    6286    6287    2
#chrI    6887    6888    1
#chrI    6956    6957    1
#chrI    7017    7018    1
#chrI    7177    7178    2
#chrI    7225    7226    1

wc -l noPaP_0rapa_sites_*.bedgraph
#  222650 noPaP_0rapa_sites_minus.bedgraph
#  174194 noPaP_0rapa_sites_plus.bedgraph
#  396844 total


grep -v ^chr noPaP_0rapa_sites_plus.bedgraph | wc -l
# 0
grep -v ^chr noPaP_0rapa_sites_minus.bedgraph | wc -l
# 0
## these files only have cerevisiae chromosomes


awk '{sum += ($3-$2)}END{print sum}' noPaP_0rapa_sites_*.bedgraph
# 402286
## this is the actual number of sites, not big difference to clusters, ie most are individual isolated sites
```



#### count pA+ positions per body and end
```{bash, eval = FALSE}
awk '{
  for(i=$2; i < $3; ++i){
    print $1"\t"i"\t"i+1"\t1"
  }
}' noPaP_0rapa_sites_plus.bedgraph  > tmp

bedtools map -a ${body_plus_bed} -b tmp -c 4 -o sum > noPap_0rapa_sites_per_cleaned_body_plus.txt


awk '{
  for(i=$2; i < $3; ++i){
    print $1"\t"i"\t"i+1"\t1"
  }
}' noPaP_0rapa_sites_minus.bedgraph  > tmp


bedtools map -a ${body_minus_bed} -b tmp -c 4 -o sum > noPap_0rapa_sites_per_cleaned_body_minus.txt


cat noPap_0rapa_sites_per_cleaned_body_plus.txt noPap_0rapa_sites_per_cleaned_body_minus.txt > noPap_0rapa_sites_per_cleaned_body.txt
rm tmp


head noPap_0rapa_sites_per_cleaned_body.txt
#chrI    30071   30704   ST0002:CUTs:CUT001:CUT001       7
#chrI    31151   32784   ST0003:ORF-T:YAL062W:GDH3       49
#chrI    33359   34696   ST0004:ORF-T:YAL061W:BDH2       2
#chrI    35096   36192   ST0005:ORF-T:YAL060W:BDH1       7
#chrI    36592   37128   ST0006:ORF-T:YAL059W:ECM1       .
#chrI    37528   38832   ST0007:ORF-T:YAL058W:CNE1       2
#chrI    39232   41768   ST0008:ORF-T:YAL056W:GPB2       5
#chrI    42168   42632   ST0009:ORF-T:YAL055W:PEX22      .
#chrI    43439   45128   ST0010:SUTs:SUT002:SUT002       11
#chrI    45759   48104   ST0011:ORF-T:YAL053W:FLC2       4




#### masked sites on gene ends ####


awk '{
  for(i=$2; i < $3; ++i){
    print $1"\t"i"\t"i+1"\t1"
  }
}' noPaP_0rapa_sites_plus.bedgraph  > tmp

bedtools map -a ${ends_plus_bed} -b tmp -c 4 -o sum > noPap_0rapa_sites_per_end_plus.txt


awk '{
  for(i=$2; i < $3; ++i){
    print $1"\t"i"\t"i+1"\t1"
  }
}' noPaP_0rapa_sites_minus.bedgraph  > tmp


bedtools map -a ${ends_minus_bed} -b tmp -c 4 -o sum > noPap_0rapa_sites_per_end_minus.txt


cat noPap_0rapa_sites_per_end_plus.txt noPap_0rapa_sites_per_end_minus.txt > noPap_0rapa_sites_per_end.txt
rm tmp


head noPap_0rapa_sites_per_end.txt
#chrI    9400    9800    ST0001:SUTs:SUT001:SUT001       11
#chrI    30704   31104   ST0002:CUTs:CUT001:CUT001       10
#chrI    32784   33184   ST0003:ORF-T:YAL062W:GDH3       24
#chrI    34696   35096   ST0004:ORF-T:YAL061W:BDH2       22
#chrI    36192   36592   ST0005:ORF-T:YAL060W:BDH1       55
#chrI    37128   37528   ST0006:ORF-T:YAL059W:ECM1       40
#chrI    38832   39232   ST0007:ORF-T:YAL058W:CNE1       46
#chrI    41768   42168   ST0008:ORF-T:YAL056W:GPB2       51
#chrI    42632   43032   ST0009:ORF-T:YAL055W:PEX22      23
#chrI    45128   45528   ST0010:SUTs:SUT002:SUT002       11

```



#### masking from normalized track files

```{bash, eval = FALSE}
### for S.p normalized files remove noPap_0rapa positions

# input-filtering A-filtered bigwigs location
filter_dir="/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph/"

cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bedgraph


#subtract noPap 0rapa sites
mkdir only_pAminus_positions

for f in norm_xPap_*input*_plus_KevinRoyAfiltered.bedgraph
do
  echo $f
  bedtools subtract -a $f -b ${filter_dir}"noPaP_0rapa_sites_plus.bedgraph" > \
  only_pAminus_positions/${f/.bedgraph/_wo_noPaP_0rapa_sites.bedgraph}
done

for f in norm_xPap_*ip*_plus_KevinRoyAfiltered_BGsub.bedgraph
do
  echo $f
  bedtools subtract -a $f -b ${filter_dir}"noPaP_0rapa_sites_plus.bedgraph" > \
  only_pAminus_positions/${f/.bedgraph/_wo_noPaP_0rapa_sites.bedgraph}
done

for f in norm_xPap_*input*_minus_KevinRoyAfiltered.bedgraph
do
  echo $f
  bedtools subtract -a $f -b ${filter_dir}"noPaP_0rapa_sites_minus.bedgraph" > \
  only_pAminus_positions/${f/.bedgraph/_wo_noPaP_0rapa_sites.bedgraph}
done

for f in norm_xPap_*ip*_minus_KevinRoyAfiltered_BGsub.bedgraph
do
  echo $f
  bedtools subtract -a $f -b ${filter_dir}"noPaP_0rapa_sites_minus.bedgraph" > \
  only_pAminus_positions/${f/.bedgraph/_wo_noPaP_0rapa_sites.bedgraph}
done


```


#### convert to bigwigs

```{bash, eval = FALSE}
## convert to bigwigs
genome_info="/Users/schmidm//Documents/genomewide_datasets/annotations/sacCer3/sacCer3_genome_latin_chrM.info"

cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bedgraph/only_pAminus_positions/

for f in *.bedgraph
  do
    echo "sorting ${f}"
    sort -k1,1 -k2,2n $f -o $f
    echo "  and converting to bigwig"
    /Users/schmidm/ms_tools/bedGraphToBigWig $f $genome_info ../../norm_and_pA_filtered_bw/only_pAminus_positions/${f/.bedgraph/.bw}
  done

```


#### remove empty files

negative IP minus background gives all 0 values -> empty bigwigs, remove those

```{bash, eval = FALSE}
rm /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/only_pAminus_positions/*ip_neg*BGsub.bw
```

ups .. perhaps not necessary?? or already deleted before???

```{r}
sessionInfo()
```



