---
title: 'snRNA signal around TES single nucleotide resolution'
author: "Manfred Schmid"
output: html_document
---

`r format(Sys.time(), "%d %B, %Y")`

```{r, echo=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, 
                      fig.path=paste0('Figures_snRNA_precise_end_counts/'), 
                      dev='pdf', echo=TRUE, warning=FALSE, message=FALSE, 
                      error=TRUE)
```


```{r load packages, echo=T, warning=F, message=F}
suppressWarnings(library(tidyverse))
suppressWarnings(library(magrittr))
suppressWarnings(library(knitr))
```

```{bash, eval=FALSE}
#!/usr/bin/env bash

cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bedgraph



sgdSNR="/Users/schmidm//Documents/genomewide_datasets/annotations/sacCer3/sgdSNRs_sacCer3.bed"

awk '{OFS="\t"}{if($6=="+"){$2=$3-10;$3+=10; print $0}}' ${sgdSNR} | cut -f 1-6 > ${sgdSNR/.bed/_TESpm10_plus.bed}
awk '{OFS="\t"}{if($6=="-"){$3=$2+10;$2-=10; print $0}}' ${sgdSNR} | cut -f 1-6 > ${sgdSNR/.bed/_TESpm10_minus.bed}


##single example plus strand
f="norm_xPappAminus_Nab2AA_input_15_1_plus_KevinRoyAfiltered.bedgraph"
bedtools intersect -wao -a ${sgdSNR/.bed/_TESpm10_plus.bed} -b $f > tmp
head tmp
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142459  142460  4.99304 1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142461  142462  4.99304 1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142462  142463  54.9234 1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142463  142464  114.84  1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142464  142465  139.805 1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142465  142466  2076.92716      1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142466  142467  66348.2747      1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142467  142468  352954.198      1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142468  142469  6353.55358      1
#chrI    142458  142478  snR18   snoRNA_gene     +       chrI    142469  142470  4517.22395      1
#--> clearly position 142467 has most reads, thats the exact 3' end of snR18

grep snR18 ${sgdSNR}
#chrI    142366  142468  snR18   snoRNA_gene     +  ##note half-open !! so the end = 142467
## a little awk to transform this into relative positions
awk '{start=$8;end=$9;for(i=start;i<end;i++){print $4"\t"i-$2-9"\t"$10}}' tmp | head
#snR18   -8      4.99304
#snR18   -6      4.99304
#snR18   -5      54.9234
#snR18   -4      114.84
#snR18   -3      139.805
#snR18   -2      2076.92716
#snR18   -1      66348.2747
#snR18   0       352954.198
#snR18   1       6353.55358
#snR18   2       4517.22395



##single example plus strand
f="norm_xPappAminus_Nab2AA_input_15_1_plus_KevinRoyAfiltered.bedgraph"
bedtools intersect -wao -a ${sgdSNR/.bed/_TESpm10_minus.bed} -b ${f/plus/minus} > tmp
head tmp
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307176  307179  4.99304 3
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307179  307180  29.9582 1
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307180  307181  14.9791 1
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307181  307182  9.98607 1
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307182  307183  44.9373 1
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307183  307184  559.22  1
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307184  307185  35461.62037     1
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307185  307186  84.8816 1
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307186  307187  339.527 1
#chrII   307174  307194  snR161  snoRNA_gene     -       chrII   307187  307188  19.9721 1

#--> clearly position 307184 has most reads, thats the exact 3' end of snR161

grep snR161 ${sgdSNR}
#chrII   307184  307345  snR161  snoRNA_gene     - ##perfect

## a little awk to transform this into relative positions
awk '{start=$8;end=$9;for(i=start;i<end;i++){print $4"\t"$3-i-10"\t"$10}}' tmp | head
#snR161  8       4.99304
#snR161  7       4.99304
#snR161  6       4.99304
#snR161  5       29.9582
#snR161  4       14.9791
#snR161  3       9.98607
#snR161  2       44.9373
#snR161  1       559.22
#snR161  0       35461.62037
#snR161  -1      84.8816




## for all samples:
for f in *ip*_plus_KevinRoyAfiltered_BGsub.bedgraph
  do
    echo $f
    bedtools intersect -wao -a ${sgdSNR/.bed/_TESpm10_plus.bed} -b $f | \
    awk '{start=$8;end=$9;for(i=start;i<end;i++){print $4"\t"i-$2-9"\t"$10}}' > tmp.plus

    bedtools intersect -wao -a ${sgdSNR/.bed/_TESpm10_minus.bed} -b ${f/plus/minus}  | \
    awk '{start=$8;end=$9;for(i=start;i<end;i++){print $4"\t"$3-i-10"\t"$10}}' > tmp.minus

    cat tmp.plus tmp.minus > bedtools_counts/${f/_plus_KevinRoyAfiltered_BGsub.bedgraph/_KevinRoyAfiltered_BGsub_snR_mature_TESpm10.counts}
  done



for f in *input*plus_KevinRoyAfiltered.bedgraph
  do
    echo $f
    bedtools intersect -wao -a ${sgdSNR/.bed/_TESpm10_plus.bed} -b $f | \
    awk '{start=$8;end=$9;for(i=start;i<end;i++){print $4"\t"i-$2-9"\t"$10}}' > tmp.plus

    bedtools intersect -wao -a ${sgdSNR/.bed/_TESpm10_minus.bed} -b ${f/plus/minus}  | \
    awk '{start=$8;end=$9;for(i=start;i<end;i++){print $4"\t"$3-i-10"\t"$10}}' > tmp.minus

    cat tmp.plus tmp.minus > bedtools_counts/${f/_plus_KevinRoyAfiltered.bedgraph/_KevinRoyAfiltered_snR_mature_TESpm10.counts}
  done

rm tmp.plus
rm tmp.minus
```


```{r}
path <- '/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bedgraph/bedtools_counts/'

sfx <- '_snR_mature_TESpm10.counts$'
fnames <- dir(path) %>% keep(grepl(sfx, .))

(d <- lapply(fnames, function(f) {read.table(paste0(path,f), 
                                           col.names=c('id', 'rel_pos', 'norm_counts')) %>%
    mutate(norm_counts = ifelse(norm_counts == '.', 0, as.numeric(norm_counts)),
           condition = sub(sfx, '', f) %>%
             sub('norm_', '', .))}) %>%
    bind_rows %>%
    separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='warn') %>%
    tbl_df)
```


script above misses empty positions ...
```{r}
d %>%
  group_by(rel_pos) %>%
  summarize(n_with_pos = n()) %>%
  kable
```


add missing values ...
```{r}
(d %<>% 
  spread(rel_pos, norm_counts) %>% 
  gather(rel_pos, norm_counts, -id, -Pap, -strain, -fraction, -rapa, -rep) %>%
  mutate(norm_counts = ifelse(is.na(norm_counts), 0, norm_counts),
         rel_pos = as.integer(rel_pos)))
```

worked, now have same amount of data for all positions
```{r}
d %>%
  group_by(rel_pos) %>%
  summarize(n_with_pos = n()) %>%
  kable
```


```{r}
medians_per_position <- d %>%
  group_by(rel_pos, Pap, strain, fraction, rapa, rep) %>%
  summarize(median_norm_counts = median(norm_counts))

means_per_position <- d %>%
  group_by(rel_pos, Pap, strain, fraction, rapa, rep) %>%
  summarize(mean_norm_counts = mean(norm_counts))
```

```{r distribution points and mean line and 0pos as text}
d %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=rel_pos, y=norm_counts)) + 
  geom_point(size=.4) + 
  geom_line(data=filter(means_per_position, rapa=='0'), 
            aes(x=rel_pos, y=mean_norm_counts),
            color='red') +
  geom_text(data=filter(means_per_position, rapa=='0', rel_pos == 0),
            aes(x=rel_pos, y=1000000, label=round(mean_norm_counts,0)), color='red') +
  scale_y_log10() +
  facet_grid(strain+fraction~Pap+rep, scales='free') +
  xlim(-10,10) +
  theme_bw()
```

```{r distribution points and median line and 0pos as text}
d %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=rel_pos, y=norm_counts)) + 
  geom_point(size=.4) + 
  geom_line(data=filter(medians_per_position, rapa=='0'), 
            aes(x=rel_pos, y=median_norm_counts),
            color='red') +
  geom_text(data=filter(medians_per_position, rapa=='0', rel_pos == 0),
            aes(x=rel_pos, y=1000000, label=round(median_norm_counts,0)), color='red') +
  scale_y_log10() +
  facet_grid(strain+fraction~Pap+rep, scales='free') +
  xlim(-10,10) +
  theme_bw()
```


```{r distribution points}
d %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=rel_pos, y=norm_counts)) + 
  geom_point(size=.4, shape=3) + 
  facet_grid(strain+fraction~Pap+rep, scales='free') +
  xlim(-10,10) +
  theme_bw()
```

```{r distribution points log y}
d %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=rel_pos, y=norm_counts)) + 
  geom_point(size=.4, shape=3) + 
  scale_y_log10() +
  facet_grid(strain+fraction~Pap+rep, scales='free') +
  xlim(-10,10) +
  theme_bw()
```

```{r distribution and mean line and 0pos as text}
means_per_position %>%
  filter(rapa == '0') %>%
ggplot(., aes(x=rel_pos, y=mean_norm_counts)) + 
  geom_line() +
  geom_text(data=filter(means_per_position, rapa=='0', rel_pos == 0),
            aes(x=rel_pos, y=250000, label=round(mean_norm_counts,0)), 
            color='red') +
  scale_y_log10(limits=c(.001,1000000)) +
  facet_grid(strain+fraction~Pap+rep, scales='free') +
  xlim(-10,10) +
  theme_bw()
```



# for mean of replicates

```{r}
d_means <- d %>%
  filter(rapa == '0') %>%
  group_by(id, Pap, strain, fraction, rapa, rel_pos) %>%
  summarise(norm_counts_mean_of_replicates = mean(norm_counts))

```

```{r}
means_of_means_per_position <- d_means %>%
  group_by(rel_pos, Pap, strain, fraction, rapa) %>%
  summarize(mean_norm_counts_mean_of_replicates = mean(norm_counts_mean_of_replicates))
```

```{r}
medians_of_means_per_position <- d_means %>%
  group_by(rel_pos, Pap, strain, fraction, rapa) %>%
  summarize(median_norm_counts_mean_of_replicates = median(norm_counts_mean_of_replicates))
```


```{r distribution points and mean line and 0pos as text mean of replicates}
ggplot(d_means, aes(x=rel_pos, y=norm_counts_mean_of_replicates)) + 
  geom_point(size=.4) + 
  geom_line(data=filter(means_of_means_per_position, rapa=='0'), 
            aes(x=rel_pos, y=mean_norm_counts_mean_of_replicates),
            color='red') +
  geom_text(data=filter(means_of_means_per_position, rapa=='0', rel_pos == 0),
            aes(x=rel_pos, y=1000000, label=round(mean_norm_counts_mean_of_replicates,0)), color='red') +
  scale_y_log10() +
  facet_grid(strain+fraction~Pap, scales='free') +
  xlim(-10,10) +
  theme_bw()
```

```{r distribution points and median line and 0pos as text mean of replicates}
ggplot(d_means, aes(x=rel_pos, y=norm_counts_mean_of_replicates)) + 
  geom_point(size=.4) + 
  geom_line(data=filter(medians_of_means_per_position, rapa=='0'), 
            aes(x=rel_pos, y=median_norm_counts_mean_of_replicates),
            color='red') +
  geom_text(data=filter(medians_of_means_per_position, rapa=='0', rel_pos == 0),
            aes(x=rel_pos, y=1000000, label=round(median_norm_counts_mean_of_replicates,0)), color='red') +
  scale_y_log10() +
  facet_grid(strain+fraction~Pap, scales='free') +
  xlim(-10,10) +
  theme_bw()
```



```{r}
sessionInfo()
```
