---
title: "Normalize to S.pombe spike-ins"
author: "Manfred Schmid"
output: html_document
---
`r format(Sys.time(), "%d %B, %Y")`


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



## S.pombe counts

#### S. pombe annotations used

```{bash, eval = FALSE}
#!/usr/bin/env bash

### get S.pombe annotations for gene TSS to TES + 300bp
cd /Users/schmidm/Documents/genomewide_datasets/annotations/Spombe_ENSEMBL_EF2

#extract exons and convert to 0-based
cat Sp_genes.gtf | awk '{if($3=="exon"){sub("\"","",$12);sub("\";","",$12);print $1"\t"$4-1"\t"$5"\t"$12"\t"$3"\t"$7}}' > Sp_exons.bed

#combine exons of each annotation and extend end by 300bp after TES
awk '$6=="+"' Sp_exons.bed | sort -k4,4 -k2,2n | \
awk '
{
  if($4==i){
    e=$3+300
  }else{
    print c"\t"s"\t"e"\t"i"\t"t"\t"str; 
    c=$1;s=$2;e=$3+300;i=$4;t=$5;str=$6
  }
}END{
  print c"\t"s"\t"e"\t"i"\t"t"\t"str
}' | sed 1d | sort -k1,1 -k2,2n > Sp_exons_ends_p300_plus.bed

awk '$6=="-"' Sp_exons.bed | sort -k4,4 -k2,2n | awk '{if($4==i){e=$3}else{print c"\t"s"\t"e"\t"i"\t"t"\t"str; c=$1;s=$2-300;if(s < 0){s=0};e=$3;i=$4;t=$5;str=$6}}END{print c"\t"s"\t"e"\t"i"\t"t"\t"str}' | sed 1d | sort -k1,1 -k2,2n > Sp_exons_ends_p300_minus.bed


## improved version removing overlaps between neighbors on same strand

awk '{OFS="\t"}{if(chr == $1 && $2 < end){if(end<$3){$2=end;$5="trimmed_gene_plus300bp";print $0}}else{$5="gene_plus300bp"; print $0};chr=$1;end=$3}' Sp_exons_ends_p300_plus.bed > Sp_gene_ends_p300_nonoverlapping_plus.bed 

awk '{OFS="\t"}{if(chr == $1 && $2 < end){if(end<$3){$2=end;$5="trimmed_gene_plus300bp";print $0}}else{$5="gene_plus300bp"; print $0};chr=$1;end=$3}' Sp_exons_ends_p300_minus.bed > Sp_gene_ends_p300_nonoverlapping_minus.bed 

```



#### counts for each Spombe transcript 10min data
```{bash, eval = FALSE}
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph

mkdir Spombe_bedgraphs
for f in *.bedgraph
  do 
    grep -v ^chr $f > Spombe_bedgraphs/${f/.bedgraph/_Sp.bedgraph}
  done

mkdir Spombe_per_gene_counts
plus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/Spombe_ENSEMBL_EF2/Sp_gene_ends_p300_nonoverlapping_plus.bed"
minus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/Spombe_ENSEMBL_EF2/Sp_gene_ends_p300_nonoverlapping_minus.bed"

cd /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph/Spombe_bedgraphs
mkdir Spombe_per_gene_counts

for b in *_plus_Sp.bedgraph
do

  echo "counting plus strand file ${b}"
  bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $plus_bed $b 4 Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_plus.raw_counts}

  echo "   minus strand"
  bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $minus_bed ${b/plus/minus} 4 Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_minus.raw_counts}

  cat Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_plus.raw_counts} Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_minus.raw_counts} | sort -k1,1 -k2,2n > Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp.raw_counts}

  rm Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_plus.raw_counts}
  rm Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_minus.raw_counts}
done
```


#### counts for each Spombe transcript 2min data
```{bash, eval = FALSE}
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/raw_bedgraph

mkdir Spombe_bedgraphs
for f in *.bedgraph
  do 
    grep -v ^chr $f > Spombe_bedgraphs/${f/.bedgraph/_Sp.bedgraph}
  done

mkdir Spombe_per_gene_counts
plus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/Spombe_ENSEMBL_EF2/Sp_gene_ends_p300_nonoverlapping_plus.bed"
minus_bed="/Users/schmidm/Documents/genomewide_datasets/annotations/Spombe_ENSEMBL_EF2/Sp_gene_ends_p300_nonoverlapping_minus.bed"

cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/raw_bedgraph/Spombe_bedgraphs
mkdir Spombe_per_gene_counts

for b in *_plus_Sp.bedgraph
do

  echo "counting plus strand file ${b}"
  bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $plus_bed $b 4 Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_plus.raw_counts}

  echo "   minus strand"
  bash /Users/schmidm/ms_tools/MS_Metagene_Tools/bed_count.sh $minus_bed ${b/plus/minus} 4 Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_minus.raw_counts}

  cat Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_plus.raw_counts} Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_minus.raw_counts} | sort -k1,1 -k2,2n > Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp.raw_counts}

  rm Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_plus.raw_counts}
  rm Spombe_per_gene_counts/${b/_plus_Sp.bedgraph/_Sp_minus.raw_counts}
done
```


## DEseq2 for Spombe size factors

```{r, message=FALSE}
suppressWarnings(library('tidyverse'))
suppressWarnings(library('magrittr'))
suppressWarnings(library('DESeq2'))
```

Next step --> feed these counts into DESeq2 and get the sizefactors.



### 10min data
```{r}
path <-
  '/Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph/Spombe_bedgraphs/Spombe_per_gene_counts/'

sfx <- '_Sp.raw_counts$'

file_list <- dir(path)[grep(sfx, dir(path))]

(
  Sp_10min_counts_df <- lapply(file_list, function(fname) {
    read.table(
      paste0(path, fname, sep = ''),
      col.names =
        c('chr', 'start', 'end', 'id', 'type', 'strand', 'count'),
      stringsAsFactors = FALSE
    ) %>%
      tbl_df %>%
      mutate(condition = sub(sfx, '', fname))
  }) %>%
    bind_rows() %>%
  mutate(count = ifelse(count == '.', 0, as.numeric(count)))
)
```



make count matrix
```{r}
Sp_10min_count_mat <- Sp_10min_counts_df %>% 
  dplyr::select(id,count,condition) %>%
  tidyr::spread(condition, count) %>%
  remove_rownames %>%
  data.frame %>%
  column_to_rownames(var='id') %>%
  as.matrix

head(Sp_10min_count_mat)
```


make colData  
```{r}
Sp_10min_coldata <- data.frame(condition = colnames(Sp_10min_count_mat) ) %>%
  tidyr::separate(condition, c('strain', 'fraction', 'rapa', 'replicate'), by='_') %>%
  mutate(condition = paste0(strain, '_', fraction, '_', rapa),
         replicate = ifelse(is.na(replicate), 0, replicate))

rownames(Sp_10min_coldata) <- colnames(Sp_10min_count_mat)

Sp_10min_coldata
```


make DESeq object and get sizeFactors
```{r}
Sp_ddsFullCountTable <- DESeqDataSetFromMatrix(
countData = Sp_10min_count_mat,
colData = Sp_10min_coldata,
design = ~ condition)

Sp_10min_genes_sf <- sizeFactors(estimateSizeFactors(Sp_ddsFullCountTable))

data.frame(Sp_10min_genes_sf)
```


```{r}
write.table(Sp_10min_genes_sf, file='data/Sp_10min_genes_sf.txt', col.names=F, quote=F)
```




### 2min data

same as above for 10min data

```{r}
path <- '/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/raw_bedgraph/Spombe_bedgraphs/Spombe_per_gene_counts/'

sfx <- '_Sp.raw_counts$'

file_list <- dir(path)[grep(sfx, dir(path))]

(Sp_counts_df <- lapply(file_list, function(fname) {read.table(paste0(path,fname,sep=''), col.names=c('chr', 'start', 'end', 'id', 'type', 'strand', 'count'), stringsAsFactors = FALSE) %>% 
    tbl_df %>%
  mutate(condition = sub(sfx, '', fname))}) %>%
  bind_rows() %>%
  mutate(count = ifelse(count == '.', 0, as.numeric(count))))
```



make count matrix
```{r}
Sp_count_mat <- Sp_counts_df %>% 
  dplyr::select(id,count,condition) %>%
  tidyr::spread(condition, count) %>%
  remove_rownames %>%
  data.frame %>%
  column_to_rownames(var='id') %>%
  as.matrix

head(Sp_count_mat)
```


make colData  
```{r}
Sp_coldata <- data.frame(condition = colnames(Sp_count_mat) ) %>%
  tidyr::separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'replicate'), by='_') %>%
  mutate(condition = paste0(Pap, '_', strain, '_', fraction, '_', rapa))

rownames(Sp_coldata) <- colnames(Sp_count_mat)

Sp_coldata
```


make DESeq object and get sizeFactors
```{r}
Sp_ddsFullCountTable <- DESeqDataSetFromMatrix(
countData = Sp_count_mat,
colData = Sp_coldata,
design = ~ condition)

Sp_genes_sf <- sizeFactors(estimateSizeFactors(Sp_ddsFullCountTable))

data.frame(Sp_genes_sf)
```



#### correct size factor for xPap_Nab2AA_ip_0_1 sample

note this was a mistake already realized during pipetting, ie only for sample xPap_Nab2AA_ip_0_1 was there used 3x more spike-in. This only affects xPap_Nab2AA_ip_0_1 as all other samples are mix of ip_0_2 and ip_0_3.

```{r}
Sp_genes_sf['xPap_Nab2AA_ip_0_1'] <- Sp_genes_sf['xPap_Nab2AA_ip_0_1']/3

data.frame(Sp_genes_sf)
```



```{r}
write.table(Sp_genes_sf, file='data/Sp_2min_genes_sf.txt', col.names=F, quote=F)
```



## use size factors for normalizing track files

### 10min tracks

```{bash, eval=FALSE}
#!/bin/bash

cd /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/KevinRoyAfiltered_bedgraph
mkdir ../norm_and_pA_filtered_bedgraph

cat /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/Rproject/data/Sp_10min_genes_sf.txt | \
while read sf
  do
    sf_map=(${sf/ / })
    bdgp="${sf_map[0]}_plus_KevinRoyAfiltered.bedgraph"
    echo "doing ${bdgp}"
    echo "  sf: ${sf}"
    awk -v sf="${sf_map[1]}" '{OFS="\t"}{$4/=sf; print $0}' ${bdgp} > ../norm_and_pA_filtered_bedgraph/"norm_"${bdgp}

    bdgp="${sf_map[0]}_minus_KevinRoyAfiltered.bedgraph"
    echo "doing ${bdgp}"
    awk -v sf="${sf_map[1]}" '{OFS="\t"}{$4/=sf; print $0}' ${bdgp} > ../norm_and_pA_filtered_bedgraph/"norm_"${bdgp}
  done
```


#### sanity check
```{bash, eval=FALSE}
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/norm_and_pA_filtered_bedgraph

rm tmp
for f in norm*_plus_KevinRoyAfiltered.bedgraph
  do
    echo $f
    echo ${f/_plus_KevinRoyAfiltered.bedgraph/} >> tmp
    awk '{sum+=$4*($3-$2)}END{print sum}' $f ${f/plus/minus} >> tmp
  done

cat tmp | paste - - > total_reads_after_norm.txt
rm tmp

cat total_reads_after_norm.txt
#norm_Mex_in_0_1	7.02814e+06
#norm_Mex_in_0_2	6.7443e+06
#norm_Mex_in_0_3	7.04208e+06
#norm_Mex_in_15_1	5.04728e+06
#norm_Mex_in_15_2	8.65957e+06
#norm_Mex_in_15_3	4.78368e+06
#norm_Mex_in_70_1	5.08323e+06
#norm_Mex_in_70_2	5.48049e+06
#norm_Mex_in_70_3	4.6289e+06
#norm_Mex_in_neg	1.17791e+07
#norm_Mex_ip_0_1	6.76162e+06
#norm_Mex_ip_0_2	5.60938e+06
#norm_Mex_ip_0_3	5.96017e+06
#norm_Mex_ip_15_1	1.3863e+06
#norm_Mex_ip_15_2	2.47173e+06
#norm_Mex_ip_15_3	1.29019e+06
#norm_Mex_ip_70_1	1.51664e+06
#norm_Mex_ip_70_2	1.63447e+06
#norm_Mex_ip_70_3	1.33936e+06
#norm_Mex_ip_neg	1.02529e+06
#norm_Nab_in_0_1	8.15257e+06
#norm_Nab_in_0_2	6.00691e+06
#norm_Nab_in_0_3	5.69447e+06
#norm_Nab_in_15_1	5.30869e+06
#norm_Nab_in_15_2	6.63003e+06
#norm_Nab_in_15_3	5.22273e+06
#norm_Nab_in_70_1	5.70082e+06
#norm_Nab_in_70_2	3.82502e+06
#norm_Nab_in_70_3	3.95804e+06
#norm_Nab_in_neg	5.99288e+06
#norm_Nab_ip_0_1	4.78605e+06
#norm_Nab_ip_0_2	3.74188e+06
#norm_Nab_ip_0_3	3.23521e+06
#norm_Nab_ip_15_1	1.58206e+06
#norm_Nab_ip_15_2	2.16421e+06
#norm_Nab_ip_15_3	1.55521e+06
#norm_Nab_ip_70_1	1.69428e+06
#norm_Nab_ip_70_2	1.21088e+06
#norm_Nab_ip_70_3	1.44463e+06
#norm_Nab_ip_neg	574306
#norm_noPap_in_0	8.21961e+06
#norm_noPap_in_70	6.68217e+06
#norm_noPap_ip_0	2.00051e+06
#norm_noPap_ip_70	1.50898e+06
#norm_xPap_in_0	3.56043e+07
#norm_xPap_in_70	4.35005e+07
#norm_xPap_ip_0	8.27244e+06
#norm_xPap_ip_70	7.88152e+06
```


```{r}
(norm_reads_10min <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/norm_and_pA_filtered_bedgraph/total_reads_after_norm.txt',
           col.names=c('file', 'norm_reads')))
```

```{r norm reads barplot 10min data}
norm_reads_10min %>%
  separate(file, c('norm', 'strain', 'fraction', 'rapa', 'rep'), sep='_') %>%
  ggplot(., aes(x=rep, y=norm_reads)) +
  geom_bar(stat='identity', position='dodge') +
  facet_grid(fraction~strain+rapa, scales='free')
```

--> similar samples have overall similar read counts 
--> Mex67AA 15 min input replicate 2 is off, but also in total mapping statistics this sample has unexpectedly few S.pombe reads. And is also off in inputs -> probably pipetting error.



### 2min tracks
```{bash, eval=FALSE}
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph
mkdir ../norm_and_pA_filtered_bedgraph

cat /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/Rproject/data/Sp_2min_genes_sf.txt | \
while read sf
  do
    sf_map=(${sf/ / })
    bdgp="${sf_map[0]}_plus_KevinRoyAfiltered.bedgraph"
    echo "doing ${bdgp}"
    echo "  sf: ${sf}"
    awk -v sf="${sf_map[1]}" '{OFS="\t"}{$4/=sf; print $0}' ${bdgp} > ../norm_and_pA_filtered_bedgraph/"norm_"${bdgp}

    bdgp="${sf_map[0]}_minus_KevinRoyAfiltered.bedgraph"
    echo "doing ${bdgp}"
    awk -v sf="${sf_map[1]}" '{OFS="\t"}{$4/=sf; print $0}' ${bdgp} > ../norm_and_pA_filtered_bedgraph/"norm_"${bdgp}
  done
```


####  sanity check

```{bash, eval=FALSE}
## before S.pombe normalization
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph/

rm tmp
for f in *_plus_KevinRoyAfiltered.bedgraph
  do
    echo ${f/_plus_KevinRoyAfiltered.bedgraph/} >> tmp
    awk '{sum+=$4*($3-$2)}END{print sum}' $f ${f/plus/minus} >> tmp
  done

cat tmp | paste - - > total_reads_before_norm.txt
rm tmp


cat total_reads_before_norm.txt
#noPap_Mex67AA_input_0_1	2202503
#noPap_Mex67AA_input_15_1	2963985
#noPap_Mex67AA_input_70_1	2387948
#noPap_Mex67AA_ip_0_1	2579454
#noPap_Mex67AA_ip_15_1	1837564
#noPap_Mex67AA_ip_70_1	1054532
#noPap_Mex67AA_ip_neg0_1	3116325
#noPap_Nab2AA_input_0_1	2313940
#noPap_Nab2AA_input_15_1	3235115
#noPap_Nab2AA_ip_0_1	1769572
#noPap_Nab2AA_ip_15_1	1905171
#xPap_Mex67AA_input_0_1	3654903
#xPap_Mex67AA_input_15_1	5390170
#xPap_Mex67AA_input_70_1	3825320
#xPap_Mex67AA_ip_0_1	2648241
#xPap_Mex67AA_ip_0_2	3022976
#xPap_Mex67AA_ip_0_3	2310471
#xPap_Mex67AA_ip_15_1	2904716
#xPap_Mex67AA_ip_15_2	3379732
#xPap_Mex67AA_ip_15_3	2072147
#xPap_Mex67AA_ip_70_1	4265182
#xPap_Mex67AA_ip_70_2	2950139
#xPap_Mex67AA_ip_70_3	2784526
#xPap_Mex67AA_ip_neg0_1	3153307
#xPap_Nab2AA_input_0_1	3822077
#xPap_Nab2AA_input_15_1	3530828
#xPap_Nab2AA_ip_0_1	1888025
#xPap_Nab2AA_ip_0_2	2815418
#xPap_Nab2AA_ip_0_3	2214857
#xPap_Nab2AA_ip_15_1	1901498
#xPap_Nab2AA_ip_15_2	2208254
#xPap_Nab2AA_ip_15_3	1902227



## after S.pombenormalization
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bedgraph

rm tmp
for f in norm*_plus_KevinRoyAfiltered.bedgraph
  do
    echo ${f/_plus_KevinRoyAfiltered.bedgraph/} >> tmp
    awk '{sum+=$4*($3-$2)}END{print sum}' $f ${f/plus/minus} >> tmp
  done

cat tmp | paste - - > total_reads_after_norm.txt
rm tmp


cat total_reads_after_norm.txt
#norm_noPap_Mex67AA_input_0_1	4.99852e+06
#norm_noPap_Mex67AA_input_15_1	3.96362e+06
#norm_noPap_Mex67AA_input_70_1	3.32303e+06
#norm_noPap_Mex67AA_ip_0_1	508882
#norm_noPap_Mex67AA_ip_15_1	320987
#norm_noPap_Mex67AA_ip_70_1	240968
#norm_noPap_Mex67AA_ip_neg0_1	349911
#norm_noPap_Nab2AA_input_0_1	5.16833e+06
#norm_noPap_Nab2AA_input_15_1	4.09354e+06
#norm_noPap_Nab2AA_ip_0_1	427518
#norm_noPap_Nab2AA_ip_15_1	299170
#norm_xPap_Mex67AA_input_0_1	1.69155e+07
#norm_xPap_Mex67AA_input_15_1	1.74843e+07
#norm_xPap_Mex67AA_input_70_1	1.49722e+07
#norm_xPap_Mex67AA_ip_0_1	3.31228e+06
#norm_xPap_Mex67AA_ip_0_2	3.46139e+06
#norm_xPap_Mex67AA_ip_0_3	3.58673e+06
#norm_xPap_Mex67AA_ip_15_1	2.73184e+06
#norm_xPap_Mex67AA_ip_15_2	3.64444e+06
#norm_xPap_Mex67AA_ip_15_3	2.92028e+06
#norm_xPap_Mex67AA_ip_70_1	2.21209e+06
#norm_xPap_Mex67AA_ip_70_2	1.92183e+06
#norm_xPap_Mex67AA_ip_70_3	2.11992e+06
#norm_xPap_Mex67AA_ip_neg0_1	1.3927e+06
#norm_xPap_Nab2AA_input_0_1	1.62348e+07
#norm_xPap_Nab2AA_input_15_1	1.79021e+07
#norm_xPap_Nab2AA_ip_0_1	2.26037e+06
#norm_xPap_Nab2AA_ip_0_2	2.89161e+06
#norm_xPap_Nab2AA_ip_0_3	2.55998e+06
#norm_xPap_Nab2AA_ip_15_1	3.54336e+06
#norm_xPap_Nab2AA_ip_15_2	3.64858e+06
#norm_xPap_Nab2AA_ip_15_3	2.92977e+06
```


#### normalized reads check in R

```{r}
(raw_reads <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph/total_reads_before_norm.txt',
           col.names=c('file', 'raw_reads')))
```

```{r raw reads barplot 2min data}
raw_reads %>%
  separate(file, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_') %>%
  ggplot(., aes(x=rep, y=raw_reads)) +
  geom_bar(stat='identity', position='dodge') +
  facet_grid(Pap+fraction~strain+rapa, scales='free')
```

```{r}
(norm_reads <- read.table('/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bedgraph/total_reads_after_norm.txt',
           col.names=c('file', 'norm_reads')))
```

```{r norm reads barplot 2min data}
norm_reads %>%
  separate(file, c('norm', 'Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_') %>%
  ggplot(., aes(x=rep, y=norm_reads)) +
  geom_bar(stat='identity', position='dodge') +
  facet_grid(Pap+fraction~strain+rapa, scales='free')
```

--> similar samples have very similar read counts but between ie ip vs input and xPap vs noPap there are huge differences.


## compare counts relative to size factors

```{r}
(Sp_genes_sf <- read.table('data/Sp_2min_genes_sf.txt', 
                          col.names=c('file', 'sf')))
```


```{r sizeFactor vs raw rel norm ratios 2min data}
left_join(raw_reads, norm_reads %>%
            mutate(file=sub('norm_', '', file))) %>%
  left_join(., Sp_genes_sf) %>%
  mutate(ratio = raw_reads/norm_reads) %>%
  ggplot(., aes(x=sf, y=ratio)) +
  geom_point() +
  geom_abline(slope=1)
```

```{r}
sessionInfo()
```

