{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Perform the QC and mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. QC_and_mapping.Rmd\n",
    "    + adapter and quality trimming of raw reads\n",
    "    + mapping using STAR aligner\n",
    "    + obtain stats from the above steps."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR6423303\n",
      "\n",
      "gzip: SRR6423303.fastq.gz: unexpected end of file\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12469m -Xms12469m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423303.fastq out=SRR6423303_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423303.fastq, out=SRR6423303_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.023 seconds.\n",
      "Initial:\n",
      "Memory: max=13077m, total=13077m, free=13045m, used=32m\n",
      "\n",
      "Added 446 kmers; time: \t0.016 seconds.\n",
      "Memory: max=13077m, total=13077m, free=13033m, used=44m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.093 seconds.\n",
      "Processing time:   \t\t5.561 seconds.\n",
      "\n",
      "Input:                  \t4481601 reads \t\t224080050 bases.\n",
      "QTrimmed:               \t325893 reads (7.27%) \t691696 bases (0.31%)\n",
      "KTrimmed:               \t209193 reads (4.67%) \t5519946 bases (2.46%)\n",
      "Total Removed:          \t74797 reads (1.67%) \t6211642 bases (2.77%)\n",
      "Result:                 \t4406804 reads (98.33%) \t217868408 bases (97.23%)\n",
      "\n",
      "Time:                         \t5.671 seconds.\n",
      "Reads Processed:       4481k \t790.26k reads/sec\n",
      "Bases Processed:        224m \t39.51m bases/sec\n",
      "Aligning...\n",
      "May 17 14:54:09 ..... started STAR run\n",
      "May 17 14:54:09 ..... loading genome\n",
      "May 17 14:54:09 ..... started mapping\n",
      "May 17 14:54:39 ..... started sorting BAM\n",
      "May 17 14:54:42 ..... finished successfully\n",
      "SRR6423305\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12647m -Xms12647m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423305.fastq out=SRR6423305_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423305.fastq, out=SRR6423305_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.022 seconds.\n",
      "Initial:\n",
      "Memory: max=13262m, total=13262m, free=13231m, used=31m\n",
      "\n",
      "Added 446 kmers; time: \t0.015 seconds.\n",
      "Memory: max=13262m, total=13262m, free=13218m, used=44m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t3.121 seconds.\n",
      "\n",
      "Input:                  \t4866432 reads \t\t243321600 bases.\n",
      "QTrimmed:               \t223997 reads (4.60%) \t455575 bases (0.19%)\n",
      "KTrimmed:               \t520395 reads (10.69%) \t21938015 bases (9.02%)\n",
      "Total Removed:          \t417681 reads (8.58%) \t22393590 bases (9.20%)\n",
      "Result:                 \t4448751 reads (91.42%) \t220928010 bases (90.80%)\n",
      "\n",
      "Time:                         \t3.141 seconds.\n",
      "Reads Processed:       4866k \t1549.53k reads/sec\n",
      "Bases Processed:        243m \t77.48m bases/sec\n",
      "Aligning...\n",
      "May 17 14:54:52 ..... started STAR run\n",
      "May 17 14:54:52 ..... loading genome\n",
      "May 17 14:54:53 ..... started mapping\n",
      "May 17 14:55:22 ..... started sorting BAM\n",
      "May 17 14:55:28 ..... finished successfully\n",
      "SRR6423306\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12644m -Xms12644m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423306.fastq out=SRR6423306_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423306.fastq, out=SRR6423306_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.022 seconds.\n",
      "Initial:\n",
      "Memory: max=13258m, total=13258m, free=13226m, used=32m\n",
      "\n",
      "Added 446 kmers; time: \t0.015 seconds.\n",
      "Memory: max=13258m, total=13258m, free=13212m, used=46m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t2.596 seconds.\n",
      "\n",
      "Input:                  \t4189502 reads \t\t209475100 bases.\n",
      "QTrimmed:               \t183139 reads (4.37%) \t379485 bases (0.18%)\n",
      "KTrimmed:               \t626994 reads (14.97%) \t26996218 bases (12.89%)\n",
      "Total Removed:          \t514331 reads (12.28%) \t27375703 bases (13.07%)\n",
      "Result:                 \t3675171 reads (87.72%) \t182099397 bases (86.93%)\n",
      "\n",
      "Time:                         \t2.616 seconds.\n",
      "Reads Processed:       4189k \t1601.22k reads/sec\n",
      "Bases Processed:        209m \t80.06m bases/sec\n",
      "Aligning...\n",
      "May 17 14:55:36 ..... started STAR run\n",
      "May 17 14:55:36 ..... loading genome\n",
      "May 17 14:55:36 ..... started mapping\n",
      "May 17 14:56:02 ..... started sorting BAM\n",
      "May 17 14:56:10 ..... finished successfully\n",
      "SRR6423307\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12658m -Xms12658m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423307.fastq out=SRR6423307_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423307.fastq, out=SRR6423307_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.022 seconds.\n",
      "Initial:\n",
      "Memory: max=13274m, total=13274m, free=13245m, used=29m\n",
      "\n",
      "Added 446 kmers; time: \t0.016 seconds.\n",
      "Memory: max=13274m, total=13274m, free=13230m, used=44m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t3.447 seconds.\n",
      "\n",
      "Input:                  \t5306967 reads \t\t265348350 bases.\n",
      "QTrimmed:               \t243235 reads (4.58%) \t534053 bases (0.20%)\n",
      "KTrimmed:               \t605371 reads (11.41%) \t20160230 bases (7.60%)\n",
      "Total Removed:          \t332911 reads (6.27%) \t20694283 bases (7.80%)\n",
      "Result:                 \t4974056 reads (93.73%) \t244654067 bases (92.20%)\n",
      "\n",
      "Time:                         \t3.467 seconds.\n",
      "Reads Processed:       5306k \t1530.72k reads/sec\n",
      "Bases Processed:        265m \t76.54m bases/sec\n",
      "Aligning...\n",
      "May 17 14:56:21 ..... started STAR run\n",
      "May 17 14:56:21 ..... loading genome\n",
      "May 17 14:56:21 ..... started mapping\n",
      "May 17 14:56:59 ..... started sorting BAM\n",
      "May 17 14:57:07 ..... finished successfully\n",
      "SRR6423308\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12645m -Xms12645m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423308.fastq out=SRR6423308_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423308.fastq, out=SRR6423308_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.023 seconds.\n",
      "Initial:\n",
      "Memory: max=13262m, total=13262m, free=13232m, used=30m\n",
      "\n",
      "Added 446 kmers; time: \t0.016 seconds.\n",
      "Memory: max=13262m, total=13262m, free=13218m, used=44m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t2.852 seconds.\n",
      "\n",
      "Input:                  \t4571542 reads \t\t228577100 bases.\n",
      "QTrimmed:               \t206114 reads (4.51%) \t491468 bases (0.22%)\n",
      "KTrimmed:               \t643508 reads (14.08%) \t21836788 bases (9.55%)\n",
      "Total Removed:          \t356006 reads (7.79%) \t22328256 bases (9.77%)\n",
      "Result:                 \t4215536 reads (92.21%) \t206248844 bases (90.23%)\n",
      "\n",
      "Time:                         \t2.873 seconds.\n",
      "Reads Processed:       4571k \t1591.45k reads/sec\n",
      "Bases Processed:        228m \t79.57m bases/sec\n",
      "Aligning...\n",
      "May 17 14:57:17 ..... started STAR run\n",
      "May 17 14:57:17 ..... loading genome\n",
      "May 17 14:57:17 ..... started mapping\n",
      "May 17 14:57:50 ..... started sorting BAM\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "May 17 14:57:55 ..... finished successfully\n",
      "SRR6423309\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12647m -Xms12647m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423309.fastq out=SRR6423309_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423309.fastq, out=SRR6423309_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.022 seconds.\n",
      "Initial:\n",
      "Memory: max=13262m, total=13262m, free=13231m, used=31m\n",
      "\n",
      "Added 446 kmers; time: \t0.015 seconds.\n",
      "Memory: max=13262m, total=13262m, free=13216m, used=46m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t2.342 seconds.\n",
      "\n",
      "Input:                  \t3587820 reads \t\t179391000 bases.\n",
      "QTrimmed:               \t164375 reads (4.58%) \t343310 bases (0.19%)\n",
      "KTrimmed:               \t425231 reads (11.85%) \t16771396 bases (9.35%)\n",
      "Total Removed:          \t308313 reads (8.59%) \t17114706 bases (9.54%)\n",
      "Result:                 \t3279507 reads (91.41%) \t162276294 bases (90.46%)\n",
      "\n",
      "Time:                         \t2.362 seconds.\n",
      "Reads Processed:       3587k \t1518.67k reads/sec\n",
      "Bases Processed:        179m \t75.93m bases/sec\n",
      "Aligning...\n",
      "May 17 14:58:04 ..... started STAR run\n",
      "May 17 14:58:04 ..... loading genome\n",
      "May 17 14:58:04 ..... started mapping\n",
      "May 17 14:58:27 ..... started sorting BAM\n",
      "May 17 14:58:32 ..... finished successfully\n",
      "SRR6423310\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12650m -Xms12650m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423310.fastq out=SRR6423310_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423310.fastq, out=SRR6423310_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.023 seconds.\n",
      "Initial:\n",
      "Memory: max=13266m, total=13266m, free=13235m, used=31m\n",
      "\n",
      "Added 446 kmers; time: \t0.015 seconds.\n",
      "Memory: max=13266m, total=13266m, free=13222m, used=44m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t2.679 seconds.\n",
      "\n",
      "Input:                  \t4690443 reads \t\t234522150 bases.\n",
      "QTrimmed:               \t209542 reads (4.47%) \t435820 bases (0.19%)\n",
      "KTrimmed:               \t669957 reads (14.28%) \t28340997 bases (12.08%)\n",
      "Total Removed:          \t536908 reads (11.45%) \t28776817 bases (12.27%)\n",
      "Result:                 \t4153535 reads (88.55%) \t205745333 bases (87.73%)\n",
      "\n",
      "Time:                         \t2.699 seconds.\n",
      "Reads Processed:       4690k \t1738.09k reads/sec\n",
      "Bases Processed:        234m \t86.90m bases/sec\n",
      "Aligning...\n",
      "May 17 14:58:41 ..... started STAR run\n",
      "May 17 14:58:41 ..... loading genome\n",
      "May 17 14:58:42 ..... started mapping\n",
      "May 17 14:59:10 ..... started sorting BAM\n",
      "May 17 14:59:16 ..... finished successfully\n",
      "SRR6423311\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12574m -Xms12574m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423311.fastq out=SRR6423311_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423311.fastq, out=SRR6423311_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.022 seconds.\n",
      "Initial:\n",
      "Memory: max=13186m, total=13186m, free=13157m, used=29m\n",
      "\n",
      "Added 446 kmers; time: \t0.015 seconds.\n",
      "Memory: max=13186m, total=13186m, free=13144m, used=42m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t3.159 seconds.\n",
      "\n",
      "Input:                  \t5035947 reads \t\t251797350 bases.\n",
      "QTrimmed:               \t235834 reads (4.68%) \t482965 bases (0.19%)\n",
      "KTrimmed:               \t520532 reads (10.34%) \t20920503 bases (8.31%)\n",
      "Total Removed:          \t390596 reads (7.76%) \t21403468 bases (8.50%)\n",
      "Result:                 \t4645351 reads (92.24%) \t230393882 bases (91.50%)\n",
      "\n",
      "Time:                         \t3.179 seconds.\n",
      "Reads Processed:       5035k \t1584.23k reads/sec\n",
      "Bases Processed:        251m \t79.21m bases/sec\n",
      "Aligning...\n",
      "May 17 14:59:26 ..... started STAR run\n",
      "May 17 14:59:26 ..... loading genome\n",
      "May 17 14:59:26 ..... started mapping\n",
      "May 17 14:59:56 ..... started sorting BAM\n",
      "May 17 15:00:02 ..... finished successfully\n",
      "SRR6423312\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12670m -Xms12670m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423312.fastq out=SRR6423312_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423312.fastq, out=SRR6423312_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.024 seconds.\n",
      "Initial:\n",
      "Memory: max=13287m, total=13287m, free=13258m, used=29m\n",
      "\n",
      "Added 446 kmers; time: \t0.015 seconds.\n",
      "Memory: max=13287m, total=13287m, free=13245m, used=42m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t2.698 seconds.\n",
      "\n",
      "Input:                  \t3847037 reads \t\t192351850 bases.\n",
      "QTrimmed:               \t183860 reads (4.78%) \t381251 bases (0.20%)\n",
      "KTrimmed:               \t307578 reads (8.00%) \t10775408 bases (5.60%)\n",
      "Total Removed:          \t188319 reads (4.90%) \t11156659 bases (5.80%)\n",
      "Result:                 \t3658718 reads (95.10%) \t181195191 bases (94.20%)\n",
      "\n",
      "Time:                         \t2.717 seconds.\n",
      "Reads Processed:       3847k \t1415.70k reads/sec\n",
      "Bases Processed:        192m \t70.79m bases/sec\n",
      "Aligning...\n",
      "May 17 15:00:10 ..... started STAR run\n",
      "May 17 15:00:10 ..... loading genome\n",
      "May 17 15:00:10 ..... started mapping\n",
      "May 17 15:00:37 ..... started sorting BAM\n",
      "May 17 15:00:42 ..... finished successfully\n",
      "SRR6423313\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12455m -Xms12455m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423313.fastq out=SRR6423313_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423313.fastq, out=SRR6423313_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.023 seconds.\n",
      "Initial:\n",
      "Memory: max=13061m, total=13061m, free=13031m, used=30m\n",
      "\n",
      "Added 446 kmers; time: \t0.015 seconds.\n",
      "Memory: max=13061m, total=13061m, free=13019m, used=42m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t3.182 seconds.\n",
      "\n",
      "Input:                  \t4738414 reads \t\t236920700 bases.\n",
      "QTrimmed:               \t217247 reads (4.58%) \t453389 bases (0.19%)\n",
      "KTrimmed:               \t573535 reads (12.10%) \t23126357 bases (9.76%)\n",
      "Total Removed:          \t430323 reads (9.08%) \t23579746 bases (9.95%)\n",
      "Result:                 \t4308091 reads (90.92%) \t213340954 bases (90.05%)\n",
      "\n",
      "Time:                         \t3.201 seconds.\n",
      "Reads Processed:       4738k \t1480.19k reads/sec\n",
      "Bases Processed:        236m \t74.01m bases/sec\n",
      "Aligning...\n",
      "May 17 15:00:52 ..... started STAR run\n",
      "May 17 15:00:52 ..... loading genome\n",
      "May 17 15:00:52 ..... started mapping\n",
      "May 17 15:01:21 ..... started sorting BAM\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "May 17 15:01:31 ..... finished successfully\n",
      "SRR6423314\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12339m -Xms12339m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423314.fastq out=SRR6423314_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423314.fastq, out=SRR6423314_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.019 seconds.\n",
      "Initial:\n",
      "Memory: max=12939m, total=12939m, free=12910m, used=29m\n",
      "\n",
      "Added 446 kmers; time: \t0.016 seconds.\n",
      "Memory: max=12939m, total=12939m, free=12910m, used=29m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t4.015 seconds.\n",
      "\n",
      "Input:                  \t6222728 reads \t\t311136400 bases.\n",
      "QTrimmed:               \t275554 reads (4.43%) \t649238 bases (0.21%)\n",
      "KTrimmed:               \t1026160 reads (16.49%) \t37455617 bases (12.04%)\n",
      "Total Removed:          \t642821 reads (10.33%) \t38104855 bases (12.25%)\n",
      "Result:                 \t5579907 reads (89.67%) \t273031545 bases (87.75%)\n",
      "\n",
      "Time:                         \t4.036 seconds.\n",
      "Reads Processed:       6222k \t1541.89k reads/sec\n",
      "Bases Processed:        311m \t77.09m bases/sec\n",
      "Aligning...\n",
      "May 17 15:01:44 ..... started STAR run\n",
      "May 17 15:01:44 ..... loading genome\n",
      "May 17 15:01:44 ..... started mapping\n",
      "May 17 15:02:23 ..... started sorting BAM\n",
      "May 17 15:02:31 ..... finished successfully\n",
      "SRR6423315\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12640m -Xms12640m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423315.fastq out=SRR6423315_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423315.fastq, out=SRR6423315_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.023 seconds.\n",
      "Initial:\n",
      "Memory: max=13254m, total=13254m, free=13222m, used=32m\n",
      "\n",
      "Added 446 kmers; time: \t0.016 seconds.\n",
      "Memory: max=13254m, total=13254m, free=13209m, used=45m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t2.455 seconds.\n",
      "\n",
      "Input:                  \t4208805 reads \t\t210440250 bases.\n",
      "QTrimmed:               \t166089 reads (3.95%) \t347627 bases (0.17%)\n",
      "KTrimmed:               \t1011443 reads (24.03%) \t45610189 bases (21.67%)\n",
      "Total Removed:          \t882006 reads (20.96%) \t45957816 bases (21.84%)\n",
      "Result:                 \t3326799 reads (79.04%) \t164482434 bases (78.16%)\n",
      "\n",
      "Time:                         \t2.476 seconds.\n",
      "Reads Processed:       4208k \t1699.98k reads/sec\n",
      "Bases Processed:        210m \t85.00m bases/sec\n",
      "Aligning...\n",
      "May 17 15:02:39 ..... started STAR run\n",
      "May 17 15:02:39 ..... loading genome\n",
      "May 17 15:02:40 ..... started mapping\n",
      "May 17 15:03:04 ..... started sorting BAM\n",
      "May 17 15:03:08 ..... finished successfully\n",
      "SRR6423316\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12379m -Xms12379m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423316.fastq out=SRR6423316_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423316.fastq, out=SRR6423316_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.023 seconds.\n",
      "Initial:\n",
      "Memory: max=12981m, total=12981m, free=12951m, used=30m\n",
      "\n",
      "Added 446 kmers; time: \t0.016 seconds.\n",
      "Memory: max=12981m, total=12981m, free=12939m, used=42m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.004 seconds.\n",
      "Processing time:   \t\t3.539 seconds.\n",
      "\n",
      "Input:                  \t5253477 reads \t\t262673850 bases.\n",
      "QTrimmed:               \t261406 reads (4.98%) \t523685 bases (0.20%)\n",
      "KTrimmed:               \t302856 reads (5.76%) \t10430290 bases (3.97%)\n",
      "Total Removed:          \t192774 reads (3.67%) \t10953975 bases (4.17%)\n",
      "Result:                 \t5060703 reads (96.33%) \t251719875 bases (95.83%)\n",
      "\n",
      "Time:                         \t3.560 seconds.\n",
      "Reads Processed:       5253k \t1475.54k reads/sec\n",
      "Bases Processed:        262m \t73.78m bases/sec\n",
      "Aligning...\n",
      "May 17 15:03:19 ..... started STAR run\n",
      "May 17 15:03:19 ..... loading genome\n",
      "May 17 15:03:19 ..... started mapping\n",
      "May 17 15:03:51 ..... started sorting BAM\n",
      "May 17 15:03:55 ..... finished successfully\n",
      "SRR6423317\n",
      "Trimming...\n",
      "java -Djava.library.path=/home/ubuntu/anaconda3/opt/bbmap-38.46-0/jni/ -ea -Xmx12224m -Xms12224m -cp /home/ubuntu/anaconda3/opt/bbmap-38.46-0/current/ jgi.BBDuk in=SRR6423317.fastq out=SRR6423317_trimmed_clean.fastq ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
      "Executing jgi.BBDuk [in=SRR6423317.fastq, out=SRR6423317_trimmed_clean.fastq, ref=/home/ubuntu/annotations/common/Lexogen_adapters_with_pA.fa.gz, k=13, ktrim=r, useshortkmers=t, mink=5, qtrim=t, trimq=10, minlength=20]\n",
      "Version 38.46\n",
      "\n",
      "maskMiddle was disabled because useShortKmers=true\n",
      "0.021 seconds.\n",
      "Initial:\n",
      "Memory: max=12817m, total=12817m, free=12791m, used=26m\n",
      "\n",
      "Added 446 kmers; time: \t0.015 seconds.\n",
      "Memory: max=12817m, total=12817m, free=12775m, used=42m\n",
      "\n",
      "Input is being processed as unpaired\n",
      "Started output streams:\t0.003 seconds.\n",
      "Processing time:   \t\t3.432 seconds.\n",
      "\n",
      "Input:                  \t5649907 reads \t\t282495350 bases.\n",
      "QTrimmed:               \t245456 reads (4.34%) \t562339 bases (0.20%)\n",
      "KTrimmed:               \t885709 reads (15.68%) \t34114517 bases (12.08%)\n",
      "Total Removed:          \t604370 reads (10.70%) \t34676856 bases (12.28%)\n",
      "Result:                 \t5045537 reads (89.30%) \t247818494 bases (87.72%)\n",
      "\n",
      "Time:                         \t3.451 seconds.\n",
      "Reads Processed:       5649k \t1637.28k reads/sec\n",
      "Bases Processed:        282m \t81.86m bases/sec\n",
      "Aligning...\n",
      "May 17 15:04:07 ..... started STAR run\n",
      "May 17 15:04:07 ..... loading genome\n",
      "May 17 15:04:07 ..... started mapping\n",
      "May 17 15:04:42 ..... started sorting BAM\n",
      "May 17 15:04:48 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "mkdir -p ~/output\n",
    "cd ~/data\n",
    "\n",
    "#Loop for all fastq files\n",
    "for FILE in `ls *fastq.gz`\n",
    "do\n",
    "\n",
    "#Get sample name\n",
    "sample=${FILE##*/}\n",
    "sample=${sample%.fastq.gz}\n",
    "echo $sample\n",
    "\n",
    "#Unpack the fastq.gz files\n",
    "gzip -d ${sample}.fastq.gz\n",
    "\n",
    "#Trim adapters with BBMAP\n",
    "echo \"Trimming...\"\n",
    "\n",
    "~/anaconda3/bin/bbduk.sh in=${sample}.fastq out=${sample}_trimmed_clean.fastq ref=~/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20\n",
    "\n",
    "#Align with STAR\n",
    "echo \"Aligning...\"\n",
    "\n",
    "mkdir -p ~/output/STAR_map/${sample}_trimmed_clean\n",
    "\n",
    "STAR \\\n",
    "--runThreadN 4 \\\n",
    "--genomeDir ~/annotations/sacCer3_Sp_merged_genome/STAR_index \\\n",
    "--readFilesIn ${sample}_trimmed_clean.fastq \\\n",
    "--outFilterType BySJout \\\n",
    "--outFilterMultimapNmax 20 \\\n",
    "--alignSJoverhangMin 8 \\\n",
    "--alignSJDBoverhangMin 1 \\\n",
    "--outFilterMismatchNmax 999 \\\n",
    "--alignIntronMin 20 \\\n",
    "--alignIntronMax 2000 \\\n",
    "--outFileNamePrefix ~/output/STAR_map/${sample}_trimmed_clean \\\n",
    "--outSAMtype BAM SortedByCoordinate\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Index the bams\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/output/STAR_map\n",
    "for f in *.bam; do samtools index $f; done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Run QC for alignment results ...doesn't make good output...\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/output/STAR_map\n",
    "\n",
    "#put the final STAR map reports into a single file \n",
    "for f in *_trimmed_cleanLog.final.out; do echo $f >> STAR_final.outs; cat $f >> STAR_final.outs ;done \n",
    "#cut down to the relevant rows\n",
    "cat STAR_final.outs | tr \"\\n\" \"\\t\" | sed s/\"%\\t390\"/\"%\\n390\"/g | cut -f1,11- > STAR_final.outs_concise"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "Make Lexogen_bam_to_bedgraph.py file\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/software/\n",
    "\n",
    "cat >Lexogen_bam_to_bedgraph.py <<EOL\n",
    "#!/usr/bin/env python\n",
    "\n",
    "'''\n",
    "usage:\n",
    "      \tpython Lexogen_bam_to_bedgraph.py bam_file\n",
    "For now just trying to get reads with NH tag = 1, obtain end position of each read, count them up and write directly to bedgraph file\n",
    "'''\n",
    "\n",
    "__author__ = 'schmidm'\n",
    "import sys\n",
    "import HTSeq\n",
    "import numpy as np\n",
    "if len(sys.argv) <= 1 or sys.argv[1] == '-h':\n",
    "    print __doc__\n",
    "    exit()\n",
    "#open a bam file\n",
    "bamfile = sys.argv[1]\n",
    "bam_reader = HTSeq.BAM_Reader( bamfile )\n",
    "plus_bg_fname = bamfile.replace('.bam', '_plus.bedgraph')\n",
    "minus_bg_fname = bamfile.replace('.bam', '_minus.bedgraph')\n",
    "#get header for chromosome information\n",
    "chromosomes = {chunk['SN']: chunk['LN'] for chunk in bam_reader.get_header_dict()['SQ']}\n",
    "print 'chromosomes from bam: ', chromosomes\n",
    "#loop through the reads\n",
    "i = 0\n",
    "NH_included = 0\n",
    "NH_excluded = 0\n",
    "cur_chr = ''\n",
    "plus_sum = 0\n",
    "minus_sum = 0\n",
    "def write_chr_values(bg_fname, chr, values):\n",
    "    with open(bg_fname, 'a+') as f:\n",
    "        start = 0\n",
    "        end = 1\n",
    "        value = 0\n",
    "        for i,v in enumerate(values):            \n",
    "            if i == end and v == value:\n",
    "                #concatenate adjacent bins with identical value into larger interval\n",
    "                end = i+1\n",
    "            else:\n",
    "                if value > 0:\n",
    "                     f.write(chr + '\\t' + str(start) + '\\t' + str(end) + '\\t' + str(value) + '\\n')\n",
    "                start = i\n",
    "                end = i+1\n",
    "                value = v\n",
    "        if value > 0:\n",
    "            f.write(chr + '\\t' + str(start) + '\\t' + str(end) + '\\t' + str(value) + '\\n')\n",
    "for aln in bam_reader:\n",
    "    if aln.optional_field('NH') != 1:\n",
    "        NH_excluded += 1\n",
    "        continue\n",
    "    NH_included += 1\n",
    "    iv = aln.iv\n",
    "    if iv.chrom != cur_chr:\n",
    "        #write out\n",
    "        if cur_chr != '':\n",
    "            write_chr_values(plus_bg_fname, cur_chr, plus_vals)\n",
    "            write_chr_values(minus_bg_fname, cur_chr, minus_vals)\n",
    "            plus_sum += sum(plus_vals)\n",
    "            minus_sum += sum(minus_vals)\n",
    "        #initialize next chr\n",
    "        cur_chr = iv.chrom\n",
    "        print 'processing reads for chr: ', cur_chr\n",
    "        plus_vals = np.zeros(chromosomes[cur_chr], dtype = np.int)\n",
    "        minus_vals = np.zeros(chromosomes[cur_chr], dtype = np.int)\n",
    "    if iv.strand == '+':\n",
    "        minus_vals[iv.start] += 1\n",
    "    elif iv.strand == '-':\n",
    "        plus_vals[iv.end - 1] += 1\n",
    "    else:\n",
    "        print 'missing strand info ...?'\n",
    "write_chr_values(plus_bg_fname, cur_chr, plus_vals)\n",
    "write_chr_values(minus_bg_fname, cur_chr, minus_vals)\n",
    "plus_sum += sum(plus_vals)\n",
    "minus_sum += sum(minus_vals)\n",
    "## some output that helped in some initial debugging\n",
    "print 'NH_excluded: ', NH_excluded\n",
    "print 'NH_included: ', NH_included\n",
    "print 'plus_sum: ', plus_sum\n",
    "print 'minus_sum: ', minus_sum\n",
    "EOL"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Run bam_to_bedgraph script on bam files\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(py27) (py27) (py27) SRR6423303_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  3407322\n",
      "NH_included:  3718645\n",
      "plus_sum:  1981402\n",
      "minus_sum:  1737243\n",
      "done\n",
      "SRR6423305_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  4486473\n",
      "NH_included:  3232905\n",
      "plus_sum:  1745167\n",
      "minus_sum:  1487738\n",
      "done\n",
      "SRR6423306_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  4486435\n",
      "NH_included:  2609529\n",
      "plus_sum:  1444607\n",
      "minus_sum:  1164922\n",
      "done\n",
      "SRR6423307_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  9433521\n",
      "NH_included:  3176830\n",
      "plus_sum:  1850902\n",
      "minus_sum:  1325928\n",
      "done\n",
      "SRR6423308_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  7675477\n",
      "NH_included:  2707460\n",
      "plus_sum:  1565056\n",
      "minus_sum:  1142404\n",
      "done\n",
      "SRR6423309_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  4505791\n",
      "NH_included:  2248911\n",
      "plus_sum:  1252268\n",
      "minus_sum:  996643\n",
      "done\n",
      "SRR6423310_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  4098075\n",
      "NH_included:  3006928\n",
      "plus_sum:  1617165\n",
      "minus_sum:  1389763\n",
      "done\n",
      "SRR6423311_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  4842811\n",
      "NH_included:  3331266\n",
      "plus_sum:  1769290\n",
      "minus_sum:  1561976\n",
      "done\n",
      "SRR6423312_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  3944078\n",
      "NH_included:  2591388\n",
      "plus_sum:  1394010\n",
      "minus_sum:  1197378\n",
      "done\n",
      "SRR6423313_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  4322146\n",
      "NH_included:  3219049\n",
      "plus_sum:  1759900\n",
      "minus_sum:  1459149\n",
      "done\n",
      "SRR6423314_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  7826906\n",
      "NH_included:  3881460\n",
      "plus_sum:  2149864\n",
      "minus_sum:  1731596\n",
      "done\n",
      "SRR6423315_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  4093009\n",
      "NH_included:  2401615\n",
      "plus_sum:  1337321\n",
      "minus_sum:  1064294\n",
      "done\n",
      "SRR6423316_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  1106540\n",
      "NH_included:  4312922\n",
      "plus_sum:  2230429\n",
      "minus_sum:  2082493\n",
      "done\n",
      "SRR6423317_trimmed_cleanAligned.sortedByCoord.out.bam\n",
      "chromosomes from bam:  {'chrV': 576874, 'chrII': 813184, 'chrVI': 270161, 'chrX': 745751, 'chrIX': 439888, 'chrM': 85779, 'chrIV': 1531933, 'chrI': 230218, 'chrVIII': 562643, 'chrXIV': 784333, 'chrXVI': 948066, 'chrXI': 666816, 'chrXIII': 924431, 'I': 5579133, 'chrXV': 1091291, 'chrIII': 316620, 'chrXII': 1078177, 'III': 2452883, 'AB325691': 20000, 'MTR': 20128, 'II': 4539804, 'chrVII': 1090940, 'MT': 19431}\n",
      "processing reads for chr:  chrI\n",
      "processing reads for chr:  chrII\n",
      "processing reads for chr:  chrIII\n",
      "processing reads for chr:  chrIV\n",
      "processing reads for chr:  chrIX\n",
      "processing reads for chr:  chrM\n",
      "processing reads for chr:  chrV\n",
      "processing reads for chr:  chrVI\n",
      "processing reads for chr:  chrVII\n",
      "processing reads for chr:  chrVIII\n",
      "processing reads for chr:  chrX\n",
      "processing reads for chr:  chrXI\n",
      "processing reads for chr:  chrXII\n",
      "processing reads for chr:  chrXIII\n",
      "processing reads for chr:  chrXIV\n",
      "processing reads for chr:  chrXV\n",
      "processing reads for chr:  chrXVI\n",
      "processing reads for chr:  I\n",
      "processing reads for chr:  II\n",
      "processing reads for chr:  III\n",
      "processing reads for chr:  MT\n",
      "processing reads for chr:  MTR\n",
      "processing reads for chr:  AB325691\n",
      "NH_excluded:  5542676\n",
      "NH_included:  4014976\n",
      "plus_sum:  2277176\n",
      "minus_sum:  1737800\n",
      "done\n",
      "(py27) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd ~/output/STAR_map\n",
    "conda activate py27\n",
    "\n",
    "for sample in `ls *bam`\n",
    "do\n",
    "\n",
    "echo $sample\n",
    "\n",
    "python ~/software/Lexogen_bam_to_bedgraph.py $sample\n",
    "echo \"done\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
