---
title: "Genomic A-rich masking"
author: "Manfred Schmid"
date: "9 August 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# Mask reads primed at genomic A-stretches

Next step is to mask genomic pA reads from the genome. This could possible be included in the above step. But the approach I used is to create a “masking file”. Ie a bed file of all genomic positions to be filtered. (This should allow for a more flexible approach for comparing different filters.) The main script to create the masking file is again implemented in python, which takes various parameters (how many nucleotides X in a window of size Y) to control what is being masked. The criteria are taken from Kevin Roy, Chanfreau lab. Which he assembled based on yeast Lexogen data. They are:  
1)	all regions with >= 4A within 6 nucleotides but no C or Ts (downstream of position).  
2)	all regions with >= 12A within 18 nucleotides (no CT restrictions, downstream of position).  
3)	All regions with >=15A within 18 nucleotides (upstream of position).  
I would say number 3 is optional, and the technical reason for this filtering is somewhat obscure. Apparently QuantSeq protocol can produce artifact reads containing polyT stretches, and these will map to A-rich sequences in the genome, which can be filtered away using criterion 3. On our first run this was not obvious and I did not include that.


## creation of the genomic A-mask bed file

#### flag_genomic_As_KevinRoy_like.py
```{python, eval=FALSE}

#!/usr/bin/env python
'''

Scans the genome for genomic A stretches and prints them in bed format.

Usage:

    flag_genomic_As_KevinRoy_like.py -i genome.fa -l [word_len] -A [min As within word] -maxCT [max C+Ts within word] -el [extend output left] -er [extend output right] -o [output_file]

-i ... genome in fasta format
-l ... length of motif, ie word length (default = 6)
-A ... minimum number of A for word to be flagged (default = 4)
-maxCT ... maximum number of C and Ts for word to be flagged (default = 2)
-el ... extend each single nucleotide hit by this number upstream (default = 0)
-er ... extend each single nucleotide hit by this number downstream (default = 0)
-o ... output file

Intended for combination as input for bedtools subtract of A-tail based 3p end sequencing.

'''


__author__ = 'schmidm'

import sys
import argparse


def parse_args():
    parser = argparse.ArgumentParser()
    parser.add_argument('fname', action='store', help="input genome file")
    parser.add_argument("-l", "--word_len", action='store', help="word length", type=int, default=5)
    parser.add_argument("-A", "--A", action='store', help="minimum As within word", type=int, default=0)
    parser.add_argument("-maxCT", "--maxCT", action='store', help="maximum C plus T within word", type=int, default=0)
    parser.add_argument("-el", "--extend_left", action='store', default=0, help="extend output hits by this upstream", type=int)
    parser.add_argument("-er", "--extend_right", action='store', default=0, help="extend output hits by this downstream", type=int)
    parser.add_argument("-o", "--output_file", action='store', default="flagged.bed", help="output bed file")
    args=parser.parse_args()
    return args

class Sequence:
    def __init__(self, name, seq):
        self.name = name
        self.seq = seq

def read_fasta(fname):
    '''
    :param fname: fasta file
    :return: Generator of Sequence objects for the individual sequences in the file
    '''
    try:
        with open(fname, 'r') as f:
            name = ''
            seq = ''
            for line in f:
                if line[0] == '>':
                    if len(seq) > 0:
                        yield Sequence(name, seq)
                    name = line.lstrip('>').rstrip()
                    seq = ''
                else:
                    seq += line.strip()
            yield Sequence(name, seq)
    except Exception as e:
        print 'ERROR: genome fasta file ', fname, 'not found!'

    return


if __name__ == '__main__':

    args = parse_args()

    flagged_regions = []
    for seq in read_fasta(args.fname):
        print 'scanning: ', seq.name
        start = 0
        end = start + args.word_len
        cnts = {'A':0, 'C':0, 'G':0, 'T':0, 'N':0}
        cnts['A'] = sum(1 for c in seq.seq[start:end] if c == 'A' )
        cnts['C'] = sum(1 for c in seq.seq[start:end] if c == 'C' )
        cnts['G'] = sum(1 for c in seq.seq[start:end] if c == 'G' )
        cnts['T'] = sum(1 for c in seq.seq[start:end] if c == 'T' )

        while end < len(seq.seq):
            #print 'A', cnts, cnts['C'] + cnts['T']
            if cnts['A'] >= args.A and (cnts['C'] + cnts['T']) <= args.maxCT:
                flagged_regions.append( (seq.name, start-1-args.extend_left, start+args.extend_right, seq.seq[start:end], '.', '+' ))
                #print 'A', cnts, cnts['C'] + cnts['T']

            if cnts['T'] >= args.A and (cnts['G'] + cnts['A']) <= args.maxCT:
                flagged_regions.append( (seq.name, end-args.extend_right, end+1+args.extend_left, seq.seq[start:end], '.', '-' ))
                #print 'T', cnts

            cnts[seq.seq[start]] -= 1
            cnts[seq.seq[end]] += 1

            start += 1
            end += 1

    print 'scanning done, found ', len(flagged_regions), ' hits'


    with open(args.output_file, 'w') as f:
        for r in flagged_regions:
            f.write('\t'.join(str(x) for x in r))
            f.write('\n')
```



again this is called via a shell script:  
```{bash, eval=FALSE}
#!usr/bin/bash

cd /Users/schmidm/ms_tools/MS_Metagene_Tools/


##get A6 regions
python flag_genomic_As_KevinRoy_like.py /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged.fa -l 6 -A 4 -maxCT 0 -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6.bed

awk '{if($6=="+") print $0}' /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6.bed |
sort -k1,1 -k2,2n -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6_plus.bed
bedtools merge -d 2 -i /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6_plus.bed > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6_plus_merged.bed

awk '{if($6=="-") print $0}' /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6.bed |
sort -k1,1 -k2,2n -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6_minus.bed
bedtools merge -d 2 -i /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6_minus.bed > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6_minus_merged.bed

##get A18 regions
python flag_genomic_As_KevinRoy_like.py /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged.fa -l 18 -A 12 -maxCT 6 -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18.bed

awk '{if($6=="+") print $0}' /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18.bed |
sort -k1,1 -k2,2n -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18_plus.bed
bedtools merge -d 2 -i /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18_plus.bed > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18_plus_merged.bed

awk '{if($6=="-") print $0}' /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18.bed |
sort -k1,1 -k2,2n -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18_minus.bed
bedtools merge -d 2 -i /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18_minus.bed > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18_minus_merged.bed


##get A15 out of A18 regions
python flag_genomic_As_KevinRoy_like.py /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged.fa -l 18 -A 15 -maxCT 3 -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18.bed

#these are for upstream filtering ... need to shift
awk '{OFS="\t"}{if($6=="+"){$2+=19; $3+=19; print $0}}' /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18.bed | \
sort -k1,1 -k2,2n -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18_plus.bed
bedtools merge -d 2 -i /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18_plus.bed > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18_plus_merged.bed

awk '{OFS="\t"}{if($6=="-"){ $2-=19; $3-=19; print $0}}' /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18.bed | \
sort -k1,1 -k2,2n -o /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18_minus.bed
bedtools merge -d 2 -i /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18_minus.bed > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A15in18_minus_merged.bed


##browser tracking indicates no reads stemming around upstream A-tracks ... do not apply this filter for now



##merge the above
cat /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18_plus_merged.bed \
/Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6_plus_merged.bed | \
sort -k1,1 -k2,2n | \
bedtools merge -i stdin > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_plus.bed

cat /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A12in18_minus_merged.bed \
/Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_Sp_merged_flagged_A4in6_minus_merged.bed | \
sort -k1,1 -k2,2n | \
bedtools merge -i stdin > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_minus.bed


## single file
awk '{if(NR == FNR){strand="+"}else{strand="-"};print $0"\t.\t0\t"strand}' /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_plus.bed /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_minus.bed | sort -k1,1 -k2,2n > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy.bed

##single mask S.cerevisiae only
grep ^chr /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy.bed > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/genomicAmask.bed

## count positions
awk '{sum+=($3-$2)}END{print sum}' /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/genomicAmask.bed
# 655732

## genome size
head /Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_chrom.sizes
#chrM	85779
#chrI	230218
#chrII	813184
#chrIII	316620
#chrIV	1531933
#chrIX	439888
#chrV	576874

awk '{sum+=$2}END{print sum*2}' /Users/schmidm/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_chrom.sizes
# 24314210 (total nt in both strands!!)


## 655732/24314210 = 0.02696909 --> 2.7% of genome are masked
```



## filtering genomic A-masked positions from bedgraphs

#### 10min data

```{bash, eval=FALSE}
#!usr/bin/bash

##subtract from bedgraph; only do S.cerevisiae for this part for simplicity
grep ^chr /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_plus.bed | \
 sort -k1,1 -k2,2n > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_Sc_plus.bed

grep ^chr /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_minus.bed |  sort -k1,1 -k2,2n > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_Sc_minus.bed


mkdir /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/KevinRoyAfiltered_bedgraph

for f in /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph/*_plus.bedgraph
do
  echo $f
  grep ^chr $f | \
  sort -k1,1 -k2,2n | \
  bedtools subtract -a - \
  -b /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_Sc_plus.bed > \
  ${f/.bedgraph/_KevinRoyAfiltered.bedgraph}
done

for f in /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph/*_minus.bedgraph
do
  echo $f
  grep ^chr $f | \
  sort -k1,1 -k2,2n | \
  bedtools subtract -a - \
  -b /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_Sc_minus.bed > \
  ${f/.bedgraph/_KevinRoyAfiltered.bedgraph}
done

mv /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph/*KevinRoyAfiltered.bedgraph \
/Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/KevinRoyAfiltered_bedgraph/.
```



#### 2min data

```{bash, eval=FALSE}
#!usr/bin/bash

##subtract from bedgraph; only do S.cerevisiae for this part for simplicity
grep ^chr /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_plus.bed | \
 sort -k1,1 -k2,2n > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_Sc_plus.bed

grep ^chr /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_minus.bed |  sort -k1,1 -k2,2n > /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_Sc_minus.bed


mkdir /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph

for f in /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/raw_bedgraph/*_plus.bedgraph
do
  echo $f
  grep ^chr $f | \
  sort -k1,1 -k2,2n | \
  bedtools subtract -a - \
  -b /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_Sc_plus.bed > \
  ${f/.bedgraph/_KevinRoyAfiltered.bedgraph}
done

for f in /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/raw_bedgraph/*_minus.bedgraph
do
  echo $f
  grep ^chr $f | \
  sort -k1,1 -k2,2n | \
  bedtools subtract -a - \
  -b /Users/schmidm/Documents/Results/Lexogen_RNAseq/bams/sacCer3_flagged_KevinRoy_Sc_minus.bed > \
  ${f/.bedgraph/_KevinRoyAfiltered.bedgraph}
done

mv /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph/*KevinRoyAfiltered.bedgraph \
/Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/KevinRoyAfiltered_bedgraph/.
```


## stats for reads in S. cerevisiae and S.pombe and A-filtered...

#### 10min data

```{bash, eval = FALSE}
### amount of reads in S.cerevisiae and S.pombe raw files
raw_count_file="/Users/schmidm/Documents/Results/Lexogen_RNAseq/trim_and_map_info/Sc_Sp_raw_bedgraph.counts"

cd /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph

echo "bedgraph Sc Sp" > ${raw_count_file}

for b in *_plus.bedgraph
do
  echo "counting ${b}"

  cat $b ${b/_plus.bedgraph/_minus.bedgraph} | \
  awk -v fname="${b/_plus.bedgraph/}" '{if($1 ~ /^chr/){Sc_sum+=$4*($3-$2)}else{Sp_sum+=$4*($3-$2)}}END{print fname" "Sc_sum" "Sp_sum}' >> ${raw_count_file}

done


cat $raw_count_file
#bedgraph Sc Sp
#Mex_in_0_1 5942399 137163
#Mex_in_0_2 4707318 112692
#Mex_in_0_3 10113976 235737
#Mex_in_15_1 5541109 176583
#Mex_in_15_2 4402660 81556
#Mex_in_15_3 8171197 273885
#Mex_in_70_1 4302130 132222
#Mex_in_70_2 4006037 111759
#Mex_in_70_3 3993881 135561
#Mex_in_neg 5261497 72637
#Mex_ip_0_1 7150452 165124
#Mex_ip_0_2 7054968 196736
#Mex_ip_0_3 5173699 138931
#Mex_ip_15_1 4930616 508465
#Mex_ip_15_2 4323077 256616
#Mex_ip_15_3 3742082 431698
#Mex_ip_70_1 6366109 558609
#Mex_ip_70_2 3934838 344116
#Mex_ip_70_3 4029790 419957
#Mex_ip_neg 5603811 851599
#Nab_in_0_1 5396295 106881
#Nab_in_0_2 5482125 147565
#Nab_in_0_3 6450346 184617
#Nab_in_15_1 2357069 71017
#Nab_in_15_2 5483223 133231
#Nab_in_15_3 6574692 197926
#Nab_in_70_1 2997165 84911
#Nab_in_70_2 4714484 194848
#Nab_in_70_3 5565636 240363
#Nab_in_neg 4077028 109698
#Nab_ip_0_1 5876655 192255
#Nab_ip_0_2 4597870 193963
#Nab_ip_0_3 5694365 284033
#Nab_ip_15_1 3924799 390638
#Nab_ip_15_2 3665898 260786
#Nab_ip_15_3 4191233 411475
#Nab_ip_70_1 3248139 288464
#Nab_ip_70_2 4751696 602516
#Nab_ip_70_3 3904713 407398
#Nab_ip_neg 5521766 1494515
#noPap_in_0 1778588 34996
#noPap_in_70 1994363 48467
#noPap_ip_0 3086874 230384
#noPap_ip_70 2698267 262979
#xPap_in_0 5704295 59633
#xPap_in_70 6447377 65637
#xPap_ip_0 6075478 236730
#xPap_ip_70 7340559 276256


### counts after A-filtering
filt_count_file="/Users/schmidm/Documents/Results/Lexogen_RNAseq/trim_and_map_info/Sc_Afiltered_bedgraph.counts"
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/KevinRoyAfiltered_bedgraph

echo "bedgraph Sc Sp" > ${filt_count_file}

for b in *_plus_KevinRoyAfiltered.bedgraph
do
  echo "counting ${b}"

  cat $b ${b/_plus/_minus} | \
  awk -v fname="${b/_plus_KevinRoyAfiltered.bedgraph/}" '{sum+=$4*($3-$2)}END{print fname" "sum}' >> ${filt_count_file}

done

cat $filt_count_file
#bedgraph Sc Sp
#Mex_in_0_1 5210550
#Mex_in_0_2 4141477
#Mex_in_0_3 9052288
#Mex_in_15_1 4851797
#Mex_in_15_2 3843771
#Mex_in_15_3 7285130
#Mex_in_70_1 3642981
#Mex_in_70_2 3439136
#Mex_in_70_3 3501942
#Mex_in_neg 4687659
#Mex_ip_0_1 6143287
#Mex_ip_0_2 6060297
#Mex_ip_0_3 4553450
#Mex_ip_15_1 3982991
#Mex_ip_15_2 3535866
#Mex_ip_15_3 3114382
#Mex_ip_70_1 4668975
#Mex_ip_70_2 3159292
#Mex_ip_70_3 3187034
#Mex_ip_neg 4910928
#Nab_in_0_1 4763184
#Nab_in_0_2 4779771
#Nab_in_0_3 5636310
#Nab_in_15_1 2076661
#Nab_in_15_2 4762969
#Nab_in_15_3 5763307
#Nab_in_70_1 2617732
#Nab_in_70_2 4112311
#Nab_in_70_3 4918285
#Nab_in_neg 3607338
#Nab_ip_0_1 5137191
#Nab_ip_0_2 3982006
#Nab_ip_0_3 4916991
#Nab_ip_15_1 3346347
#Nab_ip_15_2 3105436
#Nab_ip_15_3 3565773
#Nab_ip_70_1 2763834
#Nab_ip_70_2 4014559
#Nab_ip_70_3 3298796
#Nab_ip_neg 4786756
#noPap_in_0 1554131
#noPap_in_70 1724969
#noPap_ip_0 2600270
#noPap_ip_70 2259523
#xPap_in_0 5513515
#xPap_in_70 6271098
#xPap_ip_0 5815192
#xPap_ip_70 6949865
```


#### 2min data

```{bash, eval = FALSE}
### amount of reads in S.cerevisiae and S.pombe raw files
raw_count_file="/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/trim_and_map_info/Sc_Sp_raw_bedgraph.counts"

cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/raw_bedgraph

echo "bedgraph Sc Sp" > ${raw_count_file}

for b in *_plus.bedgraph
do
  echo "counting ${b}"

  cat $b ${b/_plus.bedgraph/_minus.bedgraph} | \
  awk -v fname="${b/_plus.bedgraph/}" '{if($1 ~ /^chr/){Sc_sum+=$4*($3-$2)}else{Sp_sum+=$4*($3-$2)}}END{print fname" "Sc_sum" "Sp_sum}' >> ${raw_count_file}

done


cat $raw_count_file
#bedgraph Sc Sp
#noPap_Mex67AA_input_0_1 2467734 41709
#noPap_Mex67AA_input_15_1 3352405 74880
#noPap_Mex67AA_input_70_1 2735663 68571
#noPap_Mex67AA_ip_0_1 2949235 456822
#noPap_Mex67AA_ip_15_1 2110422 519671
#noPap_Mex67AA_ip_70_1 1249021 399728
#noPap_Mex67AA_ip_neg0_1 3517377 795545
#noPap_Nab2AA_input_0_1 2595050 42095
#noPap_Nab2AA_input_15_1 3634113 74302
#noPap_Nab2AA_ip_0_1 2015728 377298
#noPap_Nab2AA_ip_15_1 2195539 591965
#xPap_Mex67AA_input_0_1 3781260 35565
#xPap_Mex67AA_input_15_1 5519178 59857
#xPap_Mex67AA_input_70_1 3935436 41347
#xPap_Mex67AA_ip_0_1 2855569 151361
#xPap_Mex67AA_ip_0_2 3156133 175134
#xPap_Mex67AA_ip_0_3 2459241 132147
#xPap_Mex67AA_ip_15_1 3044177 174875
#xPap_Mex67AA_ip_15_2 3668144 213321
#xPap_Mex67AA_ip_15_3 2260905 140711
#xPap_Mex67AA_ip_70_1 4690852 379459
#xPap_Mex67AA_ip_70_2 3143608 267832
#xPap_Mex67AA_ip_70_3 3029220 247148
#xPap_Mex67AA_ip_neg0_1 3579615 435360
#xPap_Nab2AA_input_0_1 3944905 40380
#xPap_Nab2AA_input_15_1 3626890 35202
#xPap_Nab2AA_ip_0_1 2057164 434311
#xPap_Nab2AA_ip_0_2 3040507 192401
#xPap_Nab2AA_ip_0_3 2446311 163220
#xPap_Nab2AA_ip_15_1 2196866 138905
#xPap_Nab2AA_ip_15_2 2547243 160215
#xPap_Nab2AA_ip_15_3 2104394 144518


### counts after A-filtering
filt_count_file="/Users/schmidm/Documents/Results/Lexogen_RNAseq_2/trim_and_map_info/Sc_Afiltered_bedgraph.counts"
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/KevinRoyAfiltered_bedgraph

echo "bedgraph Sc Sp" > ${filt_count_file}

for b in *_plus_KevinRoyAfiltered.bedgraph
do
  echo "counting ${b}"

  cat $b ${b/_plus/_minus} | \
  awk -v fname="${b/_plus_KevinRoyAfiltered.bedgraph/}" '{sum+=$4*($3-$2)}END{print fname" "sum}' >> ${filt_count_file}

done

cat $filt_count_file
#bedgraph Sc Sp
#noPap_Mex67AA_input_0_1 2202503
#noPap_Mex67AA_input_15_1 2963985
#noPap_Mex67AA_input_70_1 2387948
#noPap_Mex67AA_ip_0_1 2579454
#noPap_Mex67AA_ip_15_1 1837564
#noPap_Mex67AA_ip_70_1 1054532
#noPap_Mex67AA_ip_neg0_1 3116325
#noPap_Nab2AA_input_0_1 2313940
#noPap_Nab2AA_input_15_1 3235115
#noPap_Nab2AA_ip_0_1 1769572
#noPap_Nab2AA_ip_15_1 1905171
#xPap_Mex67AA_input_0_1 3654903
#xPap_Mex67AA_input_15_1 5390170
#xPap_Mex67AA_input_70_1 3825320
#xPap_Mex67AA_ip_0_1 2648241
#xPap_Mex67AA_ip_0_2 3022976
#xPap_Mex67AA_ip_0_3 2310471
#xPap_Mex67AA_ip_15_1 2904716
#xPap_Mex67AA_ip_15_2 3379732
#xPap_Mex67AA_ip_15_3 2072147
#xPap_Mex67AA_ip_70_1 4265182
#xPap_Mex67AA_ip_70_2 2950139
#xPap_Mex67AA_ip_70_3 2784526
#xPap_Mex67AA_ip_neg0_1 3153307
#xPap_Nab2AA_input_0_1 3822077
#xPap_Nab2AA_input_15_1 3530828
#xPap_Nab2AA_ip_0_1 1888025
#xPap_Nab2AA_ip_0_2 2815418
#xPap_Nab2AA_ip_0_3 2214857
#xPap_Nab2AA_ip_15_1 1901498
#xPap_Nab2AA_ip_15_2 2208254
#xPap_Nab2AA_ip_15_3 1902227
```
