---
title: 'Nucleosomes'
author: "Manfred Schmid"
output: pdf_document
editor_options: 
  chunk_output_type: console
---

`r format(Sys.time(), "%d %B, %Y")`


```{r, echo=TRUE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, 
                      fig.path=paste0('../../Figures/Figures_plus1_nucleosome/'), 
                      dev='pdf', echo=TRUE, warning=FALSE, message=FALSE, 
                      error=TRUE)
```


```{r load packages, echo=T, warning=F, message=F}
suppressWarnings(library(tidyverse))
suppressWarnings(library(magrittr))
suppressWarnings(library(knitr))
suppressWarnings(library(RMetaTools))
suppressWarnings(library(broom))
```


## nucleosome positions

Extract from Supplemental Additonal Data 1 Excel file from Jiang and Pugh 2009 publication. This publication uses nucleosomes info from several publication and makes a common list assembled in this sheet. Saved the first sheet "Measured nucleosomes" as *Pugh_measured_nucleosomes.txt*.

In the table nucleosomes marked with, + and - are the number of nucleosome according to a specific annotation (Steinmetz includes ORFs, SUTs and CUTs); * in position are terminal nucleosomes, all other have no info in that column.


```{bash, eval = FALSE}
# Genome release these coordinates refer to are not provided, but given its late 2009
## this should be close to sacCer2 (released mid-2008) ...


## convert to bed file

cd ~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/
cat Pugh_measured_nucleosomes.txt | tr "\r" "\n" | cut -f 1-20 | head
#Name    chrom   start   end     position        peakheight      nc      occ     fuzziness       Set1    rc1     occ1    Set2    rc2     occ2    Set3    rc3     occ3    Set4    rc4
#N1:192  chr01   119     265             1,11    3       80,5    41      151     10      89                              192     109     72
#N1:382  chr01   309     455             3,61    5       71      21,916  376     260     100     427     7       100     372     50      13
#N1:630  chr01   557     703             4,31    5       62      10,756  622     359     100     647     2       79      636     44      7
#N1:872  chr01   799     945             3,81    5       65      15,643                          864     11      100     845     65      28      880     5
#N1:1033 chr01   960     1106            3,44    4       52,5    11,325  1022    74      100                             1034    42      5
#N1:1191 chr01   1118    1264            2,99    3       100     1,732   1190    17      100
#N1:1340 chr01   1267    1413            2,05    3       50      24,007  1340    74      100                             1380    16      0
#N1:1585 chr01   1512    1658    "+5*:YAL068C; " 3,83    4       50      5,916   1594    67      100                             1582    22      0
#N1:1902 chr01   1829    1975    "+3:YAL068C; "  2,67    4       50      35,968  1893    325     100                             1972    8       0

cat Pugh_measured_nucleosomes.txt | tr "\r" "\n" | cut -f 2 | uniq -c

cat Pugh_measured_nucleosomes.txt | tr "\r" "\n" | cut -f 2-5 | sed s/"; "/";"/g | head

cat Pugh_measured_nucleosomes.txt | tr "\r" "\n" | cut -f 2-6 | sed 1d | sed s/"; "/";"/g | awk '{OFS="\t"}{if(!($4 ~ /^"/ )){$4="ND"};print $0}' | cut -f 1-4 > Pugh_measured_nucleosomes_roman0.bed
#I strongly assume that the coordinates are NOT 0-based (its called start position and end position), so I convert to bed 0-based half-open...



## liftOVer to sacCer3
### for liftOver the closest chain file I can get is:
### V62_2009_02_18_V64_2011_02_03.over.chain

cat V62_2009_02_18_V64_2011_02_03.over.chain | sed 's/_...._.._.. / /g' > V62_2009_02_18_V64_2011_02_03_plain.over.chain

~/Documents/genomewide_datasets/scripts/liftOver_and_conversionTools/liftOver Pugh_measured_nucleosomes_roman0.bed V62_2009_02_18_V64_2011_02_03_plain.over.chain Pugh_measured_nucleosomes_roman0_sacCer3.bed conversion_failed.bed

cat conversion_failed.bed
##Partially deleted in new
#chr01   171878  172025  "+10:CUT004;-1:YAR018C;"
##Partially deleted in new
#chr01   174060  174207  "+7:YAR019C;"
##Partially deleted in new
#chr01   230082  230229  ND
##Partially deleted in new
#chr02   13409   13556   "+4:YBL106C;"
##Partially deleted in new
#chr02   38854   39001   "+14*:YBL099W;-1:YBL098W;"
##Partially deleted in new
#chr02   382627  382774  "+5*:YBR072W;+3*:YBR072C-A;"
##Partially deleted in new
#chr02   392501  392648  "-1:YBR077C;"
##Partially deleted in new
#chr02   813075  813222  ND
##Partially deleted in new
#chr03   316488  316635  "-1:YCR108C;"
##Partially deleted in new
#chr06   118502  118649  ND
##Partially deleted in new
#chr06   270089  270236  ND
##Partially deleted in new
#chr08   212226  212373  ND
##Partially deleted in new
#chr09   439771  439918  ND
##Partially deleted in new
#chr10   97404   97551   "+1:CUT672;-1:YJL172W;"
##Partially deleted in new
#chr10   613775  613922  "+9:YJR098C;"
##Partially deleted in new
#chr11   190399  190546  "+7:YKL134C;"
##Partially deleted in new
#chr11   199291  199438  "+8:YKL129C;"
##Partially deleted in new
#chr15   49878   50025   "+18:YOL145C;"
##Partially deleted in new
#chr15   50032   50179   "+17:YOL145C;"
##Partially deleted in new
#chr15   220107  220254  "+6:YOL058W;+3:anti167;"



## to latin chr numbers
python ~/ms_tools/convert_chr_names2.py Pugh_measured_nucleosomes_roman0_sacCer3.bed -f roman0 -t latin > Pugh_measured_nucleosomes_sacCer3.bed
```


#### load the nucleosomes
```{r}
(nucs <- read_tsv('~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_nucleosomes_sacCer3.bed',
                  col_names = c('chr', 'start', 'end', 'position')))
```


#### uniquely assigned nucleosomes

some nucleosomes cover 2 annotations. Typically sense/antisense pairs, but this is based on SGD annotations (rather old ones given the **age** of the data?) and hence also sense overlaps exist. To make interpretation easier and more stringent I only use nucleosomes only overlapping a single annotation (ie they are asssigned only ie +1:... and not +1:geneX; -1:geneY in the position column).
Here is how this is achieved, includes also cleanup of the position formatting and spearating nucleosome position from annotation name:  
```{r}
(uniquely_assigned_nucs <- nucs %>%
   filter(position != "ND" & !grepl(';.', position)) %>%
   mutate(position = sub(';$', '', position)) %>%
   dplyr::select(chr, start, end, position) %>%
   separate(position, c('nuc', 'name'), sep=':') %>%
   mutate(nuc = ifelse(grepl('\\*', nuc), 'term', nuc)))
```


#### add annotation strand

To find out which strand the annotation covering the nucleosome is from we use an annotation file (this would be trivial for Y... gene names, but there is also SUTs,...).  
```{r}
(anno <- read_tsv('~/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin.bed', col_names = c('chr', 'start', 'end', 'anno', 'score', 'strand')) %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':'))
```


```{r}
(uniquely_assigned_nucs %<>% left_join(., 
                              dplyr::select(anno, name, strand)))
```

--> NAs are simply annotations not covered in the annotation file. Probably mostly pseudogenes not detected expressed in the Steinmetz annotation procedure. These are ambigous and we ignore those.

```{r}
table(uniquely_assigned_nucs$strand)
```

```{r}
sum(is.na(uniquely_assigned_nucs$strand))
```


#### save uniquely assigned nucleosomes with strand
```{r}
filter(uniquely_assigned_nucs, !is.na(strand)) %>%
  mutate(nuc = paste(nuc, name, sep=':'),
         name = 0) %>%
  write.table(., file='~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_nucleosomes_sacCer3_unique_with_strand.bed', col.names = FALSE, row.names = FALSE, quote=FALSE, sep='\t')
```


#### metagene profile using deeptools

```{bash, eval = FALSE}
cd ~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/
mkdir deeptools_out/nucleosomes

bed="~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_nucleosomes_sacCer3_unique_with_strand.bed"


awk '$4 ~ /\+1/' $bed | wc -l
# 9893
awk '$4 ~ /\-1/' $bed | wc -l
# 1399

awk '$4 ~ /\+1:/' $bed > ${bed/_measured_nucleosomes/_measured_plus1_nucleosomes}

### +1 nucleosomes vs both strands ###
plus1_bed="~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus1_nucleosomes_sacCer3_unique_with_strand.bed"

plus_bw="*_Mex67*_0_1*plus*.bw"
minus_bw="*_Mex67*_0_1*minus*.bw"
sample_names=$(echo $plus_bw | sed s/norm_//g | sed s/Mex67AA_//g | sed s/_0_1//g | sed s/0_1//g | sed s/_BGsub//g | sed s/_plus_KevinRoyAfiltered.bw//g)


bash ~/ms_tools/MS_Metagene_Tools/computeMatrixStrandedX.sh reference-point \
$plus1_bed "$plus_bw" "$minus_bw" \
1000 1000 TSS deeptools_out/nucleosomes/Mex_0_1_plus1nuc_startpm1000 "--binSize=10 --missingDataAsZero"

#log2
python ~/ms_tools/MS_Metagene_Tools/log2_matrix.py deeptools_out/nucleosomes/Mex_0_1_plus1nuc_startpm1000_joined.gz

plotHeatmap -m deeptools_out/nucleosomes/Mex_0_1_plus1nuc_startpm1000_log2.gz \
      --sortUsing region_length --colorMap Blues --missingDataColor white --zMax -1 \
      --samplesLabel $sample_names \
      -out deeptools_out/nucleosomes/Mex_0_1_plus1nuc_startpm1000_joined_log2_heatmap.pdf

plotProfile -m deeptools_out/nucleosomes/Mex_0_1_plus1nuc_startpm1000_joined_log2.gz \
-out deeptools_out/nucleosomes/Mex_0_1_plus1nuc_startpm1000_joined_log2_profile.pdf \
--numPlotsPerRow 2 \
--outFileNameData deeptools_out/nucleosomes/Mex_0_1_plus1nuc_startpm1000_joined_log2_profile.tab


##pA masked
### ups: not part of paper draft
cd only_pAminus_positions/
plus_bw="*_Mex67*_0_1*plus*.bw"
minus_bw="*_Mex67*_0_1*minus*.bw"
sample_names=$(echo $plus_bw | sed s/norm_//g | sed s/Mex67AA_//g | sed s/_0_1//g | sed s/_wo_noPaP_0rapa_sites//g | sed s/_BGsub//g | sed s/_plus_KevinRoyAfiltered.bw//g)


bash ~/ms_tools/MS_Metagene_Tools/computeMatrixStrandedX.sh reference-point \
$plus1_bed "$plus_bw" "$minus_bw" \
1000 1000 TSS ../deeptools_out/nucleosomes/masked_Mex_0_1_plus1nuc_startpm1000 "--binSize=10 --missingDataAsZero"

#log2
python ~/ms_tools/MS_Metagene_Tools/log2_matrix.py ../deeptools_out/nucleosomes/masked_Mex_0_1_plus1nuc_startpm1000_joined.gz

plotProfile -m ../deeptools_out/nucleosomes/masked_Mex_0_1_plus1nuc_startpm1000_joined_log2.gz \
-out ../deeptools_out/nucleosomes/masked_Mex_0_1_plus1nuc_startpm1000_joined_log2_profile.pdf \
--numPlotsPerRow 2 --plotType se \
--outFileNameData ../deeptools_out/nucleosomes/masked_Mex_0_1_plus1nuc_startpm1000_joined_log2_profile.tab
```



## 3seq data analysis R

#### common plot fun

common theme for prettier and plot funs for annotated nucleosome array plots
```{r}
nuc_borders <- data.frame(
  name = c('dn1', 'dn2', 'dn3', 'dn4'),
  border = c(0, 165, 330, 495)
)

nucleosome_rel_nuc_theme <- list(scale_x_continuous(minor_breaks=NULL, breaks=c(-1000, nuc_borders$border, 1000)),
  geom_vline(xintercept = nuc_borders$border, color='orange', linetype=2),
  geom_vline(xintercept = c(-1000,1000), color='gray', linetype=2),
  geom_vline(xintercept = 0, color='black'),
  geom_ribbon(aes(ymin=conf.low, ymax=conf.high), fill='gray', color=NA),
  geom_ribbon(aes(ymin=conf.low, ymax=conf.high), fill='gray', color=NA),
  geom_line(),
  theme_bw())
```

#### load to R

```{r}
(nuc1 <- RMetaTools::load_deeptoolsmatrix('~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/deeptools_out/nucleosomes/Mex_0_1_plus1nuc_startpm1000_joined.gz'))

nuc1 %<>%
  mutate(sample_name = sub('norm_', '', sample_name) %>%
           sub('_BGsub', '', .) %>%
           sub('_KevinRoyAfiltered', '', .)) %>%
  separate(sample_name, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_') %>%
  mutate(pA = case_when(Pap == 'noPap' ~ 'pA+', 
                        Pap == 'xPap' ~ 'pA+&-',
                        Pap == 'xPappAminus' ~ 'pA-'))
```


UPS: masked analysis not part of paper draft !
```{r}
(nuc1_masked <- RMetaTools::load_deeptoolsmatrix('~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bw/deeptools_out/nucleosomes/masked_Mex_0_1_plus1nuc_startpm1000_joined.gz'))


nuc1_masked %<>%
  mutate(sample_name = sub('norm_', '', sample_name) %>%
           sub('_BGsub', '', .) %>%
           sub('_KevinRoyAfiltered', '', .) %>%
           sub('_wo_noPaP_0rapa_sites', '', .)) %>%
  separate(sample_name, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_') %>%
  mutate(pA = 'pA-X')
```


combine the above

```{r}
nuc1 %<>% bind_rows(., nuc1_masked)
```

add pseudocount and log-scale

```{r}
(min_val <- filter(nuc1, value > 0) %$% value %>% min)
```

```{r}
nuc1 %<>%
  mutate(log2_value = ifelse(value < 0 | is.na(value), log2(min_val), log2(value + min_val)))
```

#### plot

```{r Nuc 2min data rel plus1 nucleosomes}
nuc1 %>%
  group_by(rel_pos, pA, strain, fraction, rapa, rep) %>%
  filter(is.finite(log2_value)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(pA+fraction~., scales='free') + 
  nucleosome_rel_nuc_theme
```

#### long TUs only

```{r}
long_names <- filter(anno, (end-start) > 1000) %$% name
```


```{r Nuc 2min data rel plus1 nucleosomes long ORFs}
nuc1 %>%
  group_by(rel_pos, pA, strain,fraction,rapa,rep) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), id %in% long_names, grepl('^Y', id)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(pA+fraction~., scales='free') + 
  nucleosome_rel_nuc_theme
```

n in this plot:
```{r}
nuc1 %>%
  group_by(rel_pos, pA, strain,fraction,rapa,rep) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), id %in% long_names, grepl('^Y', id)) %>%
  summarise(cnt = n()) %>%
  ungroup %>%
  distinct(cnt)
```


#### divergent mRNAs only

```{r}
(div_orfs <- read_tsv('~/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_divergent_ORFTs_pairs_leq500bp.bed', col_names = c('chr', 'start', 'end', 'anno', 'score', 'strand')) %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':'))
```


```{r Nuc 2min data rel plus1 nucleosomes divergent ORFs}
nuc1 %>%
  group_by(rel_pos, pA, strain,fraction,rapa,rep) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), id %in% div_orfs$name) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(pA+fraction~., scales='free') + 
  nucleosome_rel_nuc_theme
```


#### tandem mRNAs only

```{r}
(tandem_orfs <- read_tsv('~/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_tandem_ORFTs_pairs_leq500bp.bed', col_names = c('chr', 'start', 'end', 'anno', 'score', 'strand')) %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':'))
```


```{r Nuc 2min data rel plus1 nucleosomes tandem ORFs}
nuc1 %>%
  group_by(rel_pos, pA, strain,fraction,rapa,rep) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), id %in% tandem_orfs$name) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(pA+fraction~., scales='free') + 
  nucleosome_rel_nuc_theme
```



## Published Txn

### CRAC

```{bash, eval = FALSE}
cd ~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Tollervey_RNAP_CRAC/

plus_bw="bw/*_plus.bw"
minus_bw="bw/*_minus.bw"

sample_names=$(echo $plus_bw | sed s/_nabFT_plus.bw//g | sed s/GSM1706520_//g | sed s/GSM1706521_//g | sed s/bw\\///g)


plus1_bed="~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus1_nucleosomes_sacCer3_unique_with_strand.bed"

bash ~/ms_tools/MS_Metagene_Tools/computeMatrixStrandedX.sh reference-point \
$plus1_bed "$plus_bw" "$minus_bw" \
1000 1000 TSS deeptools_out/CRAC_plus1nuc_startpm1000 "--binSize=50 --missingDataAsZero"
```

#### load to R

```{r}
(crac <- RMetaTools::load_deeptoolsmatrix('~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Tollervey_RNAP_CRAC/deeptools_out/CRAC_plus1nuc_startpm1000_joined.gz') %>%
   mutate(log2_value = log2(value)))
```

```{r CRAC rel plus1 nucleosome}
crac %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```


#### long ORFs
```{r CRAC rel plus1 nucleosome long ORFs}
crac %>%
  group_by(rel_pos, sample_name) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), id %in% long_names, grepl('^Y', id)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```

n with positive values for this:
```{r}
crac %>%
  group_by(rel_pos, sample_name) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), id %in% long_names, grepl('^Y', id)) %>%
  summarize(cnt=n()) %>%
  ungroup %$%
  table(cnt)
```

#### SUTs
```{r CRAC rel plus1 nucleosome long SUTs}
crac %>%
  group_by(rel_pos, sample_name) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), id %in% long_names, grepl('^SUT', id)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```


#### CUTs
```{r CRAC rel plus1 nucleosome CUTs}
crac %>%
  group_by(rel_pos, sample_name) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), grepl('^CUT', id)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```



### NETseq

```{bash, eval = FALSE}
cd ~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Churchman_NETseq

plus_bw="bw/GSM617027_WT_NC_plus_sacCer3.bw bw/GSM617030_DST1D_plus_sacCer3.bw"
minus_bw=${plus_bw//plus/minus}

sample_names=$(echo $plus_bw | sed s/_plus_sacCer3.bw//g | sed s/bw\\///g)


plus1_bed="~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus1_nucleosomes_sacCer3_unique_with_strand.bed"

bash ~/ms_tools/MS_Metagene_Tools/computeMatrixStrandedX.sh reference-point \
$plus1_bed "$plus_bw" "$minus_bw" \
1000 1000 TSS deeptools_out/NETseq_plus1nuc_startpm1000 "--binSize=50 --missingDataAsZero"
```

#### load to R

```{r}
(netseq <- RMetaTools::load_deeptoolsmatrix('~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Churchman_NETseq/deeptools_out/NETseq_plus1nuc_startpm1000_joined.gz') %>%
   mutate(log2_value = log2(value)))
```

```{r NETseq rel plus1 nucleosome}
netseq %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```


```{r NETseq rel plus1 nucleosome long ORFs}
netseq %>%
  group_by(rel_pos, sample_name) %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(is.finite(log2_value), id %in% long_names, grepl('^Y', id)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```

n for this:
```{r}
netseq %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(id %in% long_names, grepl('^Y', id)) %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  summarize(cnt=n()) %>%
  ungroup %$%
  table(cnt)
```



### ChIPseq

```{bash, eval=FALSE}
cd ~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Henikoff_Rpb3FLAG_ChIPseq

bw="GSM2551210_WT_A_140.bw"

sample_names=$(echo $bw | sed s/_A_140.bw//g)

###+1 nucleosome
plus1_bed="~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus1_nucleosomes_sacCer3_unique_with_strand.bed"

computeMatrix reference-point -R $plus1_bed -S $bw \
-b 1000 -a 1000 --referencePoint TSS \
-out deeptools_out/ChIPseq_plus1nuc_startpm1000.gz \
--binSize 10 --missingDataAsZero

```

#### load to R

```{r}
(chipseq <- RMetaTools::load_deeptoolsmatrix('~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Henikoff_Rpb3FLAG_ChIPseq/deeptools_out/ChIPseq_plus1nuc_startpm1000.gz') %>%
   mutate(log2_value = log2(value)))
```


```{r ChIPseq rel plus1 nucleosome}
chipseq %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```

```{r ChIPseq rel plus1 nucleosome long ORFs}
chipseq %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(id %in% long_names, grepl('^Y', id)) %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```

n for this:
```{r}
chipseq %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(id %in% long_names, grepl('^Y', id)) %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  summarize(cnt=n()) %>%
  ungroup %$%
  table(cnt)
```

```{r ChIPseq rel plus1 nucleosome long SUTs}
chipseq %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(id %in% long_names, grepl('^SUT', id)) %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```

```{r ChIPseq rel plus1 nucleosome SUTs}
chipseq %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(grepl('^SUT', id)) %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```


```{r ChIPseq rel plus1 nucleosome CUTs}
chipseq %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  filter(grepl('^CUT', id)) %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(log2_value)) %>%
  do(tidy(t.test(.$log2_value))) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + 
  nucleosome_rel_nuc_theme
```



## Nucleosomes density relative to +1

### make a nucleosome track file
```{bash, eval = FALSE}
awk '{OFS="\t"}{$4=1; print $0}' Pugh_measured_nucleosomes_sacCer3.bed > Pugh_measured_nucleosomes_binary_sacCer3.bedgraph
~/ms_tools/bedGraphToBigWig Pugh_measured_nucleosomes_binary_sacCer3.bedgraph ~/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_chrom.sizes Pugh_measured_nucleosomes_sacCer3.bw

```

### metagene profile
```{bash, eval = FALSE}
cd ~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes


### nucleosomes metagene around +1 nucleosome ###
plus1_bed="~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_plus1_nucleosomes_sacCer3_unique_with_strand.bed"


computeMatrix reference-point \
-S Pugh_measured_nucleosomes_sacCer3.bw \
-R $plus1_bed \
-a 1000 -b 1000 --referencePoint TSS --binSize 10 --missingDataAsZero \
-out Pugh_measured_nucleosomes_rel_plus1nuc.gz

plotHeatmap -m Pugh_measured_nucleosomes_rel_plus1nuc.gz --yMin 0 --yMax 1 -o Pugh_measured_nucleosomes_rel_plus1nuc_htmp.pdf
```


```{r}
(nuc_relplus1 <- RMetaTools::load_deeptoolsmatrix('~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/Pugh_measured_nucleosomes_rel_plus1nuc.gz'))
```

```{r All nucs rel plus1 mRNA +1 nucleosome}
nuc_relplus1 %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(value)) %>%
  mutate(estimate = mean(value),
         conf.low=estimate, conf.high=estimate) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + theme_bw() +
  nucleosome_rel_nuc_theme
```

```{r all nucs rel plus1 long mRNA +1 nucleosome}
nuc_relplus1 %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(value), id %in% long_names, grepl('^Y', id)) %>%
  mutate(estimate = mean(value),
         conf.low=estimate, conf.high=estimate) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + theme_bw() +
  nucleosome_rel_nuc_theme
```

n for this:
```{r}
nuc_relplus1 %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(value), id %in% long_names, grepl('^Y', id)) %>%
  summarize(cnt=n()) %>%
  ungroup %>%
  distinct(cnt)
```

### TSS position relative to +1 nucleosome

```{bash, eval = FALSE}
ORFs="~/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin_ORFTs.bed"

awk '{OFS="\t"}{if($6=="+"){print $1"\t"$2"\t"$2+1"\t1"}}' $ORFs | sort -k1,1 -k2,2n > ${ORFs/.bed/_TSS_plus.bedgraph}
awk '{OFS="\t"}{if($6=="-"){print $1"\t"$3-1"\t"$3"\t1"}}' $ORFs | sort -k1,1 -k2,2n > ${ORFs/.bed/_TSS_minus.bedgraph}

~/ms_tools/bedGraphToBigWig ${ORFs/.bed/_TSS_plus.bedgraph} ~/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_chrom.sizes ${ORFs/.bed/_TSS_plus.bw}

~/ms_tools/bedGraphToBigWig ${ORFs/.bed/_TSS_minus.bedgraph} ~/Documents/genomewide_datasets/annotations/sacCer3/sacCer3_chrom.sizes ${ORFs/.bed/_TSS_minus.bw}

bash ~/ms_tools/MS_Metagene_Tools/computeMatrixStrandedX.sh reference-point \
$plus1_bed "${ORFs/.bed/_TSS_plus.bw}" "${ORFs/.bed/_TSS_minus.bw}" \
1000 1000 TSS ORF_TSSs_rel_plus1nuc "--binSize=10 --missingDataAsZero"

plotHeatmap -m ORF_TSSs_rel_plus1nuc_joined.gz -o ORF_TSSs_rel_plus1nuc_htmp.pdf
```


```{r}
(tsss_relplus1 <- RMetaTools::load_deeptoolsmatrix('~/Documents/genomewide_datasets/ChIP_CLiP_yeast/Pugh_nucleosomes/ORF_TSSs_rel_plus1nuc_joined.gz'))
```

```{r TSSs rel plus1 mRNA +1 nucleosome}
tsss_relplus1 %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(value)) %>%
  mutate(estimate = mean(value),
         conf.low=estimate, conf.high=estimate) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + theme_bw() +
  nucleosome_rel_nuc_theme
```

```{r TSSs rel plus1 long mRNA +1 nucleosome}
tsss_relplus1 %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(value), id %in% long_names, grepl('^Y', id)) %>%
  mutate(estimate = mean(value),
         conf.low=estimate, conf.high=estimate) %>%
ggplot(., aes(x=rel_pos, y=estimate)) +
  facet_grid(sample_name~., scales='free') + theme_bw() +
  nucleosome_rel_nuc_theme
```

n for this:
```{r}
tsss_relplus1 %>%
  separate(id, c('nuc', 'id'), sep=':') %>%
  group_by(rel_pos, sample_name) %>%
  filter(is.finite(value), id %in% long_names, grepl('^Y', id)) %>%
  summarize(cnt=n()) %>%
  ungroup %>%
  distinct(cnt)
```

```{r}
sessionInfo()
```
